--- dvgrab-1.2/avi.cc~	2003-02-22 20:02:12.000000000 -0600
+++ dvgrab-1.2/avi.cc	2003-08-09 16:31:51.000000000 -0500
@@ -102,6 +102,7 @@
 #include <iostream>
 #include <strstream>
 #include <iomanip>
+#include <cassert>
 
 using std::cout;
 using std::hex;
--- dvgrab-1.2/filehandler.cc~	2003-02-22 18:35:51.000000000 -0600
+++ dvgrab-1.2/filehandler.cc	2003-08-09 16:33:40.000000000 -0500
@@ -38,6 +38,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/stat.h>
+#include <cassert>
 
 #include "filehandler.h"
 #include "error.h"
--- dvgrab-1.2/riff.cc~	2003-02-22 20:02:12.000000000 -0600
+++ dvgrab-1.2/riff.cc	2003-08-09 16:32:46.000000000 -0500
@@ -65,6 +65,7 @@
 #include <strstream>
 #include <iomanip>
 #include <byteswap.h>
+#include <cassert>
 
 using std::cout;
 using std::hex;
