diff -uprN perl-5.8.1.old/installperl perl-5.8.1/installperl
--- perl-5.8.1.old/installperl	2003-10-02 17:44:52.000000000 -0700
+++ perl-5.8.1/installperl	2003-10-02 17:46:53.000000000 -0700
@@ -836,7 +836,7 @@ sub installlib {
 	my $xname = "$installlib/$name";
 	$xname =~ s/^\Q$destdir\E// if $destdir;
 	$packlist->{$xname} = { type => 'file' };
-	if ($force || compare($_, "$installlib/$name") || $nonono) {
+	if (1) {
 	    unlink("$installlib/$name");
 	    mkpath("$installlib/$dir", $verbose, 0777);
 	    # HP-UX (at least) needs to maintain execute permissions
@@ -895,7 +895,7 @@ sub copy_if_diff {
     }
     -f $from || $perlpodbadsymlink || warn "$0: $from not found";
     $packlist->{$xto} = { type => 'file' };
-    if ($force || compare($from, $to) || $nonono) {
+    if (1) {
 	safe_unlink($to);   # In case we don't have write permissions.
 	if ($nonono) {
 	    $from = $depth . "/" . $from if $depth;
