diff -Nur openbox-2.3.1/util/epist/keytree.hh openbox-2.3.1-gcc33fix/util/epist/keytree.hh
--- openbox-2.3.1/util/epist/keytree.hh	2003-03-22 01:56:01.000000000 +0100
+++ openbox-2.3.1-gcc33fix/util/epist/keytree.hh	2003-08-16 17:27:36.000000000 +0200
@@ -24,6 +24,7 @@
 #define _keytree_hh
 
 #include <list>
+#include <cassert>
 #include "../../src/Timer.hh"
 #include "actions.hh"
 #include "screen.hh"
