--- libsexy/sexy-url-label.c.orig	2005-10-24 20:44:55.000000000 -0400
+++ libsexy/sexy-url-label.c	2005-10-24 20:45:23.000000000 -0400
@@ -45,7 +45,6 @@
 
 	int layout_x;
 	int layout_y;
-	int wrap_width;
 
 	size_t wrap_width;
 
@@ -68,8 +67,6 @@
 										GtkRequisition *req);
 static void sexy_url_label_size_allocate(GtkWidget *widget,
 										 GtkAllocation *allocation);
-static void sexy_url_label_size_request(GtkWidget *widget,
-										GtkRequisition *req);
 static gboolean sexy_url_label_motion_notify_event(GtkWidget *widget,
 												   GdkEventMotion *event);
 static gboolean sexy_url_label_leave_notify_event(GtkWidget *widget,
@@ -85,8 +82,6 @@
 static void sexy_url_label_clear_links(SexyUrlLabel *url_label);
 static void sexy_url_label_clear_urls(SexyUrlLabel *url_label);
 static void sexy_url_label_rescan_label(SexyUrlLabel *url_label);
-static void sexy_url_label_set_wrap_width(SexyUrlLabel *url_label, int width);
-static void sexy_url_label_rewrap(SexyUrlLabel *url_label);
 
 static GtkLabelClass *parent_class = NULL;
 static guint signals[LAST_SIGNAL] = {0};
@@ -109,7 +104,6 @@
 	widget_class->unmap               = sexy_url_label_unmap;
 	widget_class->size_request        = sexy_url_label_size_request;
 	widget_class->size_allocate       = sexy_url_label_size_allocate;
-	widget_class->size_request        = sexy_url_label_size_request;
 	widget_class->motion_notify_event = sexy_url_label_motion_notify_event;
 	widget_class->leave_notify_event  = sexy_url_label_leave_notify_event;
 	widget_class->button_press_event  = sexy_url_label_button_press_event;
@@ -134,9 +128,6 @@
 	GtkWidget *item;
 	GtkWidget *image;
 
-	pango_layout_set_wrap(gtk_label_get_layout(GTK_LABEL(url_label)),
-						  PANGO_WRAP_WORD_CHAR);
-
 	priv->links        = NULL;
 	priv->active_link  = NULL;
 	priv->event_window = NULL;
@@ -407,7 +398,6 @@
 							   allocation->width, allocation->height);
 	}
 
-	sexy_url_label_set_wrap_width(url_label, allocation->width);
 	sexy_url_label_rescan_label(url_label);
 
 	pango_layout_get_pixel_size(gtk_label_get_layout(GTK_LABEL(widget)),
@@ -417,17 +407,6 @@
 }
 
 static void
-sexy_url_label_size_request(GtkWidget *widget, GtkRequisition *req)
-{
-	int height;
-
-	pango_layout_get_pixel_size(gtk_label_get_layout(GTK_LABEL(widget)),
-								NULL, &height);
-	req->width  = 0;
-	req->height = height;
-}
-
-static void
 open_link_activate_cb(GtkMenuItem *menu_item, SexyUrlLabel *url_label)
 {
 	SexyUrlLabelPrivate *priv = SEXY_URL_LABEL_GET_PRIVATE(url_label);
@@ -751,29 +730,6 @@
 	return FALSE;
 }
 
-static void
-sexy_url_label_set_wrap_width(SexyUrlLabel *url_label, int width)
-{
-	SexyUrlLabelPrivate *priv = SEXY_URL_LABEL_GET_PRIVATE(url_label);
-
-	pango_layout_set_width(gtk_label_get_layout(GTK_LABEL(url_label)),
-						   width * PANGO_SCALE);
-
-	if (priv->wrap_width != width)
-	{
-		priv->wrap_width = width;
-		gtk_widget_queue_resize(GTK_WIDGET(url_label));
-	}
-}
-
-static void
-sexy_url_label_rewrap(SexyUrlLabel *url_label)
-{
-	SexyUrlLabelPrivate *priv = SEXY_URL_LABEL_GET_PRIVATE(url_label);
-
-	sexy_url_label_set_wrap_width(url_label, priv->wrap_width);
-}
-
 /**
  * sexy_url_label_set_markup
  * @url_label: A #SexyUrlLabel.
@@ -809,7 +765,6 @@
 	else
 		gtk_label_set_markup(GTK_LABEL(url_label), "");
 
-	sexy_url_label_rewrap(url_label);
 	sexy_url_label_rescan_label(url_label);
 
 	update_wrap_width(url_label, priv->wrap_width);
