diff -Nur transcode-0.6.12/ffmpeg/libavcodec/i386/dsputil_mmx.c transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/dsputil_mmx.c
--- transcode-0.6.12/ffmpeg/libavcodec/i386/dsputil_mmx.c	2003-12-30 11:02:10.000000000 +0100
+++ transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/dsputil_mmx.c	2004-05-20 21:33:23.264585758 +0200
@@ -31,8 +31,8 @@
 static const uint64_t mm_wone __attribute__ ((aligned(8))) = 0x0001000100010001ULL;
 static const uint64_t mm_wtwo __attribute__ ((aligned(8))) = 0x0002000200020002ULL;
 
-static const uint64_t ff_pw_20 __attribute__ ((aligned(8))) = 0x0014001400140014ULL;
-static const uint64_t ff_pw_3  __attribute__ ((aligned(8))) = 0x0003000300030003ULL;
+const uint64_t ff_pw_20 __attribute__ ((aligned(8))) = 0x0014001400140014ULL;
+const uint64_t ff_pw_3  __attribute__ ((aligned(8))) = 0x0003000300030003ULL;
 static const uint64_t ff_pw_16 __attribute__ ((aligned(8))) = 0x0010001000100010ULL;
 static const uint64_t ff_pw_15 __attribute__ ((aligned(8))) = 0x000F000F000F000FULL;
 
diff -Nur transcode-0.6.12/ffmpeg/libavcodec/i386/motion_est_mmx.c transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/motion_est_mmx.c
--- transcode-0.6.12/ffmpeg/libavcodec/i386/motion_est_mmx.c	2003-12-30 11:02:10.000000000 +0100
+++ transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/motion_est_mmx.c	2004-05-20 21:33:31.073590458 +0200
@@ -26,7 +26,7 @@
 0x0002000200020002ULL,
 };
 
-static __attribute__ ((aligned(8), unused)) uint64_t bone= 0x0101010101010101LL;
+__attribute__ ((aligned(8), unused)) uint64_t bone= 0x0101010101010101LL;
 
 static inline void sad8_mmx(uint8_t *blk1, uint8_t *blk2, int stride, int h)
 {
diff -Nur transcode-0.6.12/ffmpeg/libavcodec/i386/simple_idct_mmx.c transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/simple_idct_mmx.c
--- transcode-0.6.12/ffmpeg/libavcodec/i386/simple_idct_mmx.c	2003-12-20 12:49:12.000000000 +0100
+++ transcode-0.6.12-fixed/ffmpeg/libavcodec/i386/simple_idct_mmx.c	2004-05-20 21:35:51.661671740 +0200
@@ -45,8 +45,8 @@
 #define ROW_SHIFT 11
 #define COL_SHIFT 20 // 6
 
-static const uint64_t __attribute__((aligned(8))) wm1010= 0xFFFF0000FFFF0000ULL;
-static const uint64_t __attribute__((aligned(8))) d40000= 0x0000000000040000ULL;
+const uint64_t __attribute__((aligned(8))) wm1010= 0xFFFF0000FFFF0000ULL;
+const uint64_t __attribute__((aligned(8))) d40000= 0x0000000000040000ULL;
 
 static const int16_t __attribute__((aligned(8))) coeffs[]= {
 	1<<(ROW_SHIFT-1), 0, 1<<(ROW_SHIFT-1), 0,
diff -Nur transcode-0.6.12/filter/subtitler/frame_list.c transcode-0.6.12-fixed/filter/subtitler/frame_list.c
--- transcode-0.6.12/filter/subtitler/frame_list.c	2004-01-05 14:12:55.000000000 +0100
+++ transcode-0.6.12-fixed/filter/subtitler/frame_list.c	2004-05-20 21:46:13.295441127 +0200
@@ -33,7 +33,7 @@
 
 char *strsave(char *s) /*save char array s somewhere*/
 {
-char *p, *malloc();
+char *p;
 if(p = malloc( strlen(s) +  1) ) strcpy(p, s);
 return(p);
 }
