diff -Nur BitTorrent-3.4.2/BitTorrent/zurllib.py BitTorrent-3.4.2.fix/BitTorrent/zurllib.py
--- BitTorrent-3.4.2/BitTorrent/zurllib.py	2003-06-27 02:40:21.000000000 +0200
+++ BitTorrent-3.4.2.fix/BitTorrent/zurllib.py	2004-12-05 15:34:21.423307728 +0100
@@ -35,7 +35,10 @@
         if DEBUG: 
              pprint.pprint(headers.dict)
         url = fp.url
-        return addinfourldecompress(fp, headers, url)
+        resp = addinfourldecompress(fp, headers, url)
+        resp.code = fp.code
+        resp.msg = fp.msg
+        return resp
 
 
 class addinfourldecompress(addinfourl):
