--- Makefile.in	2004-07-16 07:19:14.177751128 -0700
+++ Makefile.in	2004-07-16 07:19:41.577585720 -0700
@@ -602,9 +602,9 @@
 	tags tags-recursive uninstall uninstall-am uninstall-info-am
 
 
-rpm:    Makefile
-       make dist
-       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
+rpm:   Makefile \
+       make dist \
+       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz \
        rm $(PACKAGE)-$(VERSION).tar.gz
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
