diff -Naur kdevelop-3.1.2.orig/acinclude.m4 kdevelop-3.1.2/acinclude.m4
--- kdevelop-3.1.2.orig/acinclude.m4	2005-01-06 13:20:23.656702312 -0800
+++ kdevelop-3.1.2/acinclude.m4	2005-01-06 13:24:14.536603232 -0800
@@ -4254,11 +4254,13 @@
 
 AC_DEFUN([KDE_CHECK_PYTHON],
 [
-  KDE_CHECK_PYTHON_INTERN("2.3", 
-   [KDE_CHECK_PYTHON_INTERN("2.2", 
-     [KDE_CHECK_PYTHON_INTERN("2.1", 
-       [KDE_CHECK_PYTHON_INTERN("2.0", 
-         [KDE_CHECK_PYTHON_INTERN($1, $2) ])
+  KDE_CHECK_PYTHON_INTERN("2.4", 
+   [KDE_CHECK_PYTHON_INTERN("2.3", 
+     [KDE_CHECK_PYTHON_INTERN("2.2", 
+       [KDE_CHECK_PYTHON_INTERN("2.1", 
+         [KDE_CHECK_PYTHON_INTERN("2.0", 
+           [KDE_CHECK_PYTHON_INTERN($1, $2) ])
+         ])
        ])
      ])
    ])
diff -Naur kdevelop-3.1.2.orig/admin/acinclude.m4.in kdevelop-3.1.2/admin/acinclude.m4.in
--- kdevelop-3.1.2.orig/admin/acinclude.m4.in	2005-01-06 13:20:23.654702616 -0800
+++ kdevelop-3.1.2/admin/acinclude.m4.in	2005-01-06 13:24:36.611247376 -0800
@@ -4254,11 +4254,13 @@
 
 AC_DEFUN([KDE_CHECK_PYTHON],
 [
-  KDE_CHECK_PYTHON_INTERN("2.3", 
-   [KDE_CHECK_PYTHON_INTERN("2.2", 
-     [KDE_CHECK_PYTHON_INTERN("2.1", 
-       [KDE_CHECK_PYTHON_INTERN("2.0", 
-         [KDE_CHECK_PYTHON_INTERN($1, $2) ])
+  KDE_CHECK_PYTHON_INTERN("2.4", 
+   [KDE_CHECK_PYTHON_INTERN("2.3", 
+     [KDE_CHECK_PYTHON_INTERN("2.2", 
+       [KDE_CHECK_PYTHON_INTERN("2.1", 
+         [KDE_CHECK_PYTHON_INTERN("2.0", 
+           [KDE_CHECK_PYTHON_INTERN($1, $2) ])
+         ])
        ])
      ])
    ])
diff -Naur kdevelop-3.1.2.orig/configure kdevelop-3.1.2/configure
--- kdevelop-3.1.2.orig/configure	2005-01-06 13:20:23.650703224 -0800
+++ kdevelop-3.1.2/configure	2005-01-06 13:24:53.801634040 -0800
@@ -31994,7 +31994,7 @@
 if test -z ""2.3""; then
   version="1.5"
 else
-  version=""2.3""
+  version=""2.4""
 fi
 
 echo "$as_me:$LINENO: checking for Python$version" >&5
