--- ksensors/main.cpp.orig	Fri May 24 22:46:17 2002
+++ ksensors/main.cpp	Sun Aug  4 20:16:39 2002
@@ -46,14 +46,14 @@
   aboutData.addAuthor("Miguel Novas",0, "michaell@teleline.es");
   KCmdLineArgs::init( argc, argv, &aboutData );
   KCmdLineArgs::addCmdLineOptions( options ); // Add our own options.
-  KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
-  bool noDock    = !args->isSet("dock");
-  bool showSplash=  args->isSet("splash");
-  args->clear();
-
   KUniqueApplication::addCmdLineOptions();
+
   if(!KUniqueApplication::start())  return 0;
   KUniqueApplication a;
+ KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
+  bool noDock    = !args->isSet("dock");
+  bool showSplash=  args->isSet("splash");
+  args->clear();
   //
   KConfig *cfg= kapp->config();
   cfg->setGroup("General");
