diff -Nur util-linux-2.12.orig/disk-utils/blockdev.c util-linux-2.12/disk-utils/blockdev.c
--- util-linux-2.12.orig/disk-utils/blockdev.c	2002-03-08 23:57:02.000000000 +0100
+++ util-linux-2.12/disk-utils/blockdev.c	2004-02-11 22:07:23.000000000 +0100
@@ -9,12 +9,14 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <linux/version.h>
 
 #include "nls.h"
 
-/* Since it is impossible to include <linux/fs.h>, let us
-   give the ioctls explicitly. */
-
+#ifdef LIBC_HEADERS_VERSION
+/* Fixed headers detected */
+#include <linux/fs.h>
+#else
 #ifndef BLKROSET
 #define BLKROSET   _IO(0x12,93)
 #define BLKROGET   _IO(0x12,94)
@@ -27,6 +29,7 @@
 #define BLKBSZGET  _IOR(0x12,112,sizeof(int))
 #define BLKBSZSET  _IOW(0x12,113,sizeof(int))
 #endif
+#endif
 
 /* Maybe <linux/hdreg.h> could be included */
 #ifndef HDIO_GETGEO
diff -Nur util-linux-2.12.orig/disk-utils/elvtune.c util-linux-2.12/disk-utils/elvtune.c
--- util-linux-2.12.orig/disk-utils/elvtune.c	2002-03-08 23:57:49.000000000 +0100
+++ util-linux-2.12/disk-utils/elvtune.c	2004-02-11 22:07:23.000000000 +0100
@@ -27,6 +27,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include "nls.h"
+#include <linux/version.h>
 
 /* this has to match with the kernel structure */
 /* current version for ac19 and 2.2.16 */
@@ -37,8 +38,13 @@
 	int max_bomb_segments;
 } blkelv_ioctl_arg_t;
 
+#ifdef LIBC_HEADERS_VERSION 
+/* Fixed headers detected */
+#include <linux/fs.h>
+#else
 #define BLKELVGET   _IOR(0x12,106,sizeof(blkelv_ioctl_arg_t))
 #define BLKELVSET   _IOW(0x12,107,sizeof(blkelv_ioctl_arg_t))
+#endif
 
 static void
 usage(void) {
diff -Nur util-linux-2.12.orig/fdisk/common.h util-linux-2.12/fdisk/common.h
--- util-linux-2.12.orig/fdisk/common.h	2003-07-13 15:59:53.000000000 +0200
+++ util-linux-2.12/fdisk/common.h	2004-02-11 22:07:23.000000000 +0100
@@ -1,12 +1,18 @@
 /* common stuff for fdisk, cfdisk, sfdisk */
 
-/* including <linux/fs.h> fails */
 #include <sys/ioctl.h>
+#include <linux/version.h>
+
+#ifdef LIBC_HEADERS_VERSION
+/* Fixed headers detected */
+#include <linux/fs.h>
+#else
 #define BLKRRPART    _IO(0x12,95)    /* re-read partition table */
 #define BLKGETSIZE   _IO(0x12,96)    /* return device size */
 #define BLKFLSBUF    _IO(0x12,97)    /* flush buffer cache */
 #define BLKSSZGET    _IO(0x12,104)   /* get block device sector size */
 #define BLKGETSIZE64 _IOR(0x12,114,8)	/* 8 = sizeof(u64) */
+#endif
 
 /* including <linux/hdreg.h> also fails */
 struct hd_geometry {
diff -Nur util-linux-2.12.orig/mount/my_dev_t.h util-linux-2.12/mount/my_dev_t.h
--- util-linux-2.12.orig/mount/my_dev_t.h	2003-07-16 22:05:50.000000000 +0200
+++ util-linux-2.12/mount/my_dev_t.h	2004-02-11 22:08:20.000000000 +0100
@@ -4,4 +4,11 @@
 /* for ancient systems use "unsigned short" */
 
 #include <linux/posix_types.h>
+#include <linux/version.h>
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
 #define my_dev_t __kernel_dev_t
+#else
+#include <asm/posix_types.h>
+#define my_dev_t __kernel_old_dev_t
+#endif
