diff -urpN magic.old/magic.mime magic/magic.mime
--- magic.old/magic.mime	2004-07-26 08:28:15.216212592 -0700
+++ magic/magic.mime	2004-07-26 08:31:14.627937864 -0700
@@ -896,6 +896,14 @@
 # http://www.djvuzone.org/
 0	string		AT&TFORM	image/x.djvu
 
+#------------------------------------------------------------------------------
+# zisofs: file(1) magic for zisofs/RockRidge compressed files
+#
+# from H. Peter Anvin <hpa@zytor.com> May 4, 2001
+#
+0	string	\x37\xE4\x53\x96\xC9\xDB\xD6\x07	zisofs/Rockridge compressed file
+>8	lelong	x	- %d bytes
+
 # Danny Milosavljevic <danny.milo@gmx.net>
 # this are adrift (adventure game standard) game files, extension .taf
 # depending on version magic continues with 0x93453E6139FA (V 4.0)
