diff -ruN xosview-1.8.0/bitfieldmeter.cc xosview-1.8.0.patched/bitfieldmeter.cc
--- xosview-1.8.0/bitfieldmeter.cc	1999-11-12 16:13:21.000000000 +1100
+++ xosview-1.8.0.patched/bitfieldmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -5,7 +5,7 @@
 //  This file may be distributed under terms of the GPL
 //
 //
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include "snprintf.h"
 #include "general.h"
@@ -361,14 +361,14 @@
 bool BitFieldMeter::checkX(int x, int width) const {
   if ((x < x_) || (x + width < x_)
       || (x > x_ + width_) || (x + width > x_ + width_)){
-    cerr << "BitFieldMeter::checkX() : bad horiz values for meter : "
-         << name() << endl;
+    std::cerr << "BitFieldMeter::checkX() : bad horiz values for meter : "
+         << name() << std::endl;
 
-    cerr <<"value "<<x<<", width "<<width<<", total_ = "<<total_<<endl;
+    std::cerr <<"value "<<x<<", width "<<width<<", total_ = "<<total_<<std::endl;
 
     for (int i = 0 ; i < numfields_ ; i++)
-      cerr <<"fields_[" <<i <<"] = " <<fields_[i] <<",";
-    cerr <<endl;
+      std::cerr <<"fields_[" <<i <<"] = " <<fields_[i] <<",";
+    std::cerr <<std::endl;
 
     return false;
   }
diff -ruN xosview-1.8.0/bsd/btrymeter.cc xosview-1.8.0.patched/bsd/btrymeter.cc
--- xosview-1.8.0/bsd/btrymeter.cc	2001-10-09 12:38:09.000000000 +1000
+++ xosview-1.8.0.patched/bsd/btrymeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -12,7 +12,7 @@
 //  Only supported for NetBSD so far.
 #include "btrymeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 #include <fcntl.h>
@@ -54,7 +54,7 @@
   int loadinfo = open(APMFILENAME, O_RDONLY, 0);
 
   if ( !loadinfo ){
-    cerr <<"Can not open file : " <<APMFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<APMFILENAME <<std::endl;
     parent_->done(1);
     return;
   }
diff -ruN xosview-1.8.0/bsd/kernel.cc xosview-1.8.0.patched/bsd/kernel.cc
--- xosview-1.8.0/bsd/kernel.cc	2001-10-09 12:40:51.000000000 +1000
+++ xosview-1.8.0.patched/bsd/kernel.cc	2003-07-22 13:31:46.000000000 +1000
@@ -24,7 +24,7 @@
 #include <fcntl.h>
 #include <kvm.h>
 #include <limits.h>		/*  For _POSIX2_LINE_MAX  */
-#include <string.h>		/*  For strncpy().  */
+#include <string>		/*  For strncpy().  */
 
 #include <err.h>                /*  For err(), warn(), etc.  BCG  */
 #include <errno.h>
@@ -54,7 +54,7 @@
 
 #include <sys/param.h>	/*  Needed by both UVM and swapctl stuff.  */
 #if defined(UVM)
-#include <string.h>
+#include <string>
 #include <sys/malloc.h>
 #include <sys/sysctl.h>
 #include <sys/device.h>
diff -ruN xosview-1.8.0/bsd/loadmeter.cc xosview-1.8.0.patched/bsd/loadmeter.cc
--- xosview-1.8.0/bsd/loadmeter.cc	1999-11-19 21:38:24.000000000 +1100
+++ xosview-1.8.0.patched/bsd/loadmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -56,7 +56,7 @@
     //  prevent this whole problem, the load meter can not be a decay
     //  meter.  The load is a decaying average kind of thing anyway,
     //  so having a decaying load average is redundant.
-    cerr << "Warning:  The loadmeter can not be configured as a decay\n"
+    std::cerr << "Warning:  The loadmeter can not be configured as a decay\n"
          << "  meter.  See the source code (" << __FILE__ << ") for further\n"
          << "  details.\n";
     dodecay_ = 0;
diff -ruN xosview-1.8.0/bsd/swapinternal.cc xosview-1.8.0.patched/bsd/swapinternal.cc
--- xosview-1.8.0/bsd/swapinternal.cc	1999-11-17 18:19:26.000000000 +1100
+++ xosview-1.8.0.patched/bsd/swapinternal.cc	2003-07-22 13:31:46.000000000 +1000
@@ -82,7 +82,7 @@
 # include <stdio.h>		/*  For error messages.  */
 #endif
 #include <stdlib.h>		/*  For malloc().  */
-#include <string.h>		/*  For bzero().  */
+#include <string>		/*  For bzero().  */
 
 #ifdef XOSVIEW_BSDI      /* some more include files */
 #include <err.h>
diff -ruN xosview-1.8.0/fieldmeter.cc xosview-1.8.0.patched/fieldmeter.cc
--- xosview-1.8.0/fieldmeter.cc	1999-11-07 09:48:16.000000000 +1100
+++ xosview-1.8.0.patched/fieldmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -6,7 +6,7 @@
 //
 // $Id$
 //
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include "snprintf.h"
 #include "general.h"
@@ -302,14 +302,14 @@
 bool FieldMeter::checkX(int x, int width) const {
   if ((x < x_) || (x + width < x_)
       || (x > x_ + width_) || (x + width > x_ + width_)){
-    cerr << "FieldMeter::checkX() : bad horiz values for meter : "
-         << name() << endl;
+    std::cerr << "FieldMeter::checkX() : bad horiz values for meter : "
+         << name() << std::endl;
 
-    cerr <<"value "<<x<<", width "<<width<<", total_ = "<<total_<<endl;
+    std::cerr <<"value "<<x<<", width "<<width<<", total_ = "<<total_<<std::endl;
 
     for (int i = 0 ; i < numfields_ ; i++)
-      cerr <<"fields_[" <<i <<"] = " <<fields_[i] <<",";
-    cerr <<endl;
+      std::cerr <<"fields_[" <<i <<"] = " <<fields_[i] <<",";
+    std::cerr <<std::endl;
 
     return false;
   }
diff -ruN xosview-1.8.0/fieldmeterdecay.cc xosview-1.8.0.patched/fieldmeterdecay.cc
--- xosview-1.8.0/fieldmeterdecay.cc	2001-10-11 01:58:22.000000000 +1000
+++ xosview-1.8.0.patched/fieldmeterdecay.cc	2003-07-22 13:31:47.000000000 +1000
@@ -26,7 +26,7 @@
 //   4.  Make the checkResources () function in the meter set the 
 //	 dodecay_ variable according to the, e.g., xosview*cpuDecay resource.
 
-#include <fstream.h>
+#include <fstream>
 #include <math.h>		//  For fabs()
 #include "general.h"
 #include "fieldmeter.h"
@@ -116,9 +116,9 @@
     twidth = (int) (0.5 + (width_ * (float) fields_[i]) / total_); 
     decaytwidth = (int) (0.5 + width_ * decay_[i]);
     if (decaytwidth < 0.0) {
-        cerr << "Error:  FieldMeterDecay " << name() << ":  decaytwidth of ";
-        cerr << decaytwidth << ", width of " << width_ << ", decay_[" << i;
-        cerr << "] of " << decay_[i] << endl;
+        std::cerr << "Error:  FieldMeterDecay " << name() << ":  decaytwidth of ";
+        std::cerr << decaytwidth << ", width of " << width_ << ", decay_[" << i;
+        std::cerr << "] of " << decay_[i] << std::endl;
     }
 
     //  However, due to rounding, we may have gone one
@@ -142,13 +142,13 @@
       //    Let's correct for that here.
     if ( manditory || (twidth != lastvals_[i]) || (x != lastx_[i]) ){
       if (!checkX(x, twidth))
-        cerr <<__FILE__ << ":" << __LINE__ <<endl;
+        std::cerr <<__FILE__ << ":" << __LINE__ <<std::endl;
       parent_->drawFilledRectangle( x, y_, twidth, halfheight );
     }
 
     if ( manditory || (decay_[i] != lastDecayval_[i]) ){
       if (!checkX(decayx, decaytwidth))
-        cerr <<__FILE__ << ":" << __LINE__ <<endl;
+        std::cerr <<__FILE__ << ":" << __LINE__ <<std::endl;
       parent_->drawFilledRectangle( decayx, y_+halfheight+1,
             decaytwidth, height_ - halfheight-1);
     }
diff -ruN xosview-1.8.0/fieldmetergraph.cc xosview-1.8.0.patched/fieldmetergraph.cc
--- xosview-1.8.0/fieldmetergraph.cc	1999-02-27 10:28:38.000000000 +1100
+++ xosview-1.8.0.patched/fieldmetergraph.cc	2003-07-22 13:31:46.000000000 +1000
@@ -24,7 +24,7 @@
 //   5.  Make the checkResources () function in the meter set the 
 //	 useGraph_ variable according to the, e.g., xosview*cpuGraph resource.
 
-#include <fstream.h>
+#include <fstream>
 #include <math.h>		//  For fabs()
 #include "general.h"
 #include "fieldmeter.h"
diff -ruN xosview-1.8.0/Host.cc xosview-1.8.0.patched/Host.cc
--- xosview-1.8.0/Host.cc	1999-01-24 09:20:39.000000000 +1100
+++ xosview-1.8.0.patched/Host.cc	2003-07-22 13:59:24.000000000 +1000
@@ -7,7 +7,7 @@
 // $Id$
 //
 #include <stdlib.h>
-#include <string.h>
+#include <string>
 #include "general.h"
 #include "Host.h"
 
@@ -168,7 +168,7 @@
   return false;
 }
 
-ostream &Host::print(ostream& os) const {
+std::ostream &Host::print(std::ostream& os) const {
   /*  Cast 'this' to a char*, so we don't need to create a Host::! operator.*/
   if (!*((char*)this))
     return os <<"Invalid Host.  h_errno was = " <<_failure <<"\n";
diff -ruN xosview-1.8.0/Host.h xosview-1.8.0.patched/Host.h
--- xosview-1.8.0/Host.h	1999-01-24 09:20:39.000000000 +1100
+++ xosview-1.8.0.patched/Host.h	2003-07-22 13:58:45.000000000 +1000
@@ -17,7 +17,7 @@
 #include <netinet/in.h>
 #include <netdb.h>
 #include <arpa/inet.h>
-#include <iostream.h>
+#include <iostream>
 #include "bool.h"
 
 class Host {
@@ -66,7 +66,7 @@
 
   // Should not use this under linux for the same reashon as the above
   // function.
-  ostream &print(ostream &os) const;
+  std::ostream &print(std::ostream &os) const;
 
 protected:
 private:
@@ -84,7 +84,7 @@
 };
 
 // Do not use this under linux until inet_ntoa() is fixed.
-inline ostream &operator<<(ostream &os, const Host& host) {
+inline std::ostream &operator<<(std::ostream &os, const Host& host) {
   return host.print(os);
 }
 
diff -ruN xosview-1.8.0/hpux/loadmeter.cc xosview-1.8.0.patched/hpux/loadmeter.cc
--- xosview-1.8.0/hpux/loadmeter.cc	1999-02-01 07:18:49.000000000 +1100
+++ xosview-1.8.0.patched/hpux/loadmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -39,7 +39,7 @@
     //  prevent this whole problem, the load meter can not be a decay
     //  meter.  The load is a decaying average kind of thing anyway,
     //  so having a decaying load average is redundant.
-    cerr << "Warning:  The loadmeter can not be configured as a decay\n"
+    std::cerr << "Warning:  The loadmeter can not be configured as a decay\n"
          << "  meter.  See the source code (" << __FILE__ << ") for further\n"
          << "  details.\n";
     dodecay_ = 0;
diff -ruN xosview-1.8.0/irix65/cpumeter.cc xosview-1.8.0.patched/irix65/cpumeter.cc
--- xosview-1.8.0/irix65/cpumeter.cc	2002-02-25 08:31:02.000000000 +1100
+++ xosview-1.8.0.patched/irix65/cpumeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -7,8 +7,8 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <string.h>
-#include <strstream.h>
+#include <string>
+#include <strstream>
 
 CPUMeter::CPUMeter(XOSView *parent, int cpuid)
     : FieldMeterGraph(parent, USED_CPU_STATES, toUpper(cpuStr(cpuid)),
@@ -19,7 +19,7 @@
             cputime_[i][j] = 0;
 
     if ((sinfosz = sysmp(MP_SASZ, MPSA_SINFO)) < 0) {
-        cerr << "sysinfo scall interface not supported" << endl;
+        std::cerr << "sysinfo scall interface not supported" << std::endl;
         parent_->done(1);
         return;
     }
diff -ruN xosview-1.8.0/irix65/gfxmeter.cc xosview-1.8.0.patched/irix65/gfxmeter.cc
--- xosview-1.8.0/irix65/gfxmeter.cc	2002-02-20 22:15:04.000000000 +1100
+++ xosview-1.8.0.patched/irix65/gfxmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -71,7 +71,7 @@
         //  prevent this whole problem, the gfx meter can not be a decay
         //  meter.  The gfx is a decaying average kind of thing anyway,
         //  so having a decaying gfx average is redundant.
-        cerr << "Warning:  The gfxmeter can not be configured as a decay\n"
+        std::cerr << "Warning:  The gfxmeter can not be configured as a decay\n"
              << "  meter. See the source code (" <<__FILE__<< ") for further\n"
              << "  details.\n";
         dodecay_ = 0;
diff -ruN xosview-1.8.0/irix65/loadmeter.cc xosview-1.8.0.patched/irix65/loadmeter.cc
--- xosview-1.8.0/irix65/loadmeter.cc	2002-02-18 19:24:03.000000000 +1100
+++ xosview-1.8.0.patched/irix65/loadmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -61,7 +61,7 @@
         //  prevent this whole problem, the load meter can not be a decay
         //  meter.  The load is a decaying average kind of thing anyway,
         //  so having a decaying load average is redundant.
-        cerr << "Warning:  The loadmeter can not be configured as a decay\n"
+        std::cerr << "Warning:  The loadmeter can not be configured as a decay\n"
              << "  meter. See the source code (" <<__FILE__<< ") for further\n"
              << "  details.\n";
         dodecay_ = 0;
@@ -77,7 +77,7 @@
 void LoadMeter::getloadinfo(void)
 {
     if (rstat (hostname, &res) != 0) {
-        cerr << hostname <<endl;
+        std::cerr << hostname <<std::endl;
         perror ("rstat");
         return;
     }
diff -ruN xosview-1.8.0/linux/btrymeter.cc xosview-1.8.0.patched/linux/btrymeter.cc
--- xosview-1.8.0/linux/btrymeter.cc	1999-01-24 05:34:26.000000000 +1100
+++ xosview-1.8.0.patched/linux/btrymeter.cc	2003-07-22 13:54:28.000000000 +1000
@@ -8,7 +8,7 @@
 //
 #include "btrymeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 static const char APMFILENAME[] = "/proc/apm";
@@ -38,10 +38,10 @@
 
 
 void BtryMeter::getpwrinfo( void ){
-  ifstream loadinfo( APMFILENAME );
+  std::ifstream loadinfo( APMFILENAME );
 
   if ( !loadinfo ){
-    cerr <<"Can not open file : " <<APMFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<APMFILENAME <<std::endl;
     parent_->done(1);
     return;
   }
diff -ruN xosview-1.8.0/linux/cpumeter.cc xosview-1.8.0.patched/linux/cpumeter.cc
--- xosview-1.8.0/linux/cpumeter.cc	1999-02-25 09:16:20.000000000 +1100
+++ xosview-1.8.0.patched/linux/cpumeter.cc	2003-07-22 13:46:49.000000000 +1000
@@ -8,10 +8,10 @@
 //
 #include "cpumeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
-#include <string.h>
-#include <strstream.h>
+#include <string>
+#include <strstream>
 #include <ctype.h>
 
 static const char STATFILENAME[] = "/proc/stat";
@@ -50,10 +50,10 @@
 void CPUMeter::getcputime( void ){
   total_ = 0;
   char tmp[1024];
-  ifstream stats( STATFILENAME );
+  std::ifstream stats( STATFILENAME );
 
   if ( !stats ){
-    cerr <<"Can not open file : " <<STATFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<STATFILENAME <<std::endl;
     exit( 1 );
   }
 
@@ -79,10 +79,10 @@
 }
 
 int CPUMeter::findLine(const char *cpuID){
-  ifstream stats( STATFILENAME );
+  std::ifstream stats( STATFILENAME );
 
   if ( !stats ){
-    cerr <<"Can not open file : " <<STATFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<STATFILENAME <<std::endl;
     exit( 1 );
   }
 
@@ -104,10 +104,10 @@
 // If it finds that this patch has been applied to the current kernel 
 // then returns the number of cpus that are on this machine.
 int CPUMeter::countCPUs(void){
-  ifstream stats( STATFILENAME );
+  std::ifstream stats( STATFILENAME );
 
   if ( !stats ){
-    cerr <<"Can not open file : " <<STATFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<STATFILENAME <<std::endl;
     exit( 1 );
   }
 
@@ -126,12 +126,12 @@
 
 const char *CPUMeter::cpuStr(int num){
   static char buffer[32];
-  ostrstream str(buffer, 32);
+  std::ostrstream str(buffer, 32);
   
   str << "cpu";
   if (num != 0)
     str << (num - 1);
-  str << ends;
+  str << std::ends;
 
   return buffer;
 }
diff -ruN xosview-1.8.0/linux/diskmeter.cc xosview-1.8.0.patched/linux/diskmeter.cc
--- xosview-1.8.0/linux/diskmeter.cc	1999-11-08 07:30:58.000000000 +1100
+++ xosview-1.8.0.patched/linux/diskmeter.cc	2003-07-22 13:54:57.000000000 +1000
@@ -8,7 +8,7 @@
 
 #include "diskmeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 static const char STATFILENAME[] = "/proc/stat";
@@ -50,11 +50,11 @@
     IntervalTimerStop();
     total_ = maxspeed_;
     char buf[1024];
-    ifstream stats( STATFILENAME );
+    std::ifstream stats( STATFILENAME );
 
     if ( !stats )
         {
-        cerr <<"Can not open file : " <<STATFILENAME <<endl;
+        std::cerr <<"Can not open file : " <<STATFILENAME <<std::endl;
         exit( 1 );
         }
 
diff -ruN xosview-1.8.0/linux/intmeter.cc xosview-1.8.0.patched/linux/intmeter.cc
--- xosview-1.8.0/linux/intmeter.cc	1998-10-04 09:35:24.000000000 +1000
+++ xosview-1.8.0.patched/linux/intmeter.cc	2003-07-22 13:52:43.000000000 +1000
@@ -9,8 +9,8 @@
 #include "intmeter.h"
 #include "xosview.h"
 #include "cpumeter.h"
-#include <fstream.h>
-#include <strstream.h>
+#include <fstream>
+#include <strstream>
 #include <stdlib.h>
 #ifdef __alpha__
 #include <asm/irq.h>
@@ -67,16 +67,16 @@
 }
 
 float IntMeter::getLinuxVersion(void) {
-    ifstream vfile(VERSIONFILE);
+    std::ifstream vfile(VERSIONFILE);
     if (!vfile) {
-      cerr << "Can not open file : " << VERSIONFILE << endl;
+      std::cerr << "Can not open file : " << VERSIONFILE << std::endl;
       exit(1);
     }
 
     char buffer[128];
     vfile >> buffer >> buffer >> buffer;
     *strrchr(buffer, '.') = '\0';
-    istrstream is(buffer, 128);
+    std::istrstream is(buffer, 128);
     float rval = 0.0;
     is >> rval;
 
@@ -88,16 +88,16 @@
 }
 
 void IntMeter::getirqs( void ){
-  ifstream intfile( INTFILE );
+  std::ifstream intfile( INTFILE );
   int intno, count;
 
   if ( !intfile ){
-    cerr <<"Can not open file : " <<INTFILE <<endl;
+    std::cerr <<"Can not open file : " <<INTFILE <<std::endl;
     exit( 1 );
   }
 
   if (!_old)
-      intfile.istream::ignore(1024, '\n');
+      intfile.std::istream::ignore(1024, '\n');
 
   while ( !intfile.eof() ){
     intfile >>intno;
@@ -106,7 +106,7 @@
     if ( !intfile.eof() ){
       for (int i = 0 ; i <= _cpu ; i++)
           intfile >>count;
-      intfile.istream::ignore(1024, '\n');
+      intfile.std::istream::ignore(1024, '\n');
       
       irqs_[intno] = count;
     }
diff -ruN xosview-1.8.0/linux/lmstemp.cc xosview-1.8.0.patched/linux/lmstemp.cc
--- xosview-1.8.0/linux/lmstemp.cc	2002-02-25 09:00:47.000000000 +1100
+++ xosview-1.8.0.patched/linux/lmstemp.cc	2003-07-22 13:56:55.000000000 +1000
@@ -14,7 +14,7 @@
 //
 #include "lmstemp.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include <dirent.h>
 #include <sys/stat.h>
@@ -25,7 +25,7 @@
 		 const char *caption)
   : FieldMeter( parent, 3, label, caption, 1, 1, 0 ){
     if(!checksensors(filename)) {
-	cerr <<"Can not find file : " <<PROC_SENSORS <<"/*/" << filename <<endl;
+	std::cerr <<"Can not find file : " <<PROC_SENSORS <<"/*/" << filename <<std::endl;
     	parent_->done(1);
   }
     char *p;
@@ -65,7 +65,7 @@
 	    {
 	      d2=opendir(dirname);
 	      if(!d2)
-		cerr << "The directory " <<dirname <<"exists but cannot be read.\n";
+		std::cerr << "The directory " <<dirname <<"exists but cannot be read.\n";
 	      else
 		{
 		  while((ent2=readdir(d2)))
@@ -135,10 +135,10 @@
 //
 
 void LmsTemp::getlmstemp( void ){
-  ifstream file( _filename );
+  std::ifstream file( _filename );
 
   if ( !file ){
-    cerr <<"Can not open file : " <<file <<endl;
+    std::cerr <<"Can not open file : " <<file <<std::endl;
     parent_->done(1);
     return;
   }
diff -ruN xosview-1.8.0/linux/loadmeter.cc xosview-1.8.0.patched/linux/loadmeter.cc
--- xosview-1.8.0/linux/loadmeter.cc	1999-11-07 09:48:17.000000000 +1100
+++ xosview-1.8.0.patched/linux/loadmeter.cc	2003-07-22 13:54:01.000000000 +1000
@@ -10,7 +10,7 @@
 //
 #include "loadmeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 static const char LOADFILENAME[] = "/proc/loadavg";
@@ -49,7 +49,7 @@
     //  prevent this whole problem, the load meter can not be a decay
     //  meter.  The load is a decaying average kind of thing anyway,
     //  so having a decaying load average is redundant.
-    cerr << "Warning:  The loadmeter can not be configured as a decay\n"
+    std::cerr << "Warning:  The loadmeter can not be configured as a decay\n"
          << "  meter.  See the source code (" << __FILE__ << ") for further\n"
          << "  details.\n";
     dodecay_ = 0;
@@ -63,10 +63,10 @@
 
 
 void LoadMeter::getloadinfo( void ){
-  ifstream loadinfo( LOADFILENAME );
+  std::ifstream loadinfo( LOADFILENAME );
 
   if ( !loadinfo ){
-    cerr <<"Can not open file : " <<LOADFILENAME <<endl;
+    std::cerr <<"Can not open file : " <<LOADFILENAME <<std::endl;
     parent_->done(1);
     return;
   }
diff -ruN xosview-1.8.0/linux/memmeter.cc xosview-1.8.0.patched/linux/memmeter.cc
--- xosview-1.8.0/linux/memmeter.cc	1999-11-07 09:48:17.000000000 +1100
+++ xosview-1.8.0.patched/linux/memmeter.cc	2003-07-22 13:49:08.000000000 +1000
@@ -7,8 +7,8 @@
 //
 #include "memmeter.h"
 #include "xosview.h"
-#include <fstream.h>
-#include <strstream.h>
+#include <fstream>
+#include <strstream>
 #include <stdlib.h>
 
 static const char MEMFILENAME[] = "/proc/meminfo";
@@ -19,7 +19,7 @@
   _shAdj = -1;
 
   // Check and see if the memstat module has been loaded
-  ifstream test(MEMSTATFNAME);
+  std::ifstream test(MEMSTATFNAME);
   if (test){
     setNumFields(5);
     legend("USED/SHAR/BUFF/CACHE/FREE");
@@ -85,23 +85,23 @@
 
 MemMeter::LineInfo *MemMeter::findLines(LineInfo *tmplate, int len, 
                                              const char *fname){
-  ifstream meminfo(fname);
+  std::ifstream meminfo(fname);
   if (!meminfo){
-    cerr << "Can not open file : " << fname << endl;
+    std::cerr << "Can not open file : " << fname << std::endl;
     exit(1);
   }
 
   LineInfo *rval = new LineInfo[len];
 
   char buf[256];
-  istrstream line(buf, 256);
+  std::istrstream line(buf, 256);
 
   // Get the info from the "standard" meminfo file.
   int lineNum = 0;
   int inum = 0;  // which info are we going to insert
   while (!meminfo.eof()){
     meminfo.getline(buf, 256);
-    line.seekg(0, ios::beg);
+    line.seekg(0, std::ios::beg);
     lineNum++;
 
     for (int i = 0 ; i < len ; i++)
@@ -136,14 +136,14 @@
 }
 
 void MemMeter::getmemstat(const char *fname, LineInfo *infos, int ninfos){
-  ifstream meminfo(fname);
+  std::ifstream meminfo(fname);
   if (!meminfo){
-    cerr << "Can not open file : " << fname << endl;
+    std::cerr << "Can not open file : " << fname << std::endl;
     exit(1);
   }
 
   char buf[256];
-  istrstream line(buf, 256);
+  std::istrstream line(buf, 256);
 
   char ignore[256];
 
@@ -157,7 +157,7 @@
     if (lineNum != infos[inum].line())
       continue;
 
-    line.seekg(0, ios::beg);
+    line.seekg(0, std::ios::beg);
 
     unsigned long val;
     line >> ignore >> val;
diff -ruN xosview-1.8.0/linux/netmeter.cc xosview-1.8.0.patched/linux/netmeter.cc
--- xosview-1.8.0/linux/netmeter.cc	2002-02-25 09:55:19.000000000 +1100
+++ xosview-1.8.0.patched/linux/netmeter.cc	2003-07-22 13:51:00.000000000 +1000
@@ -30,7 +30,7 @@
 #include "xosview.h"
 
 #include <unistd.h>
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
@@ -41,8 +41,8 @@
 #endif
 #include <netinet/in.h>
 #include <errno.h>
-#include <iostream.h>
-#include <iomanip.h>
+#include <iostream>
+#include <iomanip>
 
 NetMeter::NetMeter( XOSView *parent, float max )
   : FieldMeterGraph( parent, 3, "NET", "IN/OUT/IDLE" ){
@@ -60,10 +60,10 @@
 
 void NetMeter::checkOSVersion(void)
     {
-    ifstream ifs("/proc/sys/kernel/osrelease");
+    std::ifstream ifs("/proc/sys/kernel/osrelease");
     if (!ifs)
         {
-        cerr <<"Can not open file : " << "/proc/sys/kernel/osrelease" <<endl;
+        std::cerr <<"Can not open file : " << "/proc/sys/kernel/osrelease" <<std::endl;
         exit(1);
         }
 
@@ -77,7 +77,7 @@
     if (major > 2 || (major == 2 && minor >= 1))
         {
 	// check presence of iacct and oacct chains
-        ifstream chains("/proc/net/ip_fwchains");
+        std::ifstream chains("/proc/net/ip_fwchains");
 	int n = 0;
 	char buf[1024];
 
@@ -119,7 +119,7 @@
 
   _ipsock = socket(AF_INET, SOCK_DGRAM, 0);
   if (_ipsock == -1) {
-    cerr <<"Can not open socket : " <<strerror( errno ) <<endl;
+    std::cerr <<"Can not open socket : " <<strerror( errno ) <<std::endl;
     parent_->done(1);
     return;
   }
@@ -135,11 +135,11 @@
 
 void NetMeter::checkeventNew(void)
     {
-    ifstream ifs(_netfilename);
+    std::ifstream ifs(_netfilename);
 
     if (!ifs)
         {
-        cerr <<"Can not open file : " <<_netfilename <<endl;
+        std::cerr <<"Can not open file : " <<_netfilename <<std::endl;
         parent_->done(1);
         return;
         }
@@ -210,10 +210,10 @@
     fields_[2] = maxpackets_;     // assume no
     fields_[0] = fields_[1] = 0;  // network activity
 
-    ifstream ifs(_netfilename);
+    std::ifstream ifs(_netfilename);
     if (!ifs)
         {
-        cerr <<"Can not open file : " << _netfilename <<endl;
+        std::cerr <<"Can not open file : " << _netfilename <<std::endl;
         parent_->done(1);
         return;
         }
@@ -224,7 +224,7 @@
     ifc.ifc_buf = buff;
     if (ioctl(_ipsock, SIOCGIFCONF, &ifc) < 0)
         {
-        cerr <<"Can not get interface list : " <<strerror( errno ) <<endl;
+        std::cerr <<"Can not get interface list : " <<strerror( errno ) <<std::endl;
         parent_->done(1);
         return;
         }
@@ -237,10 +237,10 @@
 
     while (ifs)
         {
-        ifs >> hex >> sa >> c >> sm >> c >> c >> da >> c >> dm;
+        ifs >> std::hex >> sa >> c >> sm >> c >> c >> da >> c >> dm;
         for (int index = 0 ; index < 7 ; index++)
             ifs.ignore(9999, ' ');
-        ifs >> dec >> bytes;
+        ifs >> std::dec >> bytes;
 
         ifs.ignore(9999, '\n');
 
diff -ruN xosview-1.8.0/linux/pagemeter.cc xosview-1.8.0.patched/linux/pagemeter.cc
--- xosview-1.8.0/linux/pagemeter.cc	1999-02-25 09:28:18.000000000 +1100
+++ xosview-1.8.0.patched/linux/pagemeter.cc	2003-07-22 13:49:45.000000000 +1000
@@ -8,7 +8,7 @@
 //
 #include "pagemeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 
@@ -49,10 +49,10 @@
 void PageMeter::getpageinfo( void ){
   total_ = 0;
   char buf[1024];
-  ifstream stats( STATFILENAME );
+  std::ifstream stats( STATFILENAME );
 
   if ( !stats ){
-    cerr <<"Cannot open file : " <<STATFILENAME <<endl;
+    std::cerr <<"Cannot open file : " <<STATFILENAME <<std::endl;
     exit( 1 );
   }
 
diff -ruN xosview-1.8.0/linux/raidmeter.cc xosview-1.8.0.patched/linux/raidmeter.cc
--- xosview-1.8.0/linux/raidmeter.cc	1999-11-12 16:28:24.000000000 +1100
+++ xosview-1.8.0.patched/linux/raidmeter.cc	2003-07-22 13:56:29.000000000 +1000
@@ -7,11 +7,11 @@
 
 #include "raidmeter.h"
 #include "xosview.h"
-#include <fstream.h>
-#include <strstream.h>
+#include <fstream>
+#include <strstream>
 #include <stdio.h>
 #include <stdlib.h>
-#include <string.h>
+#include <string>
 
 RAIDMeter::RAIDMeter( XOSView *parent, int raiddev)
   : BitFieldMeter( parent, 1, 2, "RAID") {
@@ -19,8 +19,8 @@
   getRAIDstate();
   if(disknum<1)
     disableMeter();
-  ostrstream os;
-  os << "MD" << raiddev << ends;
+  std::ostrstream os;
+  os << "MD" << raiddev << std::ends;
   legend(os.str());
   delete[] os.str();
   if(disknum>=1){
@@ -71,8 +71,8 @@
 int RAIDMeter::find1(char *key, char *findwhat, int num1){
   char buf[80];
   int rc;
-  ostrstream os(buf, 80);
-  os << findwhat << "." << num1 << ends;
+  std::ostrstream os(buf, 80);
+  os << findwhat << "." << num1 << std::ends;
   rc=!strncmp(buf,key, 80);
   return rc;
 }
@@ -80,8 +80,8 @@
 int RAIDMeter::find2(char *key, char *findwhat, int num1, int num2){
   char buf[80];
   int rc;
-  ostrstream os(buf, 80);
-  os << findwhat << "." << num1 << "." << num2 << ends;
+  std::ostrstream os(buf, 80);
+  os << findwhat << "." << num1 << "." << num2 << std::ends;
   rc=!strncmp(buf,key, 80);
   return rc;
 }
@@ -113,11 +113,11 @@
 }
 
 void RAIDMeter::getRAIDstate( void ){
-  ifstream raidfile( RAIDFILE );
+  std::ifstream raidfile( RAIDFILE );
   char l[256];
 
   if ( !raidfile ){
-    cerr <<"Can not open file : " <<RAIDFILE <<endl;
+    std::cerr <<"Can not open file : " <<RAIDFILE <<std::endl;
     exit( 1 );
   }
 
diff -ruN xosview-1.8.0/linux/serialmeter.cc xosview-1.8.0.patched/linux/serialmeter.cc
--- xosview-1.8.0/linux/serialmeter.cc	2002-02-25 06:52:01.000000000 +1100
+++ xosview-1.8.0.patched/linux/serialmeter.cc	2003-07-22 13:57:46.000000000 +1000
@@ -16,8 +16,8 @@
 #include <sys/ioctl.h>
 #include <errno.h>
 #include <stdlib.h>
-#include <strstream.h>
-#include <iomanip.h>
+#include <strstream>
+#include <iomanip>
 
 // This should go away after types.h gets fixed in the kernel.
 #ifdef __alpha__
@@ -66,8 +66,8 @@
 
   _port = getPortBase(_device);
   if (!getport(_port + UART_LSR) || !getport(_port + UART_MSR)){
-    cerr << "SerialMeter::SerialMeter() : "
-         << "xosview must be suid root to use the serial meter." <<endl;
+    std::cerr << "SerialMeter::SerialMeter() : "
+         << "xosview must be suid root to use the serial meter." <<std::endl;
     parent_->done(1);
   }
 }
@@ -122,12 +122,12 @@
 
     // get the real serial port (code stolen from setserial 2.11)
     if ((fd = open(deviceFile[dev], O_RDONLY|O_NONBLOCK)) < 0) {
-      cerr << "SerialMeter::SerialMeter() : "
-           << "failed to open " << deviceFile[dev] <<"." <<endl;
+      std::cerr << "SerialMeter::SerialMeter() : "
+           << "failed to open " << deviceFile[dev] <<"." <<std::endl;
       exit(1);
     }
     if (ioctl(fd, TIOCGSERIAL, &serinfo) < 0) {
-      cerr << "Failed to detect port base for " << deviceFile[dev] << endl;
+      std::cerr << "Failed to detect port base for " << deviceFile[dev] << std::endl;
       close(fd);
       exit(1);
     }
@@ -136,9 +136,9 @@
     return serinfo.port;
   }
   else { // Use user specified port base.
-    istrstream istrm(res);
+    std::istrstream istrm(res);
     unsigned short int tmp = 0;
-    istrm >> hex >> tmp;
+    istrm >> std::hex >> tmp;
     return tmp;
   }
 
diff -ruN xosview-1.8.0/linux/swapmeter.cc xosview-1.8.0.patched/linux/swapmeter.cc
--- xosview-1.8.0/linux/swapmeter.cc	1999-02-01 07:18:49.000000000 +1100
+++ xosview-1.8.0.patched/linux/swapmeter.cc	2003-07-22 13:31:47.000000000 +1000
@@ -9,8 +9,8 @@
 
 #include "swapmeter.h"
 #include "xosview.h"
-#include <fstream.h>
-#include <strstream.h>
+#include <fstream>
+#include <strstream>
 #include <stdlib.h>
 
 #ifdef USESYSCALLS
@@ -78,7 +78,7 @@
 void SwapMeter::getswapinfo( void ){
   ifstream meminfo( MEMFILENAME );
   if ( !meminfo ){
-    cerr <<"Con not open file : " <<MEMFILENAME <<endl;
+    std::cerr <<"Con not open file : " <<MEMFILENAME <<std::endl;
     exit( 1 );
   }
 
diff -ruN xosview-1.8.0/llist.cc xosview-1.8.0.patched/llist.cc
--- xosview-1.8.0/llist.cc	1999-11-05 16:39:03.000000000 +1100
+++ xosview-1.8.0.patched/llist.cc	2003-07-22 13:31:46.000000000 +1000
@@ -6,7 +6,7 @@
 //
 // $Id$
 //
-#include <iostream.h>
+#include <iostream>
 #include "general.h"
 #include "llist.h"
 
diff -ruN xosview-1.8.0/sunos5/cpumeter.cc xosview-1.8.0.patched/sunos5/cpumeter.cc
--- xosview-1.8.0/sunos5/cpumeter.cc	1999-02-01 07:26:38.000000000 +1100
+++ xosview-1.8.0.patched/sunos5/cpumeter.cc	2003-07-22 13:31:46.000000000 +1000
@@ -7,8 +7,8 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <string.h>
-#include <strstream.h>
+#include <string>
+#include <strstream>
 
 CPUMeter::CPUMeter(XOSView *parent, kstat_ctl_t *_kc, int cpuid)
 	: FieldMeterGraph(parent, CPU_STATES, toUpper(cpuStr(cpuid)),
diff -ruN xosview-1.8.0/sunos5/diskmeter.cc xosview-1.8.0.patched/sunos5/diskmeter.cc
--- xosview-1.8.0/sunos5/diskmeter.cc	1999-11-07 10:12:34.000000000 +1100
+++ xosview-1.8.0.patched/sunos5/diskmeter.cc	2003-07-22 13:31:46.000000000 +1000
@@ -7,7 +7,7 @@
 
 #include "diskmeter.h"
 #include "xosview.h"
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 
 
diff -ruN xosview-1.8.0/sunos5/loadmeter.cc xosview-1.8.0.patched/sunos5/loadmeter.cc
--- xosview-1.8.0/sunos5/loadmeter.cc	1999-02-01 07:26:38.000000000 +1100
+++ xosview-1.8.0.patched/sunos5/loadmeter.cc	2003-07-22 13:31:46.000000000 +1000
@@ -52,7 +52,7 @@
 		 * anyway, so having a decaying load average is
 		 * redundant.
 		 */
-		cerr << "Warning:  The loadmeter can not be configured as a decay\n"
+		std::cerr << "Warning:  The loadmeter can not be configured as a decay\n"
 		     << "  meter.  See the source code (" << __FILE__ << ") for further\n"
 		     << "  details.\n";
 		dodecay_ = 0;
diff -ruN xosview-1.8.0/timer.h xosview-1.8.0.patched/timer.h
--- xosview-1.8.0/timer.h	1999-11-19 20:50:20.000000000 +1100
+++ xosview-1.8.0.patched/timer.h	2003-07-22 13:44:57.000000000 +1000
@@ -49,7 +49,7 @@
       + stoptime_.tv_usec - starttime_.tv_usec;
   }
   
-  ostream &printOn(ostream &os) const {
+  std::ostream &printOn(std::ostream &os) const {
     return os <<"Timer : ["
       <<"starttime_ = " <<TimeVal(starttime_)
       <<", stoptime_ = " <<TimeVal(stoptime_)
@@ -62,7 +62,7 @@
 private:
 };
 
-inline ostream &operator<<(ostream &os, const Timer &t){
+inline std::ostream &operator<<(std::ostream &os, const Timer &t){
   return t.printOn(os);
 }
 
diff -ruN xosview-1.8.0/timeval.h xosview-1.8.0.patched/timeval.h
--- xosview-1.8.0/timeval.h	1999-01-24 09:20:40.000000000 +1100
+++ xosview-1.8.0.patched/timeval.h	2003-07-22 13:44:14.000000000 +1000
@@ -12,7 +12,7 @@
 #define TIMEVAL_H_CVSID "$Id$"
 
 #include <sys/time.h>
-#include <iostream.h>
+#include <iostream>
 
 class TimeVal {
 public:
@@ -29,7 +29,7 @@
 
   operator struct timeval(void) const { return _val; }
 
-  ostream &printOn(ostream &os) const {
+  std::ostream &printOn(std::ostream &os) const {
     return os <<"(" <<sec() <<" sec, " <<usec() <<" usec)";
   }
 
@@ -37,7 +37,7 @@
   struct timeval _val;
 };
 
-inline ostream &operator<<(ostream &os, const TimeVal &tv){
+inline std::ostream &operator<<(std::ostream &os, const TimeVal &tv){
   return tv.printOn(os);
 }
 
diff -ruN xosview-1.8.0/xosview.cc xosview-1.8.0.patched/xosview.cc
--- xosview-1.8.0/xosview.cc	2002-02-25 08:41:55.000000000 +1100
+++ xosview-1.8.0.patched/xosview.cc	2003-07-22 13:31:47.000000000 +1000
@@ -6,9 +6,9 @@
 //
 // $Id$
 //
-#include <iostream.h>
+#include <iostream>
 #include <unistd.h>
-#include <string.h>
+#include <string>
 #include <stdlib.h>
 #include <sys/time.h>
 #include "snprintf.h"
@@ -332,7 +332,7 @@
   {
     switch (argv[0][1]) {
       case 'v':
-      		cerr << versionString << endl;
+      		std::cerr << versionString << std::endl;
 		exit(0);
       case 'n': //  Check for -name option that was already parsed
 		//  and acted upon by main().
@@ -355,12 +355,12 @@
 #endif
       case '-':  /*  Check for --version argument.  */
               if (!strncasecmp(*argv, "--version", 10)) {
-	        cerr << versionString << endl;
+	        std::cerr << versionString << std::endl;
 		exit(0);
 	      }
 	      /*  Fall through to default/error case.  */
       default:
-      		cerr << "Ignoring unknown option '" << argv[0] << "'.\n";
+      		std::cerr << "Ignoring unknown option '" << argv[0] << "'.\n";
 	  	break;
     }
     argc--;
diff -ruN xosview-1.8.0/Xrm.cc xosview-1.8.0.patched/Xrm.cc
--- xosview-1.8.0/Xrm.cc	1999-11-17 17:36:13.000000000 +1100
+++ xosview-1.8.0.patched/Xrm.cc	2003-07-22 14:04:32.000000000 +1000
@@ -7,11 +7,11 @@
 // $Id$
 //
 
-#include <string.h>
+#include <string>
 #include <stdlib.h>
 #include <stdio.h>	//  For snprintf().
 #include <ctype.h>
-#include <iostream.h>
+#include <iostream>
 #include <unistd.h>  //  for access(), etc.  BCG
 #include "snprintf.h"
 #include "general.h"
@@ -29,7 +29,7 @@
 bool Xrm::_initialized = false;
 
 Xrm::Xrm(const char *instanceName, int argc, char **argv){
-  cerr << " Error:  This constructor is not supported yet.\n";
+  std::cerr << " Error:  This constructor is not supported yet.\n";
   exit (-1);
   _db = NULL;
   _class = _instance = NULLQUARK;
@@ -114,7 +114,7 @@
   }
   else
   {
-    cerr << "Error:  Xrm:loadAndMergeResources() called twice!\n";
+    std::cerr << "Error:  Xrm:loadAndMergeResources() called twice!\n";
     exit (-1);
   }
   //  This is ugly code.  According to X and Xt rules, many files need
@@ -126,7 +126,7 @@
 //  This all needs to be done in the proper order:
 /*
 Listed from weakest to strongest:
-  (from code-builtin-resources) (Stored in the string in defaultstring.h)
+  (from code-builtin-resources) (Stored in the string in defaultstring)
   app-defaults
   XOSView (from XAPPLRESDIR directory)
   from RESOURCE_MANAGER property on server (reads .Xdefaults if needed)
@@ -227,14 +227,14 @@
 
 
 //------------  Some debugging functions follow.  -----------------------
-inline ostream &operator<<(ostream &os, const XrmBinding &b){
+inline std::ostream &operator<<(std::ostream &os, const XrmBinding &b){
   switch (b){
   case XrmBindTightly:
     return os << ".";
   case XrmBindLoosely:
     return os << "*";
   default:
-    cerr <<"ostream operator<<(ostream &, const XrmBinding &) : "
+    std::cerr <<"std::ostream operator<<(std::ostream &, const XrmBinding &) : "
       <<"Unknown XrmBinding!";
     return os;
   }
@@ -242,7 +242,7 @@
   return os;
 }
 
-ostream &Xrm::dump(ostream &os) const {
+std::ostream &Xrm::dump(std::ostream &os) const {
   os <<"--- Xrm --- class: " <<XrmQuarkToString(_class)
      <<", instance: " <<XrmQuarkToString(_instance) <<"\n";
 
@@ -259,10 +259,10 @@
                  XrmQuarkList quarks, XrmRepresentation *type,
                  XrmValue *value, XPointer closure) {
   
-  ostream *os = (ostream *)closure;
+  std::ostream *os = (std::ostream *)closure;
   (void) type;  //  Avoid gcc warnings.
 
-  //cerr <<"type = " <<XrmQuarkToString(*type) <<endl;
+  //std::cerr <<"type = " <<XrmQuarkToString(*type) <<std::endl;
 
   int i = 0;
   while (quarks[i] != NULLQUARK){
diff -ruN xosview-1.8.0/Xrm.h xosview-1.8.0.patched/Xrm.h
--- xosview-1.8.0/Xrm.h	1999-01-24 09:20:39.000000000 +1100
+++ xosview-1.8.0.patched/Xrm.h	2003-07-22 14:06:00.000000000 +1000
@@ -31,7 +31,7 @@
   const char* getDisplayName (int argc, char** argv);
   void loadAndMergeResources(int& argc, char** argv, Display* display);
 
-  ostream &dump(ostream &os) const;
+  std::ostream &dump(std::ostream &os) const;
 
 private:
   XrmDatabase _db;
@@ -47,7 +47,7 @@
   static bool _initialized;
 };
 
-inline ostream &operator<<(ostream &os, const Xrm &xrm){
+inline std::ostream &operator<<(std::ostream &os, const Xrm &xrm){
   return xrm.dump(os);
 }
 
diff -ruN xosview-1.8.0/xwin.cc xosview-1.8.0.patched/xwin.cc
--- xosview-1.8.0/xwin.cc	1999-01-26 08:22:10.000000000 +1100
+++ xosview-1.8.0.patched/xwin.cc	2003-07-22 14:00:00.000000000 +1000
@@ -22,8 +22,8 @@
 //-----------------------------------------------------------------------------
 
 XWin::XWin( int argc, char *argv[], int x, int y, int width, int height ){
-  cerr << "This constructor call is not supported! (" << __FILE__
-       << ":" << __LINE__ << ")" << endl;
+  std::cerr << "This constructor call is not supported! (" << __FILE__
+       << ":" << __LINE__ << ")" << std::endl;
   exit (-1);
   //  FIXME BCG  This constructor needs to do much of the work of the above
   //  one.  Or, we need to drop this as a constructor.  As it is, it is
@@ -134,7 +134,7 @@
   map();
   flush();
   if(XGetWindowAttributes(display_, window_, &attr_) == 0){
-    cerr <<"Error getting attributes of Main." <<endl;
+    std::cerr <<"Error getting attributes of Main." <<std::endl;
     exit(2);
   }
 
@@ -154,8 +154,8 @@
   const char* fontName = getResource("font");
 
   if ((font_ = XLoadQueryFont(display_, fontName)) == NULL){
-    cerr <<name_ <<": display " <<DisplayString(display_)
-      <<" cannot load font " << fontName << endl;
+    std::cerr <<name_ <<": display " <<DisplayString(display_)
+      <<" cannot load font " << fontName << std::endl;
     exit(1);
   }
 }
@@ -164,7 +164,7 @@
 void XWin::setHints( int argc, char *argv[] ){
   // Set up class hint
   if((classhints_ = XAllocClassHint()) == NULL){
-    cerr <<"Error allocating class hint!" <<endl;
+    std::cerr <<"Error allocating class hint!" <<std::endl;
     exit(1);
   }
   //  We have to cast away the const's.
@@ -173,7 +173,7 @@
 
   // Set up the window manager hints
   if((wmhints_ = XAllocWMHints()) == NULL){
-    cerr <<"Error allocating Window Manager hints!" <<endl;
+    std::cerr <<"Error allocating Window Manager hints!" <<std::endl;
     exit(1);
   }
   wmhints_->flags = (InputHint|StateHint);
@@ -182,11 +182,11 @@
 
   // Set up XTextProperty for window name and icon name
   if(XStringListToTextProperty(&name_, 1, &title_) == 0){
-    cerr <<"Error creating XTextProperty!" <<endl;
+    std::cerr <<"Error creating XTextProperty!" <<std::endl;
     exit(1);
   }
   if(XStringListToTextProperty(&name_, 1, &iconname_) == 0){
-    cerr <<"Error creating XTextProperty!" <<endl;
+    std::cerr <<"Error creating XTextProperty!" <<std::endl;
     exit(1);
   }
 
@@ -204,7 +204,7 @@
 void XWin::openDisplay( void ){
   // Open connection to display selected by user
   if ((display_ = XOpenDisplay (display_name_)) == NULL) {
-    cerr <<"Can't open display named " << display_name_ <<endl;
+    std::cerr <<"Can't open display named " << display_name_ <<std::endl;
     exit(1);
   }
 
@@ -251,8 +251,8 @@
 		pixmap_att.valuemask=XpmSize|XpmReturnPixels|XpmColormap|XpmCloseness;
 		if(XpmReadFileToPixmap(display_,DefaultRootWindow(display_),pixmap_file, pixmap, NULL, &pixmap_att))
         	{
-                	cerr << "Pixmap " << pixmap_file  << " not found" << endl;
-                	cerr << "Defaulting to blank" << endl;
+                	std::cerr << "Pixmap " << pixmap_file  << " not found" << std::endl;
+                	std::cerr << "Defaulting to blank" << std::endl;
                 	pixmap=NULL;
 			return 0; // OOps
         	}
@@ -261,7 +261,7 @@
 	return 0; // No file specified, none used
 #else
 	(void) pixmap;
-	cerr << "Error:  getPixmap called, when Xpm is not enabled!\n" ;
+	std::cerr << "Error:  getPixmap called, when Xpm is not enabled!\n" ;
 	return 0;
 #endif
 }
@@ -273,7 +273,7 @@
   // Fill out a XsizeHints structure to inform the window manager
   // of desired size and location of main window.
   if((sizehints_ = XAllocSizeHints()) == NULL){
-    cerr <<"Error allocating size hints!" <<endl;
+    std::cerr <<"Error allocating size hints!" <<std::endl;
     exit(1);
   }
   sizehints_->flags = PSize;
@@ -350,7 +350,7 @@
 #if 0
 void XWin::usage( void ){
   //  FIXME  We need to update this.  BCG
-  cerr <<name_ <<" [-display name] [-geometry geom]" <<endl;
+  std::cerr <<name_ <<" [-display name] [-geometry geom]" <<std::endl;
 //    exit (1);
 }
 #endif
@@ -386,7 +386,7 @@
     return retval;
   else
   {
-    cerr << "Error:  Couldn't find '" << name << "' resource in the resource database!\n";
+    std::cerr << "Error:  Couldn't find '" << name << "' resource in the resource database!\n";
     exit (-1);
       /*  Some compilers aren't smart enough to know that exit() exits.  */
     return '\0';
@@ -395,8 +395,8 @@
 
 //-----------------------------------------------------------------------------
 
-void XWin::dumpResources( ostream &os ){
-cerr << "Function not implemented!\n";  //  BCG  FIXME  Need to make this.
+void XWin::dumpResources( std::ostream &os ){
+std::cerr << "Function not implemented!\n";  //  BCG  FIXME  Need to make this.
 (void) os;  //  Keep gcc happy.
 }
 //-----------------------------------------------------------------------------
@@ -405,7 +405,7 @@
   XColor exact, closest;
   
   if ( XAllocNamedColor( display_, colormap(), name, &closest, &exact ) == 0 )
-    cerr <<"XWin::allocColor() : failed to alloc : " <<name <<endl;
+    std::cerr <<"XWin::allocColor() : failed to alloc : " <<name <<std::endl;
 
   return exact.pixel;
 }
@@ -504,7 +504,7 @@
     mask_ = ResizeRedirectMask;
     break;
   default:
-    cerr <<"XWin::Event::Event() : unknown event type : " <<event_ <<endl;
+    std::cerr <<"XWin::Event::Event() : unknown event type : " <<event_ <<std::endl;
     mask_ = NoEventMask;
     break;
   }
diff -ruN xosview-1.8.0/xwin.h xosview-1.8.0.patched/xwin.h
--- xosview-1.8.0/xwin.h	2002-02-25 06:52:00.000000000 +1100
+++ xosview-1.8.0.patched/xwin.h	2003-07-22 13:43:33.000000000 +1000
@@ -9,8 +9,8 @@
 #ifdef HAVE_XPM
 #include <X11/xpm.h>
 #endif
-#include <iostream.h>
-#include <string.h>
+#include <iostream>
+#include <string>
 
 #define XWIN_H_CVSID "$Id$"
 
@@ -99,7 +99,7 @@
   const char *getResourceOrUseDefault( const char *name, const char* defaultVal );
   const int isResourceTrue( const char* name ) {
     return (!strncasecmp(getResource(name),"True", 5)); }
-  void dumpResources( ostream &os );
+  void dumpResources( std::ostream &os );
 
 protected:
   class Event {
