--- security/nss/lib/pki1/oiddata.h~	2005-04-23 16:23:28.000000000 +0200
+++ security/nss/lib/pki1/oiddata.h	2005-04-23 16:20:51.000000000 +0200
@@ -43,7 +43,7 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
+extern const NSSOID *nss_builtin_oids;
 extern const PRUint32 nss_builtin_oid_count;
 
 /*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
