--- find/find.c~	2005-02-16 10:51:08.000000000 +0100
+++ find/find.c	2005-03-09 08:53:36.200358665 +0100
@@ -1091,10 +1091,25 @@
 		      struct stat *statbuf_dest,
 		      enum ChdirSymlinkHandling symlink_handling)
 {
-  int extraflags = following_links() ? O_NOFOLLOW : 0;
+  int extraflags, fd;
+  extraflags = 0;
 
+  switch (symlink_handling)
+    {
+    case SymlinkFollowOk:
+      extraflags = 0;
+      break;
+      
+    case SymlinkHandleDefault:
+      if (following_links())
+	extraflags = 0;
+      else
+	extraflags = O_NOFOLLOW;
+      break;
+    }
+   
   errno = 0;
-  int fd = open(dest, O_RDONLY|extraflags);
+  fd = open(dest, O_RDONLY|extraflags);
   if (fd < 0)
     {
       switch (errno)
