--- scribus-1.1.7/scribus/scribus.cpp	2004-05-31 12:38:49.000000000 +0200
+++ /usr/src/scribus-1.1.7/scribus/scribus.cpp	2004-06-07 10:18:52.632253944 +0200
@@ -29,6 +29,7 @@
 #include <qvbox.h>
 #include <cstdio>
 #include <cstdlib>
+#include <cmath>
 #include <dlfcn.h>
 #include <unistd.h>
 #include <iostream>
--- scribus-1.1.7/scribus/scribusview.cpp	2004-05-31 10:56:14.000000000 +0200
+++ /usr/src/scribus-1.1.7/scribus/scribusview.cpp	2004-06-07 10:14:36.606175824 +0200
@@ -35,6 +35,7 @@
 #include <qcstring.h>
 #include <qfileinfo.h>
 #include "scribus.h"
+#include <cmath>
 
 extern void Level2Layer(ScribusDoc *doc, struct Layer *ll, int Level);
 extern double Cwidth(ScribusDoc *doc, QString name, QString ch, int Siz, QString ch2 = " ");
