diff -Naur qingy-0.5.3.orig/media/Makefile.in qingy-0.5.3/media/Makefile.in
--- qingy-0.5.3.orig/media/Makefile.in	2005-06-02 16:31:09.438777192 -0700
+++ qingy-0.5.3/media/Makefile.in	2005-06-02 16:41:19.050102184 -0700
@@ -254,42 +254,6 @@
 
 clean-lisp:
 	-rm -f elc-stamp $(ELCFILES)
-dist_settingsdataDATA_INSTALL = $(INSTALL_DATA)
-install-dist_settingsdataDATA: $(dist_settingsdata_DATA)
-	@$(NORMAL_INSTALL)
-	$(mkinstalldirs) $(DESTDIR)$(settingsdatadir)
-	@list='$(dist_settingsdata_DATA)'; for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " $(dist_settingsdataDATA_INSTALL) $$d$$p $(DESTDIR)$(settingsdatadir)/$$f"; \
-	  $(dist_settingsdataDATA_INSTALL) $$d$$p $(DESTDIR)$(settingsdatadir)/$$f; \
-	done
-
-uninstall-dist_settingsdataDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(dist_settingsdata_DATA)'; for p in $$list; do \
-	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " rm -f $(DESTDIR)$(settingsdatadir)/$$f"; \
-	  rm -f $(DESTDIR)$(settingsdatadir)/$$f; \
-	done
-nodist_settingsdataDATA_INSTALL = $(INSTALL_DATA)
-install-nodist_settingsdataDATA: $(nodist_settingsdata_DATA)
-	@$(NORMAL_INSTALL)
-	$(mkinstalldirs) $(DESTDIR)$(settingsdatadir)
-	@list='$(nodist_settingsdata_DATA)'; for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " $(nodist_settingsdataDATA_INSTALL) $$d$$p $(DESTDIR)$(settingsdatadir)/$$f"; \
-	  $(nodist_settingsdataDATA_INSTALL) $$d$$p $(DESTDIR)$(settingsdatadir)/$$f; \
-	done
-
-uninstall-nodist_settingsdataDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(nodist_settingsdata_DATA)'; for p in $$list; do \
-	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " rm -f $(DESTDIR)$(settingsdatadir)/$$f"; \
-	  rm -f $(DESTDIR)$(settingsdatadir)/$$f; \
-	done
 
 # This directory's subdirectories are mostly independent; you can cd
 # into them and run `make' without going through this Makefile.
@@ -507,8 +471,7 @@
 
 info-am:
 
-install-data-am: install-dist_settingsdataDATA install-lispLISP \
-	install-nodist_settingsdataDATA
+install-data-am: install-lispLISP
 
 install-exec-am:
 
@@ -534,8 +497,7 @@
 
 ps-am:
 
-uninstall-am: uninstall-dist_settingsdataDATA uninstall-info-am \
-	uninstall-lispLISP uninstall-nodist_settingsdataDATA
+uninstall-am: uninstall-info-am uninstall-lispLISP
 
 uninstall-info: uninstall-info-recursive
 
@@ -545,18 +507,18 @@
 	distclean-recursive distclean-tags distdir dvi dvi-am \
 	dvi-recursive info info-am info-recursive install install-am \
 	install-data install-data-am install-data-recursive \
-	install-dist_settingsdataDATA install-exec install-exec-am \
+	install-exec install-exec-am \
 	install-exec-recursive install-info install-info-am \
 	install-info-recursive install-lispLISP install-man \
-	install-nodist_settingsdataDATA install-recursive install-strip \
+	install-recursive install-strip \
 	installcheck installcheck-am installdirs installdirs-am \
 	installdirs-recursive maintainer-clean maintainer-clean-generic \
 	maintainer-clean-recursive mostlyclean mostlyclean-generic \
 	mostlyclean-libtool mostlyclean-recursive pdf pdf-am \
 	pdf-recursive ps ps-am ps-recursive tags tags-recursive \
-	uninstall uninstall-am uninstall-dist_settingsdataDATA \
+	uninstall uninstall-am \
 	uninstall-info-am uninstall-info-recursive uninstall-lispLISP \
-	uninstall-nodist_settingsdataDATA uninstall-recursive
+	uninstall-recursive
 
 
 settings: Makefile
