diff -uprN nag-1.1/config/prefs.php.dist nag/config/prefs.php.dist
--- nag-1.1/config/prefs.php.dist	2002-10-15 11:13:43.000000000 -0400
+++ nag/config/prefs.php.dist	2003-02-17 10:28:13.000000000 -0500
@@ -2,7 +2,7 @@
 // $Horde: nag/config/prefs.php.dist,v 1.12.2.3 2002/10/15 15:13:43 jan Exp $
 
 // Make sure that constants are defined.
-require_once dirname(__FILE__) . '/../lib/constants.php';
+require_once NAG_BASE . '/lib/constants.php';
 
 $prefGroups['language'] = array(
     'column' => _("Your Information"),
