diff -urN viewcvs-0.9.2/cgi/viewcvs.conf.dist viewcvs/cgi/viewcvs.conf.dist
--- viewcvs-0.9.2/cgi/viewcvs.conf.dist	2001-12-22 23:13:34.000000000 -0500
+++ viewcvs/cgi/viewcvs.conf.dist	2002-04-11 19:55:56.000000000 -0400
@@ -67,7 +67,6 @@
 # roots should be separated by commas.
 #
 cvs_roots =
-	Development : /home/cvsroot
 
 # this is the name of the default CVS root.
 default_root = Development
@@ -360,7 +359,7 @@
 diff_font_size = -1
 
 # should we use 'enscript' for syntax coloring?
-use_enscript = 0
+use_enscript = 1
 
 #
 # if the enscript program is not on the path, set this value
@@ -379,15 +378,13 @@
 #
 # ViewCVS can generate tarball from a repository on the fly.
 #
-allow_tar = 0
-# allow_tar = 1
+allow_tar = 1
 
 #
 # Use CvsGraph. See http://www.akhphd.au.dk/~bertho/cvsgraph/ for
 # documentation and download. 
 #
-use_cvsgraph = 0
-# use_cvsgraph = 1
+use_cvsgraph = 1
 
 #
 # if the cvsgraph program is not on the path, set this value
@@ -399,7 +396,7 @@
 # Location of the customized cvsgraph configuration file.  
 # You will need an absolute pathname here:
 #
-cvsgraph_conf = <VIEWCVS_INSTALL_DIRECTORY>/cvsgraph.conf
+cvsgraph_conf = /usr/share/viewcvs/cvsgraph.conf
 
 #
 # Set to enable regular expression search of all files in a directory
diff -urN viewcvs-0.9.2/lib/cvsdb.py viewcvs/lib/cvsdb.py
--- viewcvs-0.9.2/lib/cvsdb.py	2001-11-18 05:29:31.000000000 -0500
+++ viewcvs/lib/cvsdb.py	2002-04-11 20:46:08.000000000 -0400
@@ -412,11 +412,8 @@
             ## is broken, dateobject.ticks() returns somthing like
             ## GMT ticks, except it forgets about daylight savings
             ## time -- we handle it ourself in the following painful way
-            gmt_time = time.mktime(
-                (dbCI_When.year, dbCI_When.month, dbCI_When.day,
-                 dbCI_When.hour, dbCI_When.minute, dbCI_When.second,
-                 0, 0, dbCI_When.dst))
-    
+            gmt_time = time.mktime(time.strptime(dbCI_When, '%Y-%m-%d %T'))
+            
             commit.SetTime(gmt_time)
             
             commit.SetFile(self.GetFile(dbFileID))
diff -urN viewcvs-0.9.2/viewcvs-install viewcvs/viewcvs-install
--- viewcvs-0.9.2/viewcvs-install	2001-12-21 06:59:45.000000000 -0500
+++ viewcvs/viewcvs-install	2002-04-11 19:55:56.000000000 -0400
@@ -51,7 +51,7 @@
 """ % version
 
 ## installer defaults
-ROOT_DIR = "/usr/local/viewcvs-" + version
+ROOT_DIR = "/usr/share/viewcvs"
 
 
 ## list of files for installation
@@ -244,12 +244,6 @@
 if __name__ == "__main__":
     print INFO_TEXT
     
-    ## get the install path
-    temp = raw_input("Installation Path [%s]: " % ROOT_DIR)
-    temp = string.strip(temp)
-    if len(temp):
-        ROOT_DIR = temp
-        
     ## install the files
     print
     print "Installing ViewCVS to:", ROOT_DIR
