Index: nsplugin.cpp
===================================================================
RCS file: /home/kde/kdebase/nsplugins/viewer/nsplugin.cpp,v
retrieving revision 1.70
diff -u -3 -d -p -r1.70 nsplugin.cpp
--- nsplugin.cpp	2002/03/16 17:57:26	1.70
+++ nsplugin.cpp	2002/06/25 05:01:16
@@ -54,6 +54,19 @@
 #include <X11/StringDefs.h>
 #include <Xm/DrawingA.h>
 
+// provide these symbols when compiling with gcc 3.x
+#if defined(__GNUC__) && __GNUC_PREREQ(3,0)
+extern "C" void* __builtin_vec_new(size_t s)
+{
+   return operator new[](s);
+}
+
+extern "C" void __builtin_vec_delete(void* p)
+{
+   operator delete[](p);
+}
+#endif
+
 // server side functions -----------------------------------------------------
 
 // allocate memory
