--- e2fsprogs-1.34/misc/util.c	2003-05-04 08:46:47.000000000 +1000
+++ util.c	2003-12-16 15:01:11.000000000 +1100
@@ -108,6 +108,21 @@
 #define MAJOR(dev)	((dev)>>8)
 #define MINOR(dev)	((dev) & 0xff)
 #endif
+/*
+ * Tests for SCSI devices.
+ */
+
+#define SCSI_DISK_MAJOR(M) ((M) == SCSI_DISK0_MAJOR || \
+  ((M) >= SCSI_DISK1_MAJOR && (M) <= SCSI_DISK7_MAJOR))
+
+#define SCSI_BLK_MAJOR(M) \
+  (SCSI_DISK_MAJOR(M)   \
+   || (M) == SCSI_CDROM_MAJOR)
+
+static __inline__ int scsi_blk_major(int m) {
+        return SCSI_BLK_MAJOR(m);
+}
+
 #ifndef SCSI_BLK_MAJOR
 #define SCSI_BLK_MAJOR(M)  ((M) == SCSI_DISK_MAJOR || (M) == SCSI_CDROM_MAJOR)
 #endif
