--- avidemux/prefs.cpp~	2003-05-12 20:26:03.000000000 +0200
+++ avidemux/prefs.cpp	2003-05-12 20:26:48.000000000 +0200
@@ -74,8 +74,8 @@
 */
 static xmlDocPtr xdoc;
 
-int xmlSaveNoEmptyTags  = 1;      /* save empty tags as <empty></empty> */
-int xmlIndentTreeOutput = 1;      /* try to indent the tree dumps */
+//int xmlSaveNoEmptyTags  = 1;      /* save empty tags as <empty></empty> */
+//int xmlIndentTreeOutput = 1;      /* try to indent the tree dumps */
 
 void erase_blank_nodes(xmlNodePtr cur){
   /* erase all blank-nodes recursive; they deny IndentTreeOutput !!! */
