--- configure.in	2004-01-04 11:23:42.000000000 +0100
+++ configure.in.cp	2004-01-04 11:41:03.000000000 +0100
@@ -72,7 +72,7 @@
 AC_FUNC_ERROR_AT_LINE
 
 
-QHACCDIRS="/usr/local/qhacc"
+QHACCDIRS="/usr/qhacc"
 AC_DEFUN(FIND_QHACC,[
   echo -n "checking for qhacc-config... "
   for dir in $1
@@ -82,7 +82,7 @@
 		  dir=$dir/$2
 		fi
 
-		prog=$dir/qhacc-config
+		prog=$dir/bin/qhacc-config
 		
 	  if test -x $prog
 		then
@@ -109,7 +109,7 @@
 
 ])
 
-AC_ARG_WITH(qhacc,[  --with-qhacc-config=DIR qhacc-config script in DIR],
+AC_ARG_WITH(qhacc-config,[  --with-qhacc-config=DIR qhacc-config script in DIR],
 													 [FIND_QHACC($withval,"")],
 													 [FIND_QHACC($QHACCDIRS, "bin")])
 
