diff -uprN wipe-2.1.0.old/blkdev.c wipe-2.1.0/blkdev.c
--- wipe-2.1.0.old/blkdev.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/blkdev.c	2003-09-27 23:09:38.000000000 -0700
@@ -58,7 +58,7 @@
 #include "wipe.h"
 #include "blkdev.h"
 
-extern int errno;
+/*extern int errno;*/
 extern int exit_code;
 extern char *argvzero;
 extern struct opt_s options;
diff -uprN wipe-2.1.0.old/file.c wipe-2.1.0/file.c
--- wipe-2.1.0.old/file.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/file.c	2003-09-27 23:09:38.000000000 -0700
@@ -42,7 +42,7 @@
 #include "rand.h"
 #include "prompt.h"
 
-extern int errno;
+/*extern int errno;*/
 extern int exit_code;
 extern char *argvzero;
 
diff -uprN wipe-2.1.0.old/io.c wipe-2.1.0/io.c
--- wipe-2.1.0.old/io.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/io.c	2003-09-27 23:09:38.000000000 -0700
@@ -42,7 +42,7 @@
 #include "main.h"
 #include "io.h"
 
-extern int errno;
+/*extern int errno;*/
 extern int exit_code;
 extern char *argvzero;
 extern struct opt_s options;
diff -uprN wipe-2.1.0.old/main.c wipe-2.1.0/main.c
--- wipe-2.1.0.old/main.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/main.c	2003-09-27 23:09:38.000000000 -0700
@@ -45,7 +45,7 @@
 char *argvzero;                       /* what wipe was called as, argv[0]        */
 int exit_code = 0;                    /* final exit code for main()              */
 
-extern int errno;
+/*extern int errno;*/
 
 struct opt_s options;                 /* cmd line options                        */
 
diff -uprN wipe-2.1.0.old/rand.c wipe-2.1.0/rand.c
--- wipe-2.1.0.old/rand.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/rand.c	2003-09-27 23:09:52.000000000 -0700
@@ -25,6 +23,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <errno.h>
 
 #include "std.h"
 #include "io.h"
@@ -33,7 +32,7 @@
 #include "rand.h"
 
 extern char *argvzero;
-extern int errno, exit_code;
+extern int exit_code;
 extern struct opt_s options;
 
 int entropyfd;
diff -uprN wipe-2.1.0.old/wipe.c wipe-2.1.0/wipe.c
--- wipe-2.1.0.old/wipe.c	2003-09-27 23:07:03.000000000 -0700
+++ wipe-2.1.0/wipe.c	2003-09-27 23:09:38.000000000 -0700
@@ -53,7 +53,7 @@
 #include "rand.h"
 #include "wipe.h"
 
-extern int errno;
+/*extern int errno;*/
 extern int exit_code;
 extern char *argvzero;
 extern struct opt_s options;
