--- abi/src/text/fmt/xp/fl_TableLayout.cpp.orig	2004-07-13 14:45:17.834180615 +0200
+++ abi/src/text/fmt/xp/fl_TableLayout.cpp	2004-07-13 14:45:30.185606358 +0200
@@ -1044,7 +1044,7 @@
 	/* table-border properties:
 	 */
 	const char * pszColor = NULL;
-	pSectionAP->getProperty ("color", reinterpret_cast<const XML_Char *>(pszColor));
+	pSectionAP->getProperty ("color", (const XML_Char *&)pszColor);
 	if (pszColor)
 		UT_parseColor (pszColor, m_colorDefault);
 	else
@@ -1054,9 +1054,9 @@
 	const char * pszBorderStyle = NULL;
 	const char * pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("bot-color",       reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("bot-style",       reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("bot-thickness",   reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("bot-color",       (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("bot-style",       (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("bot-thickness",   (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineBottom);
 
@@ -1064,9 +1064,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("left-color",      reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("left-style",      reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("left-thickness",  reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("left-color",      (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("left-style",      (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("left-thickness",  (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineLeft);
 
@@ -1074,9 +1074,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("right-color",     reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("right-style",     reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("right-thickness", reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("right-color",     (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("right-style",     (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("right-thickness", (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineRight);
 
@@ -1084,9 +1084,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("top-color",       reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("top-style",       reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("top-thickness",   reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("top-color",       (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("top-style",       (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("top-thickness",   (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineTop);
 
@@ -1098,9 +1098,9 @@
 	const char * pszBgColor = NULL;
 	const char * pszBackgroundColor = NULL;
 
-	pSectionAP->getProperty ("bg-style",         reinterpret_cast<const XML_Char *>(pszBgStyle));
-	pSectionAP->getProperty ("bgcolor",          reinterpret_cast<const XML_Char *>(pszBgColor));
-	pSectionAP->getProperty ("background-color", reinterpret_cast<const XML_Char *>(pszBackgroundColor));
+	pSectionAP->getProperty ("bg-style",         (const XML_Char *&)pszBgStyle);
+	pSectionAP->getProperty ("bgcolor",          (const XML_Char *&)pszBgColor);
+	pSectionAP->getProperty ("background-color", (const XML_Char *&)pszBackgroundColor);
 
 	s_background_properties (pszBgStyle, pszBgColor, pszBackgroundColor, m_background);
 }
@@ -1826,15 +1826,15 @@
 	/* cell-border properties:
 	 */
 	const char * pszColor = NULL;
-	pSectionAP->getProperty ("color", reinterpret_cast<const XML_Char *>(pszColor));
+	pSectionAP->getProperty ("color", (const XML_Char *&)pszColor);
 
 	const char * pszBorderColor = NULL;
 	const char * pszBorderStyle = NULL;
 	const char * pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("bot-color",       reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("bot-style",       reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("bot-thickness",   reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("bot-color",       (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("bot-style",       (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("bot-thickness",   (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineBottom);
 
@@ -1842,9 +1842,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("left-color",      reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("left-style",      reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("left-thickness",  reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("left-color",      (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("left-style",      (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("left-thickness",  (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineLeft);
 
@@ -1852,9 +1852,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("right-color",     reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("right-style",     reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("right-thickness", reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("right-color",     (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("right-style",     (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("right-thickness", (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineRight);
 
@@ -1862,9 +1862,9 @@
 	pszBorderStyle = NULL;
 	pszBorderWidth = NULL;
 
-	pSectionAP->getProperty ("top-color",       reinterpret_cast<const XML_Char *>(pszBorderColor));
-	pSectionAP->getProperty ("top-style",       reinterpret_cast<const XML_Char *>(pszBorderStyle));
-	pSectionAP->getProperty ("top-thickness",   reinterpret_cast<const XML_Char *>(pszBorderWidth));
+	pSectionAP->getProperty ("top-color",       (const XML_Char *&)pszBorderColor);
+	pSectionAP->getProperty ("top-style",       (const XML_Char *&)pszBorderStyle);
+	pSectionAP->getProperty ("top-thickness",   (const XML_Char *&)pszBorderWidth);
 
 	s_border_properties (pszBorderColor, pszBorderStyle, pszBorderWidth, pszColor, m_lineTop);
 
@@ -1876,9 +1876,9 @@
 	const char * pszBgColor = NULL;
 	const char * pszBackgroundColor = NULL;
 
-	pSectionAP->getProperty ("bg-style",         reinterpret_cast<const XML_Char *>(pszBgStyle));
-	pSectionAP->getProperty ("bgcolor",          reinterpret_cast<const XML_Char *>(pszBgColor));
-	pSectionAP->getProperty ("background-color", reinterpret_cast<const XML_Char *>(pszBackgroundColor));
+	pSectionAP->getProperty ("bg-style",         (const XML_Char *&)pszBgStyle);
+	pSectionAP->getProperty ("bgcolor",          (const XML_Char *&)pszBgColor);
+	pSectionAP->getProperty ("background-color", (const XML_Char *&)pszBackgroundColor);
 
 	s_background_properties (pszBgStyle, pszBgColor, pszBackgroundColor, m_background);
 }
