--- NVIDIA-Linux-x86-1.0-4496-pkg2/usr/src/nv/nv-linux.h.old	2003-09-10 19:30:03.734499232 -0700
+++ NVIDIA-Linux-x86-1.0-4496-pkg2/usr/src/nv/nv-linux.h	2003-09-10 19:30:36.522514696 -0700
@@ -358,6 +358,7 @@ extern struct mem_track_t *fp_list;
 #define NV_VMA_PRIVATE(vma)           ((vma)->vm_private_data)
 
 #ifdef KERNEL_2_5
+#  define kdev_val(d)                   ((unsigned int)(d))
 #  define NV_DEVICE_NUMBER(_minor)      ((kdev_val(_minor)) & 0x0f)
 #  define NV_IS_CONTROL_DEVICE(_minor)  (((kdev_val(_minor)) & 0xff) == 0xff)
 #  define NV_IS_SUSER()                 capable(CAP_SYS_ADMIN)
