diff -Nur MPlayer-0.92/configure MPlayer-0.92-alsa-1.0/configure
--- MPlayer-0.92/configure	2003-08-09 16:12:30.000000000 +0200
+++ MPlayer-0.92-alsa-1.0/configure	2003-11-21 13:16:13.000000000 +0100
@@ -3425,7 +3425,7 @@
   cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-sys'
   cat > $TMPC << EOF
 #include <alsa/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==9)); }
+int main(void) { return (!((SND_LIB_MAJOR==0 && SND_LIB_MINOR==9) || SND_LIB_MAJOR==1)); }
 EOF
   cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-alsa'
 fi
diff -Nur MPlayer-0.92/libao2/ao_alsa9.c MPlayer-0.92-alsa-1.0/libao2/ao_alsa9.c
--- MPlayer-0.92/libao2/ao_alsa9.c	2003-08-09 16:12:34.000000000 +0200
+++ MPlayer-0.92-alsa-1.0/libao2/ao_alsa9.c	2003-11-21 13:22:29.000000000 +0100
@@ -22,6 +22,7 @@
 #if HAVE_SYS_ASOUNDLIB_H
 #include <sys/asoundlib.h>
 #elif HAVE_ALSA_ASOUNDLIB_H
+#define ALSA_PCM_OLD_HW_PARAMS_API
 #include <alsa/asoundlib.h>
 #else
 #error "asoundlib.h is not in sys/ or alsa/ - please bugreport"
