--- Makefile~	2006-03-01 15:02:34.000000000 +0100
+++ Makefile	2006-03-01 15:04:39.000000000 +0100
@@ -476,11 +476,6 @@
 	done
 endif
 	@if test ! -d $(DATADIR) ; then mkdir -p $(DATADIR) ; fi
-	@if test ! -d $(DATADIR)/font ; then mkdir -p $(DATADIR)/font ; fi
-	@if test ! -f $(DATADIR)/font/font.desc ; then \
-	echo "*** Download font at http://www.mplayerhq.hu/dload.html" ; \
-	echo "*** for OSD/Subtitles support and extract to $(DATADIR)/font/" ; \
-	fi
 ifeq ($(GUI),yes)
 	@if test ! -d $(DATADIR)/Skin ; then mkdir -p $(DATADIR)/Skin ; fi
 	@echo "*** Download skin(s) at http://www.mplayerhq.hu/dload.html"
