diff -Nur MPlayer-1.0pre2/configure MPlayer-1.0pre2-fontconfig/configure
--- MPlayer-1.0pre2/configure	2003-10-05 17:34:40.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/configure	2003-11-28 14:19:08.000000000 +0100
@@ -165,6 +165,7 @@
   --disable-css          Disable old-style libcss DVD support [autodetect]
   --disable-cdparanoia   Disable cdparanoia support [autodetect]
   --disable-freetype     Disable freetype2 font rendering support [autodetect]
+  --disable-fontconfig   Disable fontconfig font lookup support [autodetect]
   --disable-unrarlib     Disable Unique RAR File Library [enabled]
   --enable-menu          Enable OSD menu support (NOT DVD MENU) [disabled]
   --disable-sortsub      Disable subtitles sorting [enabled]
@@ -1157,6 +1158,7 @@
 _cdparanoia=auto
 _big_endian=auto
 _freetype=auto
+_fontconfig=auto
 _shared_pp=no
 _menu=no
 _qtx=auto
@@ -1371,6 +1373,8 @@
   --disable-big-endian) _big_endian=no  ;;
   --enable-freetype)    _freetype=yes   ;;
   --disable-freetype)   _freetype=no    ;;
+  --enable-fontconfig)  _fontconfig=yes ;;
+  --disable-fontconfig) _fontconfig=no  ;;
   --enable-unrarlib)	_unrarlib=yes	;;
   --disable-unrarlib)	_unrarlib=no	;;
   --enable-ftp)         _ftp=yes        ;;
@@ -4151,6 +4155,40 @@
 fi
 echores "$_freetype"
 
+if test "$_freetype" = no ; then
+    _fontconfig=no
+fi
+echocheck "fontconfig"
+if test "$_fontconfig" = auto ; then
+    if ( pkg-config fontconfig --modversion) > /dev/null 2>&1 ; then
+        cat > $TMPC << EOF
+#include <stdio.h>
+#include <fontconfig/fontconfig.h>
+int main()
+{
+    int err = FcInit();
+    if(err == FcFalse){
+        printf("Couldn't initialize fontconfig lib\n");
+        exit(err);
+    }
+    return 0;
+    
+}
+EOF
+        _fontconfig=no
+        cc_check `pkg-config fontconfig --cflags --libs` && ( $TMPO >> "$TMPLOG" ) && _fontconfig=yes
+    else
+        _fontconfig=no
+    fi
+fi
+if test "$_fontconfig" = yes ; then
+    _def_fontconfig='#define HAVE_FONTCONFIG'
+    _inc_fontconfig=`pkg-config fontconfig --cflags`
+    _ld_fontconfig=`pkg-config fontconfig --libs`
+else
+    _def_fontconfig='#undef HAVE_FONTCONFIG'
+fi
+echores "$_fontconfig"
 
 echocheck "fribidi with charsets"
 if test "$_fribidi" = yes ; then
@@ -5728,6 +5766,8 @@
 CDPARANOIA_LIB = $_ld_cdparanoia
 FREETYPE_INC = $_inc_freetype
 FREETYPE_LIB = $_ld_freetype
+FONTCONFIG_INC = $_inc_fontconfig
+FONTCONFIG_LIB = $_ld_fontconfig
 FRIBIDI_INC = $_inc_fribidi
 FRIBIDI_LIB = $_ld_fribidi
 LIBLZO_LIB= $_ld_liblzo
@@ -6168,6 +6208,9 @@
 /* enable FreeType support */
 $_def_freetype
 
+/* enable Fontconfig support */
+$_def_fontconfig
+
 /* enable FriBiDi usage */
 $_def_fribidi
 
diff -Nur MPlayer-1.0pre2/DOCS/de/documentation.html MPlayer-1.0pre2-fontconfig/DOCS/de/documentation.html
--- MPlayer-1.0pre2/DOCS/de/documentation.html	2003-07-09 22:10:34.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/DOCS/de/documentation.html	2003-11-28 14:25:42.000000000 +0100
@@ -1108,6 +1108,11 @@
         ~/.mplayer/subfont.ttf</CODE></LI>
     </UL>
   </LI>
+  <LI>Mit Fontconfig in Kombination mit freetype erwartet der -font Parameter
+    den fontconfig Namen der Schriftart:
+    <BR>
+    <CODE>-font 'Bitstream Vera Sans'</CODE>
+  </LI>
 </UL>
 
 <P>Wenn du eine Nicht-TrueType-Schrift benutzt, dann entpacke die Dateien  nach
diff -Nur MPlayer-1.0pre2/DOCS/de/mplayer.1 MPlayer-1.0pre2-fontconfig/DOCS/de/mplayer.1
--- MPlayer-1.0pre2/DOCS/de/mplayer.1	2003-09-21 16:21:40.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/DOCS/de/mplayer.1	2003-11-28 14:19:43.000000000 +0100
@@ -912,6 +912,8 @@
 Die \-subfont-*-Optionen sind nur verfügbar, wenn FreeType eincompieliert
 wurde. Wenn die FreeType-Unterstützung aktiviert wurde, dann kann die
 alte Schriftunterstützung nicht mehr benutzt werden.
+.br
+Bei Fontconfig gibt diese Option den fontconfig Namen der Schriftart an.
 
 .I BEISPIELE:
 .PD 0
@@ -919,6 +921,8 @@
 \-font ~/\:.mplayer/\:arial\-14/\:font.desc
 .br
 \-font ~/\:.mplayer/\:arialuni.ttf
+.br
+\-font 'Bitstream Vera Sans'
 .RE
 .PD 1
 .
diff -Nur MPlayer-1.0pre2/DOCS/en/documentation.html MPlayer-1.0pre2-fontconfig/DOCS/en/documentation.html
--- MPlayer-1.0pre2/DOCS/en/documentation.html	2003-10-04 03:45:14.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/DOCS/en/documentation.html	2003-11-28 14:23:27.000000000 +0100
@@ -1016,6 +1016,11 @@
       <LI>create a symlink: <CODE>ln -s /path/to/arial.ttf ~/.mplayer/subfont.ttf</CODE></LI>
     </UL>
   </LI>
+  <LI>using Fontconfig in combination with freetype, you use the -font
+    parameter with a fontconfig font name:
+    <BR>
+    <CODE>-font 'Bitstream Vera Sans'</CODE>
+  </LI>
   <LI>download ready-to-use font packages from MPlayer site.
     Note: Currently available fonts are limited for iso 8859-1/2 support,
     but there are some other (including Korean, Russian, 8859-8 etc) fonts
diff -Nur MPlayer-1.0pre2/DOCS/en/mplayer.1 MPlayer-1.0pre2-fontconfig/DOCS/en/mplayer.1
--- MPlayer-1.0pre2/DOCS/en/mplayer.1	2003-10-04 00:30:02.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/DOCS/en/mplayer.1	2003-11-28 14:19:51.000000000 +0100
@@ -1112,6 +1112,8 @@
 With FreeType, this option determines path to the text font file.
 .br
 The \-subfont-* options are available only with FreeType support compiled in.
+.br
+With Fontconfig, this option determines the fontconfig font name.
 
 .I EXAMPLE:
 .PD 0
@@ -1119,6 +1121,8 @@
 \-font ~/\:.mplayer/\:arial\-14/\:font.desc
 .br
 \-font ~/\:.mplayer/\:arialuni.ttf
+.br
+\-font 'Bitstream Vera Sans'
 .RE
 .PD 1
 .
diff -Nur MPlayer-1.0pre2/libvo/font_load_ft.c MPlayer-1.0pre2-fontconfig/libvo/font_load_ft.c
--- MPlayer-1.0pre2/libvo/font_load_ft.c	2003-06-09 14:15:47.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/libvo/font_load_ft.c	2003-11-28 14:20:13.000000000 +0100
@@ -23,6 +23,10 @@
 #include <freetype/freetype.h>
 #include <freetype/ftglyph.h>
 
+#ifdef HAVE_FONTCONFIG
+#include <fontconfig/fontconfig.h>
+#endif
+
 #include "../bswap.h"
 #include "font_load.h"
 #include "mp_msg.h"
@@ -1112,6 +1116,11 @@
 
 void load_font_ft(int width, int height) 
 {
+#ifdef HAVE_FONTCONFIG
+  FcPattern *fc_pattern;
+  FcChar8 *s;
+  FcBool scalable;
+#endif
     vo_image_width = width;
     vo_image_height = height;
 
@@ -1120,9 +1129,29 @@
 
     if (vo_font) free_font_desc(vo_font);
 
+    
 #ifdef USE_OSD
+#ifdef HAVE_FONTCONFIG
+    FcInit();
+    fc_pattern = FcNameParse(font_name);
+    FcConfigSubstitute(0, fc_pattern, FcMatchPattern);
+    FcDefaultSubstitute(fc_pattern);
+    fc_pattern = FcFontMatch(0, fc_pattern, 0);
+    FcPatternGetBool(fc_pattern, FC_SCALABLE, 0, &scalable);
+    if (scalable != FcTrue) {
+      fc_pattern = FcNameParse("sans-serif");
+      FcConfigSubstitute(0, fc_pattern, FcMatchPattern);
+      FcDefaultSubstitute(fc_pattern);
+      fc_pattern = FcFontMatch(0, fc_pattern, 0);
+    }
+    // s doesn't need to be freed according to fontconfig docs
+    FcPatternGetString(fc_pattern, FC_FILE, 0, &s);
+    vo_font=read_font_desc_ft(s, width, height);
+    free(fc_pattern);
+#else
     vo_font=read_font_desc_ft(font_name, width, height);
 #endif
+#endif
 }
 
 #endif /* HAVE_FREETYPE */
diff -Nur MPlayer-1.0pre2/Makefile MPlayer-1.0pre2-fontconfig/Makefile
--- MPlayer-1.0pre2/Makefile	2003-10-05 00:00:01.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/Makefile	2003-11-28 14:19:08.000000000 +0100
@@ -40,6 +40,8 @@
 
 CFLAGS = $(OPTFLAGS) -Ilibmpdemux -Iloader -Ilibvo $(FREETYPE_INC) $(EXTRA_INC) $(CDPARANOIA_INC) $(SDL_INC) $(X11_INC) $(FRIBIDI_INC) $(DVB_INC) # -Wall
 
+COMMON_LIBS += $(FONTCONFIG_LIB)
+CFLAGS += $(FONTCONFIG_INC)
 PARTS = libmpdemux libmpcodecs mp3lib liba52 libmpeg2 libavcodec libao2 drivers osdep postproc input libvo libaf
 ifeq ($(INTERNAL_FAAD),yes)
 COMMON_LIBS += libfaad2/libfaad2.a 
diff -Nur MPlayer-1.0pre2/mencoder.c MPlayer-1.0pre2-fontconfig/mencoder.c
--- MPlayer-1.0pre2/mencoder.c	2003-08-13 18:29:00.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/mencoder.c	2003-11-28 14:19:08.000000000 +0100
@@ -418,6 +418,10 @@
 #ifdef USE_OSD
 #ifdef HAVE_FREETYPE
   init_freetype();
+#ifdef HAVE_FONTCONFIG
+  if(!font_name)
+    font_name = "sans-serif";
+#endif
 #else
   if(font_name){
        vo_font=read_font_desc(font_name,font_factor,verbose>1);
diff -Nur MPlayer-1.0pre2/mplayer.c MPlayer-1.0pre2-fontconfig/mplayer.c
--- MPlayer-1.0pre2/mplayer.c	2003-10-04 03:24:50.000000000 +0200
+++ MPlayer-1.0pre2-fontconfig/mplayer.c	2003-11-28 14:19:08.000000000 +0100
@@ -1054,8 +1054,15 @@
 
 //------ load global data first ------
 
-#ifdef USE_OSD
 // check font
+#ifdef USE_OSD
+#ifdef HAVE_FREETYPE
+  init_freetype();
+#ifdef HAVE_FONTCONFIG
+  if(!font_name)
+    font_name = "sans-serif";
+#endif
+#else
   if(font_name){
        vo_font=read_font_desc(font_name,font_factor,verbose>1);
        if(!vo_font) mp_msg(MSGT_CPLAYER,MSGL_ERR,MSGTR_CantLoadFont,font_name);
@@ -1065,9 +1072,6 @@
        if(!vo_font)
        vo_font=read_font_desc(MPLAYER_DATADIR "/font/font.desc",font_factor,verbose>1);
   }
-#ifdef HAVE_FREETYPE
-  if (!vo_font)
-	init_freetype();
 #endif
 #endif
   vo_init_osd();
