diff -u ElectricFence-2.1.orig/ElectricFence.shar ElectricFence-2.1/ElectricFence.shar
--- ElectricFence-2.1.orig/ElectricFence.shar	1998-03-12 20:52:33.000000000 +0100
+++ ElectricFence-2.1/ElectricFence.shar	2003-04-15 16:27:51.000000000 +0200
@@ -1365,10 +1365,10 @@
 X
 Xstatic caddr_t	startAddr = (caddr_t) 0;
 X
-X#if ( !defined(sgi) && !defined(_AIX) )
+X/*#if ( !defined(sgi) && !defined(_AIX) )
 Xextern int	sys_nerr;
 Xextern char *	sys_errlist[];
-X#endif
+X#endif*/
 X
 Xstatic const char *
 XstringErrorReport(void)
@@ -1378,8 +1378,8 @@
 X#elif ( defined(_AIX) )
 X	return strerror(errno);
 X#else
-X	if ( errno > 0 && errno < sys_nerr )
-X		return sys_errlist[errno];
+X	if ( errno > 0 )
+X		return strerror(errno);
 X	else
 X		return "Unknown error.\n";
 X#endif
diff -u ElectricFence-2.1.orig/page.c ElectricFence-2.1/page.c
--- ElectricFence-2.1.orig/page.c	1996-04-27 00:57:38.000000000 +0200
+++ ElectricFence-2.1/page.c	2003-04-15 16:28:14.000000000 +0200
@@ -29,10 +29,11 @@
 
 static caddr_t	startAddr = (caddr_t) 0;
 
-#if ( !defined(sgi) && !defined(_AIX) )
+/*#if ( !defined(sgi) && !defined(_AIX) )
 extern int	sys_nerr;
 extern char *	sys_errlist[];
 #endif
+*/
 
 static const char *
 stringErrorReport(void)
@@ -42,8 +43,8 @@
 #elif ( defined(_AIX) )
 	return strerror(errno);
 #else
-	if ( errno > 0 && errno < sys_nerr )
-		return sys_errlist[errno];
+	if ( errno > 0 )
+		return strerror(errno);
 	else
 		return "Unknown error.\n";
 #endif
