diff -ruN python-ldap-2.0.0pre05/setup.cfg python-ldap/setup.cfg
--- python-ldap-2.0.0pre05/setup.cfg	2002-07-20 11:56:15.000000000 +0200
+++ python-ldap/setup.cfg	2002-08-06 22:07:19.000000000 +0200
@@ -5,8 +5,8 @@
 
 class = OpenLDAP2
 
-library_dirs = /usr/local/openldap2/lib
-include_dirs = /usr/local/openldap2/include
+library_dirs = /usr/lib
+include_dirs = /usr/include
 
 # Standard build, basic functionality
 libs = ldap lber
