diff -Naur oo_1.1_src/openssl/makefile.mk oo_1.1_src.new/openssl/makefile.mk
--- oo_1.1_src/openssl/makefile.mk	2003-04-29 03:00:54.000000000 +1000
+++ oo_1.1_src.new/openssl/makefile.mk	2003-10-27 19:29:01.000000000 +1100
@@ -77,9 +77,9 @@
 .EXPORT : LD_LIBRARY_PATH
 .ENDIF
 
-TARFILE_NAME=openssl-0.9.5a
+TARFILE_NAME=openssl-0.9.7c
 
-PATCH_FILE_NAME=openssl-0.9.5a.patch
+#PATCH_FILE_NAME=openssl-0.9.5a.patch
 
 .IF "$(GUI)"=="UNX"
 #relative to CONFIGURE_DIR
