--- misc/build/Python-2.3.2/setup.py	2003-09-22 11:19:45.000000000 +1000
+++ misc/build/Python-2.3.2/setup.py.new	2003-10-20 11:15:51.000000000 +1000
@@ -594,7 +594,7 @@
             if find_file("ndbm.h", inc_dirs, []) is not None:
                 # Some systems have -lndbm, others don't
                 if self.compiler.find_library_file(lib_dirs, 'ndbm'):
-                    ndbm_libs = ['ndbm']
+                    ndbm_libs = ['ndbm', 'gdm_compat']
                 else:
                     ndbm_libs = []
                 exts.append( Extension('dbm', ['dbmmodule.c'],
