--- jigdo-0.7.0.orig/src/cachefile.cc	2003-11-16 23:15:18.000000000 +0100
+++ jigdo-0.7.0/src/cachefile.cc	2003-11-16 23:16:48.000000000 +0100
@@ -36,7 +36,7 @@ CacheFile::CacheFile(const char* dbName)
   db->set_cachesize(db, 0, 4*1024*1024, 1);
 
   // Use a btree, create database file if not yet present
-  e = db->open(db, dbName, "jigdo filecache v0", DB_BTREE, DB_CREATE, 0666);
+  e = db->open(db, NULL, dbName, "jigdo filecache v0", DB_BTREE, DB_CREATE, 0666);
   if (e != 0) {
     // Re-close, in case it is necessary
     db->close(db, 0);
@@ -47,7 +47,7 @@ CacheFile::CacheFile(const char* dbName)
 #   if DEBUG
     cerr << "Cache file corrupt, recreating it" << endl;
 #   endif
-    if (db->open(db, dbName, "jigdo filecache v0", DB_BTREE,
+    if (db->open(db, NULL, dbName, "jigdo filecache v0", DB_BTREE,
                  DB_CREATE | DB_TRUNCATE, 0666) != 0)
       throw DbError(e);
   }
