diff -urpN esdl-0.93.0909.old/c_src/Makefile esdl-0.93.0909/c_src/Makefile
--- esdl-0.93.0909.old/c_src/Makefile	2003-10-18 18:41:36.446857712 -0700
+++ esdl-0.93.0909/c_src/Makefile	2003-10-18 18:42:23.278738184 -0700
@@ -18,7 +18,7 @@ else
 #GL_INCLUDE_DIR      = OpenGL
 #USE_GL_INCLUDE_DIR  = -DFULL_PATH_TO_GL_INCLUDE_DIR
 
-  GL_LIBS		= -L$(OGLDIR)/lib -lGL -lGLU 
+  GL_LIBS		= -L/usr/X11/lib -L$(OGLDIR)/lib -lGL -lGLU 
   GL_INCS		= -I$(OGLDIR)/include/$(GL_INCLUDE_DIR)
 
   SDL_LIBS		= $(shell sdl-config --libs)
