--- lpr-0.72/lpd.init	Sat Mar  2 23:03:59 2002
+++ lpr-0.72/lpd.init   Sat Mar  2 23:03:59 2002
@@ -16,15 +16,15 @@
 	export KILLCMD="killproc lpd"
 else
-	export STARTCMD="startproc /usr/sbin/lpd"
-	export KILLCMD="killproc /usr/sbin/lpd"
+	export STARTCMD="/usr/sbin/lpd"
+	export KILLCMD="pkill lpd"
 fi
 
 
 # Source networking configuration.
-if [ -f /etc/sysconfig/network ] ; then
-	. /etc/sysconfig/network
-fi
+#if [ -f /etc/init.d/networking.sh ] ; then
+#	. /etc/init.d/networking.sh
+#fi
 
-# Check that networking is up.
+#Check that networking is up.
 [ ${NETWORKING}x = "nox" ] && exit 0
 
@@ -43,5 +43,5 @@
 	RETVAL=$?
         echo
-        [ $RETVAL -eq 0 ] && touch /var/lock/subsys/lpd
+        [ $RETVAL -eq 0 ] && touch /var/lock/lpd
         ;;
   stop)
@@ -51,5 +51,5 @@
 	RETVAL=$?
         echo
-        [ $RETVAL -eq 0 ] && rm -f /var/lock/subsys/lpd
+        [ $RETVAL -eq 0 ] && rm -f /var/lock/lpd
         ;;
   status)
