--- app/paint-funcs/paint-funcs.h.old	2004-05-26 18:15:55.061372024 +0200
+++ app/paint-funcs/paint-funcs.h	2004-05-26 18:14:19.666874192 +0200
@@ -356,7 +356,7 @@
 					   gint,
                                            GimpLayerModeEffects,
                                            gboolean *,
-                                           CombinationMode);
+                                           InitialMode);
 
 void  combine_regions                     (PixelRegion *, PixelRegion *,
 					   PixelRegion *, PixelRegion *,
