--- Makefile.old	2004-10-08 18:54:39.681392295 +0200
+++ Makefile	2004-10-08 18:55:10.512462707 +0200
@@ -388,8 +388,7 @@
 	- rm $(mandir)/man8/udevd.8
 	- rm $(mandir)/man8/udevsend.8
 
-install: install-initscript install-config install-man install-dev.d all
-	$(INSTALL) -d $(DESTDIR)$(udevdir)
+install: install-man install-dev.d all
 	$(INSTALL) -d $(DESTDIR)$(hotplugdir)
 	$(INSTALL_PROGRAM) -D $(ROOT) $(DESTDIR)$(sbindir)/$(ROOT)
 	$(INSTALL_PROGRAM) -D $(DAEMON) $(DESTDIR)$(sbindir)/$(DAEMON)
@@ -400,10 +399,6 @@
 	- ln -f -s $(sbindir)/udev $(DESTDIR)$(sbindir)/$(STARTER)
 	- ln -f -s $(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/10-udev.hotplug
 	- ln -f -s $(sbindir)/$(WAIT) $(DESTDIR)$(hotplugdir)/05-wait_for_sysfs.hotplug
-ifndef DESTDIR
-	- killall udevd
-	- rm -f $(udevdir)/.udev.tdb
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
