--- Makefile.old	2005-07-23 09:27:02.000000000 -0700
+++ Makefile	2005-07-24 22:52:21.000000000 -0700
@@ -145,7 +145,8 @@
 	udev_db.o			\
 	udev_rules.o			\
 	udev_rules_parse.o		\
-	udev_libc_wrapper.o
+	udev_libc_wrapper.o             \
+	udev_logger.o
 
 OBJS = \
 	udev.a			\
@@ -377,11 +378,6 @@
 	$(INSTALL_PROGRAM) -D $(INFO) $(DESTDIR)$(usrbindir)/$(INFO)
 	$(INSTALL_PROGRAM) -D $(TESTER) $(DESTDIR)$(usrbindir)/$(TESTER)
 	$(INSTALL_PROGRAM) -D $(STARTER) $(DESTDIR)$(sbindir)/$(STARTER)
-ifndef DESTDIR
-	- killall $(DAEMON)
-	- rm -rf $(udevdb)
-	- $(sbindir)/$(DAEMON) --daemon
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
