--- Makefile~	2004-03-25 11:09:50.000000000 +1100
+++ Makefile	2004-04-07 11:58:02.963362821 +1000
@@ -109,7 +109,6 @@
 	OPTIMIZATION+=-fstrict-aliasing
 endif
 ifeq ($(strip $(TARGET_ARCH)),i386)
-	OPTIMIZATION+=-march=i386
 	OPTIMIZATION += ${shell if $(CC) -mpreferred-stack-boundary=2 -S -o /dev/null -xc \
 		/dev/null >/dev/null 2>&1; then echo "-mpreferred-stack-boundary=2"; fi}
 	OPTIMIZATION += ${shell if $(CC) -malign-functions=0 -malign-jumps=0 -S -o /dev/null -xc \
@@ -354,8 +353,7 @@
 	$(INSTALL) -d $(DESTDIR)$(dev_ddir)
 	$(INSTALL) -d $(DESTDIR)$(dev_ddir)default/
 
-install: install-initscript install-config install-dev.d all
-	$(INSTALL) -d $(DESTDIR)$(udevdir)
+install: install-dev.d all
 	$(INSTALL) -d $(DESTDIR)$(hotplugdir)
 	$(INSTALL_PROGRAM) -D $(ROOT) $(DESTDIR)$(sbindir)/$(ROOT)
 	$(INSTALL_PROGRAM) -D $(DAEMON) $(DESTDIR)$(sbindir)/$(DAEMON)
@@ -370,10 +368,6 @@
 	$(INSTALL_DATA) -D udevd.8 $(DESTDIR)$(mandir)/man8/udevd.8
 	- ln -f -s udevd.8 $(DESTDIR)$(mandir)/man8/udevsend.8
 	- ln -f -s $(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/$(ROOT).hotplug
-ifndef DESTDIR
-	- killall udevd
-	- rm -f $(udevdir)/.udev.tdb
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
