--- Makefile.orig	2005-07-07 22:00:50.000000000 +0000
+++ Makefile	2005-07-13 09:27:00.000000000 +0000
@@ -366,8 +366,7 @@
 	- rm $(mandir)/man8/udevd.8
 	- rm $(mandir)/man8/udevsend.8
 
-install: install-config install-man all
-	$(INSTALL) -d $(DESTDIR)$(udevdir)
+install: install-man all
 	$(INSTALL_PROGRAM) -D $(ROOT) $(DESTDIR)$(sbindir)/$(ROOT)
 	$(INSTALL_PROGRAM) -D $(DAEMON) $(DESTDIR)$(sbindir)/$(DAEMON)
 	$(INSTALL_PROGRAM) -D $(SENDER) $(DESTDIR)$(sbindir)/$(SENDER)
@@ -375,11 +374,6 @@
 	$(INSTALL_PROGRAM) -D $(INFO) $(DESTDIR)$(usrbindir)/$(INFO)
 	$(INSTALL_PROGRAM) -D $(TESTER) $(DESTDIR)$(usrbindir)/$(TESTER)
 	$(INSTALL_PROGRAM) -D $(STARTER) $(DESTDIR)$(sbindir)/$(STARTER)
-ifndef DESTDIR
-	- killall $(DAEMON)
-	- $(sbindir)/$(DAEMON) --daemon
-	- rm -rf $(udevdb)
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
