--- Makefile.old	2005-08-13 19:10:50.000000000 +0200
+++ Makefile	2005-08-16 23:33:58.000000000 +0200
@@ -148,7 +148,8 @@
 	udev_utils.o			\
 	udev_utils_file.o		\
 	udev_utils_run.o		\
-	udev_libc_wrapper.o
+	udev_libc_wrapper.o		\
+	udev_logger.o
 
 OBJS = \
 	udev.a			\
@@ -387,11 +388,6 @@
 	$(INSTALL_PROGRAM) -D $(INFO) $(DESTDIR)$(usrbindir)/$(INFO)
 	$(INSTALL_PROGRAM) -D $(TESTER) $(DESTDIR)$(usrbindir)/$(TESTER)
 	$(INSTALL_PROGRAM) -D $(STARTER) $(DESTDIR)$(sbindir)/$(STARTER)
-ifndef DESTDIR
-	- killall $(DAEMON)
-	- rm -rf $(udevdb)
-	- $(sbindir)/$(DAEMON) --daemon
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
