diff -urpN udev-038.orig/Makefile udev-038/Makefile
--- udev-038.orig/Makefile	2004-10-15 13:20:14.560018248 -0700
+++ udev-038/Makefile	2004-10-15 13:20:36.005758000 -0700
@@ -413,8 +413,7 @@ uninstall-man:
 	- rm $(mandir)/man8/udevd.8
 	- rm $(mandir)/man8/udevsend.8
 
-install: install-initscript install-config install-man install-dev.d all
-	$(INSTALL) -d $(DESTDIR)$(udevdir)
+install: install-man install-dev.d all
 	$(INSTALL) -d $(DESTDIR)$(hotplugdir)
 	$(INSTALL_PROGRAM) -D $(ROOT) $(DESTDIR)$(sbindir)/$(ROOT)
 	$(INSTALL_PROGRAM) -D $(DAEMON) $(DESTDIR)$(sbindir)/$(DAEMON)
@@ -425,10 +424,6 @@ install: install-initscript install-conf
 	- ln -f -s $(sbindir)/udev $(DESTDIR)$(sbindir)/$(STARTER)
 	- ln -f -s $(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/10-udev.hotplug
 	- ln -f -s $(sbindir)/$(WAIT) $(DESTDIR)$(hotplugdir)/05-wait_for_sysfs.hotplug
-ifndef DESTDIR
-	- killall udevd
-	- rm -f $(udevdir)/.udev.tdb
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
