--- Makefile	2004-04-03 12:20:04.000000000 +0200
+++ Makefile.cp	2004-04-03 12:26:20.000000000 +0200
@@ -109,7 +109,6 @@
 	OPTIMIZATION+=-fstrict-aliasing
 endif
 ifeq ($(strip $(TARGET_ARCH)),i386)
-	OPTIMIZATION+=-march=i386
 	OPTIMIZATION += ${shell if $(CC) -mpreferred-stack-boundary=2 -S -o /dev/null -xc \
 		/dev/null >/dev/null 2>&1; then echo "-mpreferred-stack-boundary=2"; fi}
 	OPTIMIZATION += ${shell if $(CC) -malign-functions=0 -malign-jumps=0 -S -o /dev/null -xc \
@@ -384,7 +383,7 @@
 	- rm $(mandir)/man8/udevd.8
 	- rm $(mandir)/man8/udevsend.8
 
-install: install-initscript install-config install-man install-dev.d all
+install: install-man install-dev.d all
 	$(INSTALL) -d $(DESTDIR)$(udevdir)
 	$(INSTALL) -d $(DESTDIR)$(hotplugdir)
 	$(INSTALL_PROGRAM) -D $(ROOT) $(DESTDIR)$(sbindir)/$(ROOT)
@@ -394,10 +393,6 @@
 	$(INSTALL_PROGRAM) -D $(TESTER) $(DESTDIR)$(usrbindir)/$(TESTER)
 	$(INSTALL_PROGRAM) -D $(STARTER) $(DESTDIR)$(sbindir)/$(STARTER)
 	- ln -f -s $(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/$(ROOT).hotplug
-ifndef DESTDIR
-	- killall udevd
-	- rm -f $(udevdir)/.udev.tdb
-endif
 	@extras="$(EXTRAS)" ; for target in $$extras ; do \
 		echo $$target ; \
 		$(MAKE) prefix=$(prefix) LD="$(LD)" SYSFS="$(SYSFS)" \
