Index: lib/local.c
===================================================================
RCS file: /cvs/gnome/gftp/lib/local.c,v
retrieving revision 1.26
diff -u -r1.26 local.c
--- lib/local.c	12 Aug 2003 01:05:02 -0000	1.26
+++ lib/local.c	21 Sep 2003 16:46:57 -0000
@@ -315,23 +315,15 @@
     {
       closedir (lpd->dir);
       lpd->dir = NULL;
-      return (GFTP_ERETRYABLE);
+      return (GFTP_EFATAL);
     }
 
   fle->file = g_strdup (dirp->d_name);
   if (lstat (fle->file, &st) != 0)
-    {
-      closedir (lpd->dir);
-      lpd->dir = NULL;
-      return (GFTP_ERETRYABLE);
-    }
+    return (GFTP_ERETRYABLE);
 
   if (stat (fle->file, &fst) != 0)
-    {
-      closedir (lpd->dir);
-      lpd->dir = NULL;
-      return (GFTP_ERETRYABLE);
-    }
+    return (GFTP_ERETRYABLE);
 
   if ((user = g_hash_table_lookup (lpd->userhash, 
                                    GUINT_TO_POINTER(st.st_uid))) != NULL)
cvs server: Diffing src
cvs server: Diffing src/gtk
Index: src/gtk/transfer.c
===================================================================
RCS file: /cvs/gnome/gftp/src/gtk/transfer.c,v
retrieving revision 1.39
diff -u -r1.39 transfer.c
--- src/gtk/transfer.c	12 Aug 2003 01:05:02 -0000	1.39
+++ src/gtk/transfer.c	21 Sep 2003 16:46:58 -0000
@@ -104,8 +104,15 @@
       request->gotbytes = 0; 
       havedotdot = 0; 
       fle = g_malloc0 (sizeof (*fle));
-      while ((got = gftp_get_next_file (request, NULL, fle)) > 0)
+      while ((got = gftp_get_next_file (request, NULL, fle)) > 0 ||
+             got == GFTP_ERETRYABLE)
         { 
+          if (got < 0)
+            {
+              gftp_file_destroy (fle);
+              continue;
+            }
+
           request->gotbytes += got;
           if (strcmp (fle->file, ".") == 0)
             {
cvs server: Diffing src/text
Index: src/text/gftp-text.c
===================================================================
RCS file: /cvs/gnome/gftp/src/text/gftp-text.c,v
retrieving revision 1.26
diff -u -r1.26 gftp-text.c
--- src/text/gftp-text.c	24 Jul 2003 01:48:33 -0000	1.26
+++ src/text/gftp-text.c	21 Sep 2003 16:46:58 -0000
@@ -595,7 +595,7 @@
 {
   GList * files, * templist, * delitem;
   char *color, *filespec, *tempstr;
-  int sortcol, sortasds;
+  int sortcol, sortasds, got;
   gftp_file * fle;
   time_t curtime;
 
@@ -613,9 +613,10 @@
 
   files = NULL;
   fle = g_malloc0 (sizeof (*fle));
-  while (gftp_get_next_file (request, NULL, fle) > 0)
+  while ((got = gftp_get_next_file (request, NULL, fle)) > 0 ||
+         got == GFTP_ERETRYABLE)
     {
-      if (strcmp (fle->file, ".") == 0)
+      if (got < 0 || strcmp (fle->file, ".") == 0)
         {
           gftp_file_destroy (fle);
           continue;


