diff -uNr openal_cvs-src.old/include/AL/al.h openal_cvs-src/include/AL/al.h
--- openal_cvs-src.old/include/AL/al.h	2005-06-13 14:46:18.000000000 -0700
+++ openal_cvs-src/include/AL/al.h	2005-06-24 06:21:02.000000000 -0700
@@ -74,7 +74,7 @@
 /*
  * State retrieval
  */
-ALAPI const ALchar* ALAPIENTRY alGetString( ALenum param );
+ALAPI const ALubyte* ALAPIENTRY alGetString( ALenum param );
 
 ALAPI void ALAPIENTRY alGetBooleanv( ALenum param, ALboolean* data );
 
@@ -105,11 +105,11 @@
  * Query for the presence of an extension, and obtain any appropriate
  * function pointers and enum values.
  */
-ALAPI ALboolean ALAPIENTRY alIsExtensionPresent( const ALchar* extname );
+ALAPI ALboolean ALAPIENTRY alIsExtensionPresent( const ALubyte* extname );
 
-ALAPI void* ALAPIENTRY alGetProcAddress( const ALchar* fname );
+ALAPI void* ALAPIENTRY alGetProcAddress( const ALubyte* fname );
 
-ALAPI ALenum ALAPIENTRY alGetEnumValue( const ALchar* ename );
+ALAPI ALenum ALAPIENTRY alGetEnumValue( const ALubyte* ename );
 
 
 /*
diff -uNr openal_cvs-src.old/include/AL/alc.h openal_cvs-src/include/AL/alc.h
--- openal_cvs-src.old/include/AL/alc.h	2005-06-13 14:46:18.000000000 -0700
+++ openal_cvs-src/include/AL/alc.h	2005-06-24 06:21:02.000000000 -0700
@@ -61,9 +61,9 @@
 /*
  * Device Management
  */
-ALCAPI ALCdevice *     ALCAPIENTRY alcOpenDevice( const ALchar *devicename );
+ALCAPI ALCdevice *     ALCAPIENTRY alcOpenDevice( const ALubyte *devicename );
 
-ALCAPI ALCboolean      ALCAPIENTRY alcCloseDevice( ALCdevice *device );
+ALCAPI void      ALCAPIENTRY alcCloseDevice( ALCdevice *device );
 
 
 /*
@@ -78,17 +78,17 @@
  * Query for the presence of an extension, and obtain any appropriate
  * function pointers and enum values.
  */
-ALCAPI ALCboolean      ALCAPIENTRY alcIsExtensionPresent( ALCdevice *device, const ALCchar *extname );
+ALCAPI ALCboolean      ALCAPIENTRY alcIsExtensionPresent( ALCdevice *device, const ALCubyte *extname );
 
-ALCAPI void  *         ALCAPIENTRY alcGetProcAddress( ALCdevice *device, const ALCchar *funcname );
+ALCAPI void  *         ALCAPIENTRY alcGetProcAddress( ALCdevice *device, const ALCubyte *funcname );
 
-ALCAPI ALCenum         ALCAPIENTRY alcGetEnumValue( ALCdevice *device, const ALCchar *enumname );
+ALCAPI ALCenum         ALCAPIENTRY alcGetEnumValue( ALCdevice *device, const ALCubyte *enumname );
 
 
 /*
  * Query functions
  */
-ALCAPI const ALCchar * ALCAPIENTRY alcGetString( ALCdevice *device, ALCenum param );
+ALCAPI const ALCubyte * ALCAPIENTRY alcGetString( ALCdevice *device, ALCenum param );
 
 ALCAPI void            ALCAPIENTRY alcGetIntegerv( ALCdevice *device, ALCenum param, ALCsizei size, ALCint *data );
 
diff -uNr openal_cvs-src.old/include/AL/altypes.h openal_cvs-src/include/AL/altypes.h
--- openal_cvs-src.old/include/AL/altypes.h	2005-06-13 14:46:18.000000000 -0700
+++ openal_cvs-src/include/AL/altypes.h	2005-06-24 06:46:51.000000000 -0700
@@ -18,13 +18,13 @@
 typedef char ALboolean;
 
 /** OpenAL 8bit signed byte. */
-typedef char ALbyte;
+typedef signed char ALbyte;
 
 /** OpenAL 8bit unsigned byte. */
 typedef unsigned char ALubyte;
 
 /** OpenAL 8bit char */
-typedef char ALchar;
+typedef unsigned char ALchar;
 
 /** OpenAL 16bit signed short integer type. */
 typedef short ALshort;
@@ -45,7 +45,7 @@
 typedef double ALdouble;
 
 /** OpenAL 32bit type. */
-typedef unsigned int ALsizei;
+typedef int ALsizei;
 
 /** OpenAL void type (for params, not returns). */
 typedef void ALvoid;
diff -uNr openal_cvs-src.old/include/AL/alut.h openal_cvs-src/include/AL/alut.h
--- openal_cvs-src.old/include/AL/alut.h	2005-06-13 14:46:18.000000000 -0700
+++ openal_cvs-src/include/AL/alut.h	2005-06-24 06:21:02.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 
 #include "altypes.h"
+#include "alc.h"
 
 #ifdef _WIN32
 #define ALUTAPI
@@ -48,7 +49,8 @@
 
 #ifndef ALUT_NO_PROTOTYPES
 
-ALUTAPI ALboolean ALUTAPIENTRY alutInit(ALCchar *szDeviceName, ALCdevice **ppDevice, ALCcontext **ppContext);
+/* ALUTAPI ALboolean ALUTAPIENTRY alutInit(ALCbyte *szDeviceName, ALCdevice **ppDevice, ALCcontext **ppContext); */
+ALUTAPI void alutInit(int *argc, char *argv[]);
 ALUTAPI void ALUTAPIENTRY alutExit(ALvoid);
 
 #ifndef MACINTOSH_AL
diff -uNr openal_cvs-src.old/linux/src/al_ext.h openal_cvs-src/linux/src/al_ext.h
--- openal_cvs-src.old/linux/src/al_ext.h	2003-06-20 12:23:05.000000000 -0700
+++ openal_cvs-src/linux/src/al_ext.h	2005-06-24 06:21:06.000000000 -0700
@@ -31,6 +31,61 @@
 #define PASTE_(a1, a2)          PASTE__(a1 ## _ ## a2)
 #define PASTE(a1, a2)           PASTE_(a1, a2)
 
+
+/* stuff that was in the standard 1.0 headers, but shouldn't have been... */
+#ifndef AL_BYTE_LOKI
+#define AL_BYTE_LOKI                              0x100C
+#endif
+
+#ifndef AL_ENV_ROOM_IASIG
+#define AL_ENV_ROOM_IASIG                         0x3001
+#endif
+
+#ifndef AL_ENV_ROOM_HIGH_FREQUENCY_IASIG
+#define AL_ENV_ROOM_HIGH_FREQUENCY_IASIG          0x3002
+#endif
+
+#ifndef AL_ENV_ROOM_ROLLOFF_FACTOR_IASIG
+#define AL_ENV_ROOM_ROLLOFF_FACTOR_IASIG          0x3003
+#endif
+
+#ifndef AL_ENV_DECAY_TIME_IASIG                   
+#define AL_ENV_DECAY_TIME_IASIG                   0x3004
+#endif
+
+#ifndef AL_ENV_DECAY_HIGH_FREQUENCY_RATIO_IASIG
+#define AL_ENV_DECAY_HIGH_FREQUENCY_RATIO_IASIG   0x3005
+#endif
+
+#ifndef AL_ENV_REFLECTIONS_IASIG                  
+#define AL_ENV_REFLECTIONS_IASIG                  0x3006
+#endif
+
+#ifndef AL_ENV_REFLECTIONS_DELAY_IASIG
+#define AL_ENV_REFLECTIONS_DELAY_IASIG            0x3006
+#endif
+
+#ifndef AL_ENV_REVERB_IASIG
+#define AL_ENV_REVERB_IASIG                       0x3007
+#endif
+
+#ifndef AL_ENV_REVERB_DELAY_IASIG
+#define AL_ENV_REVERB_DELAY_IASIG                 0x3008
+#endif
+
+#ifndef AL_ENV_DIFFUSION_IASIG
+#define AL_ENV_DIFFUSION_IASIG                    0x3009
+#endif
+
+#ifndef AL_ENV_DENSITY_IASIG
+#define AL_ENV_DENSITY_IASIG                      0x300A
+#endif
+
+#ifndef AL_ENV_HIGH_FREQUENCY_REFERENCE_IASIG
+#define AL_ENV_HIGH_FREQUENCY_REFERENCE_IASIG     0x300B
+#endif
+
+
 /* bookkeeping stuff */
 
 /*
diff -uNr openal_cvs-src.old/linux/src/al_ext_iasig.c openal_cvs-src/linux/src/al_ext_iasig.c
--- openal_cvs-src.old/linux/src/al_ext_iasig.c	2005-04-11 06:51:11.000000000 -0700
+++ openal_cvs-src/linux/src/al_ext_iasig.c	2005-06-24 06:21:06.000000000 -0700
@@ -11,6 +11,7 @@
 #include "al_main.h"
 #include "al_debug.h"
 #include "al_error.h"
+#include "al_ext.h"
 
 #include <AL/al.h>
 
diff -uNr openal_cvs-src.old/linux/src/al_source.h openal_cvs-src/linux/src/al_source.h
--- openal_cvs-src.old/linux/src/al_source.h	2004-01-15 03:49:45.000000000 -0800
+++ openal_cvs-src/linux/src/al_source.h	2005-06-24 06:21:06.000000000 -0700
@@ -13,6 +13,7 @@
 #include <stdio.h>
 
 #include "al_types.h"
+#include "al_ext.h"
 
 /*
  * _alIsSource( ALuint sid )
diff -uNr openal_cvs-src.old/linux/Makefile.in openal_cvs-src/linux/Makefile.in
--- openal_cvs_src.old/linux/Makefile.in.old	2005-03-23 08:45:24.000000000 -0800
+++ openal_cvs_src/linux/Makefile.in	2005-06-19 13:31:58.000000000 -0700
@@ -65,7 +65,6 @@
 	$(INSTALL) -m 444 ../include/AL/alut.h $(DESTINC)/AL
 	$(INSTALL) -m 444 ../include/AL/altypes.h $(DESTINC)/AL
 	$(INSTALL) -m 444 ../include/AL/alctypes.h $(DESTINC)/AL
-	$(INSTALL) -m 444 ../include/AL/aluttypes.h $(DESTINC)/AL
 	$(INSTALL) -m 444 include/AL/alext.h $(DESTINC)/AL
 	$(INSTALL) -m 444 include/AL/alexttypes.h $(DESTINC)/AL
 
