--- linux/configure.in~	2003-01-13 23:28:25.000000000 +0100
+++ linux/configure.in	2003-04-26 00:25:37.000000000 +0200
@@ -87,22 +87,6 @@ enable_empty_locks=no
 
 CFLAGS="-O6 -fexpensive-optimizations -funroll-all-loops -funroll-loops -fomit-frame-pointer -finline-functions -ffast-math $CFLAGS"
 
-dnl add -march
-case "$target" in 
-    *i386*)
-    CFLAGS="$CFLAGS -march=i386" ;;
-    *i486*)
-    CFLAGS="$CFLAGS -march=i486" ;;
-    *i586*)
-    CFLAGS="$CFLAGS -march=i586" ;;
-    *i686*)
-    CFLAGS="$CFLAGS -march=i686" ;;
-    mips*)
-    CFLAGS="$CFLAGS -march=mips" ;;
-    *)
-    echo "Unknown target $target" ;;
-esac
-
 fi
 
 dnl keep CFLAGS variable if set, otherwise set it to just debug
