--- Makefile-old	2003-08-10 00:21:59.000000000 +0200
+++ Makefile	2003-09-02 12:38:28.000000000 +0200
@@ -66,6 +66,7 @@
 
 CFLAGS += \
 	-I. \
+	-I/usr/include/a52dec \
 	-I$(A52DIR)/include \
 	-I$(A52DIR)/liba52
 
@@ -132,18 +133,13 @@
 	$(OBJDIR)/audio \
 	$(OBJDIR)/video
 
-include Makefile.a52
-
-DIRS += $(A52DIRS)
-
-
-OUTPUT = $(OBJDIR)/libmpeg3.a
+OUTPUT = $(OBJDIR)/libmpeg3.so
 UTILS = $(OBJDIR)/mpeg3dump $(OBJDIR)/mpeg3toc  $(OBJDIR)/mpeg3cat
 
 #$(OBJDIR)/mpeg3split
 
 
-LIBS = -lm -lpthread
+LIBS = -lm -lpthread -Li686
 
 $(shell if ! test -d $(OBJDIR) \; then mkdir -p $(OBJDIR) \; fi )
 
@@ -156,18 +152,18 @@
 
 
 $(OUTPUT): $(OBJS) $(ASMOBJS) $(NASMOBJS) $(A52OBJS)
-	ar rcs $(OUTPUT) `cat $(OBJDIR)/objs`
+	ld -shared -o $(OUTPUT) `cat $(OBJDIR)/objs` $(LIBS) -la52
 
 
 
 $(OBJDIR)/mpeg3dump: $(OUTPUT) mpeg3dump.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3dump mpeg3dump.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3dump mpeg3dump.c -lmpeg3 $(LIBS)
 
 $(OBJDIR)/mpeg3toc: $(OUTPUT) mpeg3toc.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3toc mpeg3toc.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3toc mpeg3toc.c -lmpeg3 $(LIBS)
 
 $(OBJDIR)/mpeg3cat: $(OUTPUT) mpeg3cat.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3cat mpeg3cat.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3cat mpeg3cat.c -lmpeg3 $(LIBS)
 
 #$(OBJDIR)/mpeg3split: $(OUTPUT)
 #	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3split mpeg3split.c $(OUTPUT) $(LIBS)
@@ -184,7 +180,11 @@
 		-ldl
 
 install: 
+	mkdir -p /usr/include/libmpeg3
+	cp libmpeg3.h mpeg3private.h mpeg3protos.h /usr/include/libmpeg3
 	cp $(UTILS) /usr/bin
+	cp $(OUTPUT) /usr/lib
+	ldconfig
 
 clean:
 	rm -rf $(OBJDIR)
