diff -urN httpd-2.1.6-alpha/modules/ssl/ssl_engine_pphrase.c httpd/modules/ssl/ssl_engine_pphrase.c
--- httpd-2.1.6-alpha/modules/ssl/ssl_engine_pphrase.c	2005-02-04 15:28:49.000000000 -0500
+++ httpd/modules/ssl/ssl_engine_pphrase.c	2005-08-18 20:55:30.000000000 -0400
@@ -681,7 +681,7 @@
                     ap_log_error(APLOG_MARK, APLOG_ERR, 0, s,
                                  "Init: Failed to create pass phrase pipe '%s'",
                                  sc->server->pphrase_dialog_path);
-                    PEMerr(PEM_F_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
+                    PEMerr(PEM_F_PEM_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
                     memset(buf, 0, (unsigned int)bufsize);
                     return (-1);
                 }
@@ -691,7 +691,7 @@
         }
         else { /* sc->server->pphrase_dialog_type == SSL_PPTYPE_BUILTIN */ 
 #ifdef WIN32
-            PEMerr(PEM_F_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
+            PEMerr(PEM_F_PEM_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
             memset(buf, 0, (unsigned int)bufsize);
             return (-1);
 #else
@@ -742,7 +742,7 @@
                 i = EVP_read_pw_string(buf, bufsize, "", FALSE); 
             }  
             if (i != 0) {
-                PEMerr(PEM_F_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
+                PEMerr(PEM_F_PEM_DEF_CALLBACK,PEM_R_PROBLEMS_GETTING_PASSWORD);
                 memset(buf, 0, (unsigned int)bufsize);
                 return (-1);
             }
diff -urN httpd-2.1.6-alpha/modules/ssl/ssl_toolkit_compat.h httpd/modules/ssl/ssl_toolkit_compat.h
--- httpd-2.1.6-alpha/modules/ssl/ssl_toolkit_compat.h	2005-02-04 15:28:49.000000000 -0500
+++ httpd/modules/ssl/ssl_toolkit_compat.h	2005-08-18 20:55:30.000000000 -0400
@@ -162,8 +162,8 @@
 
 #define modssl_free free
 
-#ifndef PEM_F_DEF_CALLBACK
-#define PEM_F_DEF_CALLBACK PEM_F_DEF_CB
+#ifndef PEM_F_PEM_DEF_CALLBACK
+#define PEM_F_PEM_DEF_CALLBACK PEM_F_DEF_CB
 #endif
 
 #if SSLC_VERSION_NUMBER < 0x2000
