Index: configure.in
===================================================================
RCS file: /cvs/gnome/seahorse/configure.in,v
retrieving revision 1.46
diff -u -r1.46 configure.in
--- configure.in	19 Oct 2004 22:11:21 -0000	1.46
+++ configure.in	20 Oct 2004 16:28:24 -0000
@@ -121,8 +121,12 @@
 if test "$enable_gedit" = "no"; then
 	echo "disabling gedit plugin"
 else
-	echo "enabling gedit plugin"
-	PKG_CHECK_MODULES(GEDIT, gedit-2.6 >= 2.6.0)
+	echo "enabling gedit plugin..."
+	echo "checking for supported versions of gedit..."
+	
+	PKG_CHECK_MODULES(GEDIT, gedit-2.8 >= 2.8.0, , 
+		[PKG_CHECK_MODULES(GEDIT, gedit-2.6 >= 2.6.0)])
+	
 	AC_SUBST(GEDIT_CFLAGS)
 fi
 
