--- Makefile.alt.old	2004-07-08 22:44:07.619288072 +0000
+++ Makefile.alt	2004-07-08 22:44:16.283970840 +0000
@@ -36,10 +36,6 @@
 CFLAGS += $(INCLUDES)
 CFLAGS += -DSVGALIB_HELPER_MAJOR=$(SVGALIB_HELPER_MAJOR)
                                                                                                   
-ifeq (1,$(findstring 1,$(MODVER)))
-	CFLAGS += -DMODVERSIONS -include $(INCLUDEDIR)/linux/modversions.h
-endif
-
 TARGET = svgalib_helper
 OBJS = $(TARGET).o
 SRC = main.c interrupt.c i810.c virtual.c displaystart.c
