--- Makefile	2005-05-15 11:59:04.000000000 -0700
+++ Makefile	2005-05-15 11:58:50.000000000 -0700
@@ -349,27 +349,27 @@
 	rm -f _match.s
 
 $(G)zdiff: zdiff.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zdiff.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/zdiff.in > $@
 	chmod 755 $@
 
 $(G)zgrep: zgrep.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zgrep.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/zgrep.in > $@
 	chmod 755 $@
 
 $(G)zmore: zmore.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zmore.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/zmore.in > $@
 	chmod 755 $@
 
 $(G)znew: znew.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/znew.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/znew.in > $@
 	chmod 755 $@
 
 $(G)zforce: zforce.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zforce.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/zforce.in > $@
 	chmod 755 $@
 
 gzexe: gzexe.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/gzexe.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|$(xbindir)|" $(srcdir)/gzexe.in > $@
 	chmod 755 $@
 
 gzip.info: gzip.texi
