--- transfig.3.2.3d/fig2dev/Imakefile	Thu Apr 19 11:19:38 2001
+++ transfig.3.2.3d/fig2dev/Imakefile	Sun Feb 10 23:29:42 2002
@@ -29,5 +29,5 @@
 XCOMM and installed xfig.  This will be made automatic in the future.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	/usr/lib/X11/xfig
 
 XCOMM ******
@@ -105,5 +105,5 @@
 #ifdef USEJPEG
 #ifdef USEINSTALLEDJPEG
-JPEGLIBDIR = /usr/local/lib
+JPEGLIBDIR = /usr/lib
 JPEGINCDIR = /usr/include/X11
 #else
@@ -159,5 +159,5 @@
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = /usr/lib/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
