--- fig2dev/Imakefile.old	2003-01-06 09:39:32.000000000 -0800
+++ fig2dev/Imakefile	2003-01-06 09:40:05.000000000 -0800
@@ -20,7 +20,7 @@
 XCOMM You should point XFIGLIBDIR to the same directory you did when you compiled
 XCOMM and installed xfig.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	/usr/lib/X11/xfig
 
 XCOMM ******
 XCOMM If your system has the strerror() function (doesn't have sys_errlist) then
@@ -126,7 +126,7 @@
 
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = /usr/lib/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
 install::
