--- texk/web2c/fmtutil.in	2004-07-21 15:19:45.009927112 +0200
+++ texk/web2c/fmtutil.in	2004-07-21 15:20:04.349986976 +0200
@@ -92,10 +92,10 @@
 #! pdfemex-pl		pdfetex		mexconf.tex	nls=pdfetex-pl,il2-pl *pdfemex.ini
 
 # - Czech / Slovak:
-#! csplain	tex		-		csplain.ini
-#! cslatex	tex		-	cslatex.ini
-#! pdfcsplain	pdftex		-		csplain.ini
-#! pdfcslatex	pdftex		-    cslatex.ini
+csplain	tex		-		csplain.ini
+cslatex	tex		-	cslatex.ini
+pdfcsplain	pdftex		-		csplain.ini
+pdfcslatex	pdftex		-    cslatex.ini
 
 # Other formats:
 
