--- setup.cfg.cp	2004-09-12 12:01:04.155545736 +0200
+++ setup.cfg	2004-09-12 12:03:57.936127072 +0200
@@ -14,11 +14,11 @@
 # On Windows, setup.py will try to determine whether the database engine
 # against which we're compiling is Firebird 1.0 or later by reading the
 # registry.  To force the database into Firebird mode, uncomment this:
-#database_is_firebird=1
+database_is_firebird=1
 
 # Installation directory of the database.  Unless otherwise specified, include
 # and lib paths are relative to the database_home_dir.
-#database_home_dir=C:/dev/db/firebird151
+database_home_dir=/usr/firebird
 
 #database_include_dir=C:/dev/db/firebird151/include
 # For IB 5.5:
