diff -Naur iproute2-050124.orig/ip/iptunnel.c iproute2-050124/ip/iptunnel.c
--- iproute2-050124.orig/ip/iptunnel.c	2005-01-25 11:15:22.054005392 -0800
+++ iproute2-050124/ip/iptunnel.c	2005-01-25 11:15:48.264020864 -0800
@@ -30,6 +30,9 @@
 #include <linux/if_arp.h>
 #include <linux/ip.h>
 #include <linux/if_tunnel.h>
+#include <endian.h>
+#include <asm/byteorder.h>
+#include <linux/byteorder/swab.h>
 
 #include "rt_names.h"
 #include "utils.h"
