--- ip/Makefile    2004-10-19 13:49:02.000000000 -0700
+++ ip/Makefile    2005-01-13 20:37:22.240049568 -0800
@@ -6,7 +6,8 @@
 RTMONOBJ=rtmon.o
 
 ALLOBJ=$(IPOBJ) $(RTMONOBJ)
-TARGETS=ip rtmon ifcfg rtpr
+TARGETS=ip rtmon
+SCRIPTS=ifcfg rtpr
 
 all: $(TARGETS)
 
@@ -16,6 +17,7 @@
 
 install: all
	install -m 0755 -s $(TARGETS) $(DESTDIR)$(SBINDIR)
+	install -m 0755 $(SCRIPTS) $(DESTDIR)$(SBINDIR)
	install -m 0755 routel routef $(DESTDIR)$(SBINDIR)
 
 clean:

