--- gtk-sharp-2.4.0/vte/glue/Makefile.am.vte	2006-01-09 17:46:04.000000000 +0100
+++ gtk-sharp-2.4.0/vte/glue/Makefile.am	2006-01-09 17:46:34.000000000 +0100
@@ -12,9 +12,9 @@
 
 libvtesharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
 
-libvtesharpglue_2_la_LIBADD = $(VTE_LIBS)
+libvtesharpglue_2_la_LIBADD = $(VTE_LIBS) $(GTK_LIBS)
 
-INCLUDES = $(VTE_DEPENDENCIES_CFLAGS) -I$(top_srcdir)
+INCLUDES = $(VTE_DEPENDENCIES_CFLAGS) $(GTK_CFLAGS) -I$(top_srcdir)
 
 libvtesharpglue.dll: $(libvtesharpglue_2_la_OBJECTS) libvtesharpglue.rc libvtesharpglue.def
 	./build-dll libvtesharpglue-2 $(VERSION)
