diff -urpN grub-0.95.orig/util/grub-install.in grub-0.95/util/grub-install.in
--- grub-0.95.orig/util/grub-install.in	2004-11-09 13:25:38.159282168 -0800
+++ grub-0.95/util/grub-install.in	2004-11-09 13:53:29.027271624 -0800
@@ -331,6 +331,10 @@ else
     # Create a safe temporary file.
     test -n "$mklog" && log_file=`$mklog`
 
+    # Before all invocations of the grub shell, call sync to make sure
+    # the raw device is in sync with any bufferring in filesystems.
+    sync
+ 
     $grub_shell --batch $no_floppy --device-map=$device_map <<EOF >$log_file
 quit
 EOF
@@ -442,6 +446,10 @@ rm -f $log_file
 # Create a safe temporary file.
 test -n "$mklog" && log_file=`$mklog`
 
+# Before all invocations of the grub shell, call sync to make sure
+# the raw device is in sync with any bufferring in filesystems.
+sync
+
 # Now perform the installation.
 $grub_shell --batch $no_floppy --device-map=$device_map <<EOF >$log_file
 root $root_drive
