diff -urpN grub-0.95.orig/stage2/cmdline.c grub-0.95/stage2/cmdline.c
--- grub-0.95.orig/stage2/cmdline.c	2004-11-09 13:25:38.187277912 -0800
+++ grub-0.95/stage2/cmdline.c	2004-11-09 13:48:32.363371392 -0800
@@ -48,12 +48,17 @@ skip_to (int after_equal, char *cmdline)
 
 /* Print a helpful message for the command-line interface.  */
 void
-print_cmdline_message (int forever)
+print_cmdline_message (int type)
 {
   printf (" [ Minimal BASH-like line editing is supported.  For the first word, TAB\n"
 	  "   lists possible command completions.  Anywhere else TAB lists the possible\n"
-	  "   completions of a device/filename.%s ]\n",
-	  (forever ? "" : "  ESC at any time exits."));
+	  "   completions of a device/filename.");
+  if (type == CMDLINE_NORMAL_MODE)
+    printf("  ESC at any time exits.");
+  if (type == CMDLINE_EDIT_MODE)
+    printf("  ESC at any time cancels.  ENTER \n"
+           "   at any time accepts your changes.");
+  printf("]\n");
 }
 
 /* Find the builtin whose command name is COMMAND and return the
@@ -128,7 +133,7 @@ enter_cmdline (char *heap, int forever)
   print_network_configuration ();
   grub_putchar ('\n');
 #endif
-  print_cmdline_message (forever);
+  print_cmdline_message (forever ? CMDLINE_FOREVER_MODE : CMDLINE_NORMAL_MODE);
   
   while (1)
     {
diff -urpN grub-0.95.orig/stage2/shared.h grub-0.95/stage2/shared.h
--- grub-0.95.orig/stage2/shared.h	2004-11-09 13:25:38.188277760 -0800
+++ grub-0.95/stage2/shared.h	2004-11-09 13:48:32.364371240 -0800
@@ -854,9 +854,15 @@ void init_builtins (void);
 void init_config (void);
 char *skip_to (int after_equal, char *cmdline);
 struct builtin *find_command (char *command);
-void print_cmdline_message (int forever);
 void enter_cmdline (char *heap, int forever);
 int run_script (char *script, char *heap);
+
+/* the flags for the cmdline message */
+#define CMDLINE_FOREVER_MODE 0x0
+#define CMDLINE_NORMAL_MODE 0x1
+#define CMDLINE_EDIT_MODE 0x2
+
+void print_cmdline_message (int type);
 #endif
 
 /* C library replacement functions with identical semantics. */
diff -urpN grub-0.95.orig/stage2/stage2.c grub-0.95/stage2/stage2.c
--- grub-0.95.orig/stage2/stage2.c	2004-11-09 13:25:38.189277608 -0800
+++ grub-0.95/stage2/stage2.c	2004-11-09 13:48:32.365371088 -0800
@@ -655,7 +655,7 @@ restart:
 		  else
 		    {
 		      cls ();
-		      print_cmdline_message (0);
+		      print_cmdline_message (CMDLINE_EDIT_MODE);
 
 		      new_heap = heap + NEW_HEAPSIZE + 1;
 
