diff -urpN grub-0.95.orig/lib/device.c grub-0.95/lib/device.c
--- grub-0.95.orig/lib/device.c	2004-11-09 13:25:38.157282472 -0800
+++ grub-0.95/lib/device.c	2004-11-09 13:52:24.824032000 -0800
@@ -858,7 +858,14 @@ write_to_partition (char **map, int driv
       if (strcmp (dev + strlen(dev) - 5, "/disc") == 0)
 	strcpy (dev + strlen(dev) - 5, "/part");
     }
-  sprintf (dev + strlen(dev), "%d", ((partition >> 16) & 0xFF) + 1);
+
+  sprintf (dev + strlen(dev), "%s%d", 
+	   /* Compaq smart and others */
+	   (strncmp(dev, "/dev/ida/", 9) == 0 ||
+            strncmp(dev, "/dev/ataraid/", 13) == 0 ||
+	    strncmp(dev, "/dev/cciss/", 11) == 0 ||
+	    strncmp(dev, "/dev/rd/", 8) == 0) ? "p" : "",
+	   ((partition >> 16) & 0xFF) + 1);
   
   /* Open the partition.  */
   fd = open (dev, O_RDWR);
