Index: grub/stage2/builtins.c
diff -u grub/stage2/builtins.c:1.120 grub/stage2/builtins.c:1.121
--- grub/stage2/builtins.c:1.120	Wed Jan  2 16:56:40 2002
+++ grub/stage2/builtins.c	Mon Feb  4 17:47:51 2002
@@ -949,7 +949,7 @@
 	  else
 	    str = "Reserved";
 	  grub_printf ("   %s:  Base Address:  0x%x X 4GB + 0x%x,\n"
-		       "      Length:   %u X 4GB + %u bytes\n",
+		       "      Length:   0x%x X 4GB + 0x%x bytes\n",
 		       str,
 		       (unsigned long) (map->BaseAddr >> 32),
 		       (unsigned long) (map->BaseAddr & 0xFFFFFFFF),
