--- patch-1.4.sde.libesmtp.3	2003-03-05 17:10:28.000000000 -0600
+++ patch-1.4.sde.libesmtp.3.smgl	2005-03-19 19:16:13.989185856 -0600
@@ -11,13 +11,12 @@ diff -urN --exclude-from=bin/diff.exclud
  
  EXTRA_DIST = COPYRIGHT GPL OPS OPS.PGP TODO configure acconfig.h account.h \
  	attach.h buffy.h charset.h copy.h dotlock.h functions.h gen_defs \
-@@ -79,7 +80,8 @@
- 	_regex.h OPS.MIX README.SECURITY remailer.c remailer.h browser.h \
+@@ -79,7 +80,7 @@
  	mbyte.h	lib.h extlib.c pgpewrap.c pgplib.h Muttrc.head Muttrc \
  	makedoc.c stamp-doc-rc README.SSL \
--	muttbug pgppacket.h depcomp ascii.h BEWARE PATCHES patchlist.sh
-+	muttbug pgppacket.h depcomp ascii.h BEWARE PATCHES patchlist.sh \
-+	mutt_libesmtp.h
+ 	muttbug pgppacket.h depcomp ascii.h BEWARE PATCHES patchlist.sh \
+-	snprintf.c regex.c
++	snprintf.c regex.c mutt_libesmtp.h
  
  mutt_dotlock_SOURCES = mutt_dotlock.c
  mutt_dotlock_LDADD = @LIBOBJS@
