--- patch-1.5.3.sde.libesmtp.3	2003-03-05 17:10:28.000000000 -0600
+++ patch-1.5.3.sde.libesmtp.3.smgl	2005-03-20 17:58:10.781345176 -0600
@@ -2,25 +2,24 @@ diff -urN --exclude-from=bin/diff.exclud
 --- mutt-1.5.3/Makefile.am	Mon Dec  9 13:06:22 2002
 +++ mutt-1.5.3/Makefile.am	Wed Mar  5 11:50:47 2003
 @@ -60,7 +60,8 @@
- EXTRA_mutt_SOURCES = account.c md5c.c mutt_sasl.c mutt_socket.c mutt_ssl.c \
- 	mutt_tunnel.c pop.c pop_auth.c pop_lib.c crypt.c smime.c pgp.c pgpinvoke.c pgpkey.c \
- 	pgplib.c sha1.c pgpmicalg.c gnupgparse.c resize.c dotlock.c remailer.c \
--	browser.h mbyte.h remailer.h url.h mutt_ssl_nss.c pgppacket.c 
-+	browser.h mbyte.h remailer.h url.h mutt_ssl_nss.c pgppacket.c \
+	browser.h mbyte.h remailer.h url.h mutt_ssl_nss.c \
+	crypt-mod-pgp-classic.c crypt-mod-smime-classic.c \
+	pgppacket.c mutt_idna.h hcache.c mutt_ssl_gnutls.c \
+-	crypt-gpgme.c crypt-mod-pgp-gpgme.c crypt-mod-smime-gpgme.c
++	crypt-gpgme.c crypt-mod-pgp-gpgme.c crypt-mod-smime-gpgme.c \
 +	mutt_libesmtp.c
  
  EXTRA_DIST = COPYRIGHT GPL OPS OPS.PGP OPS.CRYPT OPS.SMIME TODO \
  	configure acconfig.h account.h \
-@@ -74,7 +75,8 @@
- 	mbyte.h	lib.h extlib.c pgpewrap.c smime_keys.pl pgplib.h Muttrc.head Muttrc \
+@@ -74,7 +75,7 @@
  	makedoc.c stamp-doc-rc README.SSL crypt.h smime.h\
  	muttbug pgppacket.h depcomp ascii.h BEWARE PATCHES patchlist.sh	\
--	ChangeLog.old mkchangelog.sh cvslog2changelog.pl
-+	ChangeLog.old mkchangelog.sh cvslog2changelog.pl \
-+	mutt_libesmtp.h
+	ChangeLog.old mkchangelog.sh cvslog2changelog.pl mutt_idna.h \
+-	snprintf.c regex.c crypt-gpgme.h
++	snprintf.c regex.c crypt-gpgme.h mutt_libesmtp.h
+
+ EXTRA_SCRIPTS = smime_keys
  
- mutt_dotlock_SOURCES = mutt_dotlock.c
- mutt_dotlock_LDADD = @LIBOBJS@
 diff -urN --exclude-from=bin/diff.excludes mutt-1.5.3/configure.in mutt-1.5.3/configure.in
 --- mutt-1.5.3/configure.in	Tue Dec 17 04:35:21 2002
 +++ mutt-1.5.3/configure.in	Wed Mar  5 11:50:47 2003
@@ -75,8 +74,8 @@ diff -urN --exclude-from=bin/diff.exclud
  struct option_t MuttVars[] = {
 @@ -1654,7 +1657,39 @@
    ** keyid (the hash-value that OpenSSL generates) to work properly
+   ** (S/MIME only)
    */
- #endif /* HAVE_SMIME */
 -  
 +
 +#if defined(USE_LIBESMTP)
@@ -111,9 +110,9 @@ diff -urN --exclude-from=bin/diff.exclud
 +  */
 +#endif
 +
- #if defined(USE_SSL)||defined(USE_NSS)
- # ifndef USE_NSS  
-   { "ssl_starttls", DT_QUAD, R_NONE, OPT_SSLSTARTTLS, M_YES },
+ #if defined(USE_SSL)||defined(USE_NSS)||defined(USE_GNUTLS)
+ #ifdef USE_SSL
+   { "ssl_client_cert", DT_PATH, R_NONE, UL &SslClientCert, 0 },
 diff -urN --exclude-from=bin/diff.excludes mutt-1.5.3/m4/libesmtp.m4 mutt-1.5.3/m4/libesmtp.m4
 --- mutt-1.5.3/m4/libesmtp.m4	Wed Dec 31 18:00:00 1969
 +++ mutt-1.5.3/m4/libesmtp.m4	Wed Mar  5 11:52:36 2003
