--- patch-1.4.dgc.attach.5	2002-06-18 12:32:50.000000000 -0500
+++ patch-1.4.dgc.attach.5.smgl	2005-03-19 19:45:19.903766496 -0600
@@ -1,3 +1,8 @@
+diff -Pur mutt-1.4-base/PATCHES mutt-1.4/PATCHES
+--- mutt-1.4-base/PATCHES	Mon Mar 20 04:25:49 2000
++++ mutt-1.4/PATCHES	Fri Jun 14 11:18:35 2002
+@@ -1,0 +1 @@
++patch-1.4.dgc.attach.5
 diff -Pur mutt-1.4-base/Muttrc.head.in mutt-1.4/Muttrc.head.in
 --- mutt-1.4-base/Muttrc.head.in	Mon Mar 20 04:25:49 2000
 +++ mutt-1.4/Muttrc.head.in	Fri Jun 14 11:18:35 2002
@@ -432,12 +437,13 @@ diff -Pur mutt-1.4-base/init.h mutt-1.4/
    { "lists",		parse_list,		UL &MailLists },
    { "macro",		mutt_parse_macro,	0 },
    { "mailboxes",	mutt_parse_mailboxes,	0 },
-@@ -2493,9 +2512,13 @@
+@@ -2493,10 +2512,14 @@
    { "subscribe",	parse_subscribe,	0 },
    { "toggle",		parse_set,		M_SET_INV },
    { "unalias",		parse_unalias,		0 },
 +  { "unattach-allow",	parse_unattach_list,	UL &AttachAllow },
 +  { "unattach-exclude",	parse_unattach_list,	UL &AttachExclude },
+   { "unauto_view",	parse_unlist,		UL &AutoViewList },
    { "unhdr_order",	parse_unlist,		UL &HeaderOrderList },
    { "unhook",		mutt_parse_unhook,	0 },
    { "unignore",		parse_unignore,		0 },
