diff -Nur gst-plugins-0.8.7/configure gst-plugins-0.8.7-libvisual/configure
--- gst-plugins-0.8.7/configure	2005-01-05 11:46:07.000000000 +0100
+++ gst-plugins-0.8.7-libvisual/configure	2005-02-19 17:06:01.802594097 +0100
@@ -47473,22 +47473,22 @@
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
         echo "$as_me:$LINENO: checking for libvisual = 0.1.7" >&5
-echo $ECHO_N "checking for libvisual = 0.1.7... $ECHO_C" >&6
+echo $ECHO_N "checking for libvisual >= 0.2.0... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "libvisual = 0.1.7" ; then
+        if $PKG_CONFIG --exists "libvisual >= 0.2.0" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking LIBVISUAL_CFLAGS" >&5
 echo $ECHO_N "checking LIBVISUAL_CFLAGS... $ECHO_C" >&6
-            LIBVISUAL_CFLAGS=`$PKG_CONFIG --cflags "libvisual = 0.1.7"`
+            LIBVISUAL_CFLAGS=`$PKG_CONFIG --cflags "libvisual >= 0.2.0"`
             echo "$as_me:$LINENO: result: $LIBVISUAL_CFLAGS" >&5
 echo "${ECHO_T}$LIBVISUAL_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking LIBVISUAL_LIBS" >&5
 echo $ECHO_N "checking LIBVISUAL_LIBS... $ECHO_C" >&6
-            LIBVISUAL_LIBS=`$PKG_CONFIG --libs "libvisual = 0.1.7"`
+            LIBVISUAL_LIBS=`$PKG_CONFIG --libs "libvisual >= 0.2.0"`
             echo "$as_me:$LINENO: result: $LIBVISUAL_LIBS" >&5
 echo "${ECHO_T}$LIBVISUAL_LIBS" >&6
         else
@@ -47496,7 +47496,7 @@
             LIBVISUAL_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            LIBVISUAL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libvisual = 0.1.7"`
+            LIBVISUAL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libvisual >= 0.2.0"`
 
         fi
 
diff -Nur gst-plugins-0.8.7/ext/libvisual/visual.c gst-plugins-0.8.7-libvisual/ext/libvisual/visual.c
--- gst-plugins-0.8.7/ext/libvisual/visual.c	2004-11-12 19:00:38.000000000 +0100
+++ gst-plugins-0.8.7-libvisual/ext/libvisual/visual.c	2005-02-19 17:07:02.508856752 +0100
@@ -282,7 +282,7 @@
     return GST_PAD_LINK_REFUSED;
 
   if (visual->video)
-    visual_video_free (visual->video);
+    visual_object_unref (VISUAL_OBJECT (visual->video));
   visual->video = visual_video_new ();
   visual_actor_set_video (visual->actor, visual->video);
   visual_video_set_depth (visual->video,
@@ -352,7 +352,7 @@
         return GST_STATE_FAILURE;
 
       if (visual_actor_realize (visual->actor) != 0) {
-        visual_actor_free (visual->actor);
+        visual_object_unref (VISUAL_OBJECT (visual->actor));
         visual->actor = NULL;
         return GST_STATE_FAILURE;
       }
@@ -368,7 +368,7 @@
     case GST_STATE_PAUSED_TO_READY:
       break;
     case GST_STATE_READY_TO_NULL:
-      visual_actor_free (visual->actor);
+      visual_object_unref (VISUAL_OBJECT (visual->actor));
       visual->actor = NULL;
       break;
     default:
