diff -urpN chora-1.2/config/conf.php.dist chora/config/conf.php.dist
--- chora-1.2/config/conf.php.dist	2002-12-16 17:46:30.000000000 -0800
+++ chora/config/conf.php.dist	2003-08-04 10:57:19.000000000 -0700
@@ -9,7 +9,7 @@
  */
 
 // Make sure that constants are defined.
-require_once dirname(__FILE__) . '/../lib/constants.php';
+require_once HORDE_BASE . '/chora/lib/constants.php';
 
 /**
  ** Paths and Locations
