diff -r -U2 ntop/Makefile.in ntop/Makefile.in
--- ntop/Makefile.in	Thu Dec 27 17:00:25 2001
+++ ntop/Makefile.in	Mon Feb  4 17:19:25 2002
@@ -432,19 +432,8 @@
 install-man1:
 	$(mkinstalldirs) $(DESTDIR)$(man1dir)
-	@list='$(man1_MANS)'; \
-	l2='$(man_MANS)'; for i in $$l2; do \
-	  case "$$i" in \
-	    *.1*) list="$$list $$i" ;; \
-	  esac; \
-	done; \
-	for i in $$list; do \
-	  if test -f $(srcdir)/$$i; then file=$(srcdir)/$$i; \
-	  else file=$$i; fi; \
-	  ext=`echo $$i | sed -e 's/^.*\\.//'`; \
-	  inst=`echo $$i | sed -e 's/\\.[0-9a-z]*$$//'`; \
-	  inst=`echo $$inst | sed '$(transform)'`.$$ext; \
+	file="intop/intop.1"; \
+	inst="intop.1"; \
 	  echo " $(INSTALL_DATA) $$file $(DESTDIR)$(man1dir)/$$inst"; \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(man1dir)/$$inst; \
-	done
+	  $(INSTALL_DATA) $$file $(DESTDIR)$(man1dir)/$$inst; 
 
 uninstall-man1:
@@ -863,5 +852,5 @@
 
 install-data-local:
-	@cp -p ntop-cert.pem $(bindir)/ntop-cert.pem
+	@cp -p ntop-cert.pem /etc/ssl/certs/ntop-cert.pem
 	@$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(libdir)/plugins;
 	@$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/ntop;
diff -r -U2 ntop/configure ntop/configure
--- ntop/configure	Thu Dec 27 18:40:54 2001
+++ ntop/configure	Mon Feb  4 15:20:06 2002
@@ -5374,9 +5374,5 @@
 
 
-CONFIGFILE_DIR=$sysconfdir/ntop
-CONFIGFILE_DIR=`(
-    test "x$prefix" = xNONE && prefix=$ac_default_prefix
-    eval echo "$CONFIGFILE_DIR"
-)`
+CONFIGFILE_DIR=/etc/ssl/certs
 cat >> confdefs.h <<EOF
 #define CONFIGFILE_DIR "$CONFIGFILE_DIR"
