diff -ur wvstreams-3.75.0.orig/include/uniconftree.h wvstreams-3.75.0/include/uniconftree.h
--- wvstreams-3.75.0.orig/include/uniconftree.h	2004-03-13 18:59:36.000000000 +0000
+++ wvstreams-3.75.0/include/uniconftree.h	2004-08-22 19:32:46.000000000 +0100
@@ -48,7 +48,7 @@
 
     /** Returns a pointer to the parent node, or NULL if there is none. */
     Sub *parent() const
-        { return static_cast<Sub*>(xparent); }
+        { return static_cast<Sub*>(Base::xparent); }
 
     /** Reparents this node. */
     void setparent(Sub *parent)
@@ -93,13 +93,13 @@
     /** Removes and deletes all children of this node. */
     void zap()
     {
-        if (!xchildren)
+        if (!Base::xchildren)
             return;
         // set xchildren to NULL first so that the zap() will happen faster
         // otherwise, each child will attempt to unlink itself uselessly
 
-        typename Base::Container *oldchildren = xchildren;
-        xchildren = NULL;
+        typename Base::Container *oldchildren = Base::xchildren;
+        Base::xchildren = NULL;
 
         // delete all children
         typename Base::Container::Iter i(*oldchildren);
diff -ur wvstreams-3.75.0.orig/include/wvsorter.h wvstreams-3.75.0/include/wvsorter.h
--- wvstreams-3.75.0.orig/include/wvsorter.h	2004-03-13 18:59:36.000000000 +0000
+++ wvstreams-3.75.0/include/wvsorter.h	2004-08-22 19:24:38.000000000 +0100
@@ -90,7 +90,7 @@
     for (i.rewind(); i.next(); )
 	n++;
     
-    array = new (void *) [n+2];
+    array = new void* [n+2];
     void **aptr = array;
 
     *aptr++ = NULL; // initial link is NULL, to act like a normal iterator
diff -ur wvstreams-3.75.0.orig/utils/wvbuffer.cc wvstreams-3.75.0/utils/wvbuffer.cc
--- wvstreams-3.75.0.orig/utils/wvbuffer.cc	2004-03-13 18:59:36.000000000 +0000
+++ wvstreams-3.75.0/utils/wvbuffer.cc	2004-08-22 19:39:11.000000000 +0100
@@ -8,15 +8,6 @@
 
 /***** Specialization for raw memory buffers *****/
 
-// Instantiate some commonly used templates
-template WvBufBaseCommonImpl<unsigned char>;
-template WvInPlaceBufBase<unsigned char>;
-template WvConstInPlaceBufBase<unsigned char>;
-template WvCircularBufBase<unsigned char>;
-template WvDynBufBase<unsigned char>;
-template WvNullBufBase<unsigned char>;
-template WvBufCursorBase<unsigned char>;
-
 void WvBufBase<unsigned char>::putstr(WvStringParm str)
 {
     put((const unsigned char*)str.cstr(), str.len());
