--- configure.in	2003-12-22 17:57:17.000000000 +0100
+++ configure.in.cp	2003-12-22 18:10:31.000000000 +0100
@@ -15,7 +15,7 @@
 
 CPPFLAGS=-I/usr/X11R6/include
 #LDFLAGS="-L/usr/X11R6/lib -lpbm -lpgm -lppm -lpnm"
-LDFLAGS="-L/usr/X11R6/lib -L/usr/local/netpbm/lib"
+LDFLAGS="-L/usr/X11R6/lib -L/usr/lib -lnetpbm"
 
 dnl Checks for libraries.
 ## pnm can only be found, if LDFLAGS contains -lpgm -lpnm ... etc. 
