Index: src/nsMsgCompose.cpp
===================================================================
RCS file: /cvsroot/mozilla/mailnews/compose/src/nsMsgCompose.cpp,v
retrieving revision 1.370
diff -u -2 -w -r1.370 nsMsgCompose.cpp
--- src/nsMsgCompose.cpp	3 Dec 2002 19:53:53 -0000	1.370
+++ src/nsMsgCompose.cpp	10 Dec 2002 20:18:03 -0000
@@ -548,20 +548,4 @@
         PRInt32 end;
         nsAutoString bodyAttributes;
-        nsAutoString headContent;
-
-        /* Before we can insert the data into editor, we need to remove the <head> tag
-           content and then insert it back using ReplaceHeadContentsWithHTML */
-        start = aBuf.Find("<head>", PR_TRUE);
-        if (start != kNotFound)
-        {
-          start += 6; // move pass the <head> tag
-          end = aBuf.Find("</head>", PR_TRUE, start);
-          if (end != kNotFound)
-          {
-            const PRUnichar* data = aBuf.get();
-            headContent.Adopt(nsCRT::strndup(&data[start], end - start));
-            aBuf.Cut(start, end - start);
-          }
-        }
 
         /* If we have attribute for the body tag, we need to save them in order
@@ -579,6 +563,4 @@
         }
 
-        if (!headContent.IsEmpty())
-          htmlEditor->ReplaceHeadContentsWithHTML(headContent);
         htmlEditor->InsertHTML(aBuf);
 

