diff -urpN mozilla.orig/xpfe/bootstrap/Makefile.in mozilla/xpfe/bootstrap/Makefile.in
--- mozilla.orig/xpfe/bootstrap/Makefile.in	2004-04-10 15:54:43.000000000 -0700
+++ mozilla/xpfe/bootstrap/Makefile.in	2004-12-09 17:22:00.000000000 -0800
@@ -361,7 +361,7 @@ endif
 
 CXXFLAGS	+= $(MOZ_TOOLKIT_REGISTRY_CFLAGS)
 
-LOCAL_INCLUDES	= -I$(srcdir) -I.
+LOCAL_INCLUDES	= -I$(srcdir) -I. -I$(DIST)/sdk/include
 
 ifneq (,$(filter-out OS2 WINNT,$(OS_ARCH)))
 $(MOZ_APP_NAME).1: mozilla.man.in Makefile.in Makefile $(DEPTH)/config/autoconf.mk
