--- xpfe/bootstrap/Makefile.in	2003-04-18 16:54:02.000000000 -0400
+++ xpfe/bootstrap/Makefile.in	2003-06-15 00:52:07.000000000 -0400
@@ -340,7 +340,7 @@
 
 CXXFLAGS	+= $(MOZ_TOOLKIT_REGISTRY_CFLAGS)
 
-LOCAL_INCLUDES	= -I$(srcdir) -I.
+LOCAL_INCLUDES	= -I$(srcdir) -I. -I$(DIST)/sdk/include
 
 ifneq (,$(filter-out OS2 WINNT,$(OS_ARCH)))
 mozilla.1: mozilla.man.in Makefile.in Makefile $(DEPTH)/config/autoconf.mk
