--- src/Makefile.old	2003-12-03 17:54:49.000000000 -0800
+++ src/Makefile	2003-12-03 17:57:50.000000000 -0800
@@ -18,16 +18,16 @@
 srcdir = .
 top_srcdir = ..
 
-prefix = /usr
+prefix = /usr/local
 exec_prefix = ${prefix}
 
 bindir = ${exec_prefix}/bin
 sbindir = ${exec_prefix}/sbin
 libexecdir = ${exec_prefix}/libexec
 datadir = ${prefix}/share
-sysconfdir = /etc
+sysconfdir = ${prefix}/etc
 sharedstatedir = ${prefix}/com
-localstatedir = /var
+localstatedir = ${prefix}/var
 libdir = ${exec_prefix}/lib
 infodir = ${prefix}/info
 mandir = ${prefix}/man
@@ -96,9 +96,7 @@
 
 sbin_PROGRAMS = prelink
 
-prelink_SYSDEPS = arch-i386.c arch-alpha.c arch-ppc.c arch-ppc64.c \
-		  arch-sparc.c arch-sparc64.c arch-x86_64.c \
-		  arch-s390.c arch-s390x.c arch-arm.c arch-sh.c arch-ia64.c
+prelink_SYSDEPS = arch-i386.c
 
 prelink_SOURCES = cache.c conflict.c cxx.c data.c doit.c dso.c dwarf2.c	     \
 		  dwarf2.h exec.c execle_open.c fptr.c fptr.h get.c gather.c \
@@ -117,11 +115,7 @@
 sbin_PROGRAMS = prelink$(EXEEXT)
 PROGRAMS = $(sbin_PROGRAMS)
 
-am__objects_1 = arch-i386.$(OBJEXT) arch-alpha.$(OBJEXT) \
-	arch-ppc.$(OBJEXT) arch-ppc64.$(OBJEXT) arch-sparc.$(OBJEXT) \
-	arch-sparc64.$(OBJEXT) arch-x86_64.$(OBJEXT) \
-	arch-s390.$(OBJEXT) arch-s390x.$(OBJEXT) arch-arm.$(OBJEXT) \
-	arch-sh.$(OBJEXT) arch-ia64.$(OBJEXT)
+am__objects_1 = arch-i386.$(OBJEXT)
 am_prelink_OBJECTS = cache.$(OBJEXT) conflict.$(OBJEXT) cxx.$(OBJEXT) \
 	data.$(OBJEXT) doit.$(OBJEXT) dso.$(OBJEXT) dwarf2.$(OBJEXT) \
 	exec.$(OBJEXT) execle_open.$(OBJEXT) fptr.$(OBJEXT) \
@@ -134,18 +128,12 @@
 prelink_DEPENDENCIES =
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
 CPPFLAGS = 
-LDFLAGS =  -z combreloc -s
+LDFLAGS = 
 LIBS = -lelf 
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
-DEP_FILES = ./$(DEPDIR)/arch-alpha.Po \
-	./$(DEPDIR)/arch-arm.Po ./$(DEPDIR)/arch-i386.Po \
-	./$(DEPDIR)/arch-ia64.Po ./$(DEPDIR)/arch-ppc.Po \
-	./$(DEPDIR)/arch-ppc64.Po ./$(DEPDIR)/arch-s390.Po \
-	./$(DEPDIR)/arch-s390x.Po ./$(DEPDIR)/arch-sh.Po \
-	./$(DEPDIR)/arch-sparc.Po \
-	./$(DEPDIR)/arch-sparc64.Po \
-	./$(DEPDIR)/arch-x86_64.Po ./$(DEPDIR)/cache.Po \
+DEP_FILES = ./$(DEPDIR)/arch-i386.Po \
+	./$(DEPDIR)/cache.Po \
 	./$(DEPDIR)/conflict.Po ./$(DEPDIR)/crc32.Po \
 	./$(DEPDIR)/cxx.Po ./$(DEPDIR)/data.Po \
 	./$(DEPDIR)/doit.Po ./$(DEPDIR)/dso.Po \
@@ -165,7 +153,7 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-CFLAGS = -march=athlon-xp -mmmx -mfpmath=sse -msse -m3dnow -O3
+CFLAGS = -g -O2
 DIST_SOURCES = $(prelink_SOURCES)
 DIST_COMMON = Makefile.am Makefile.in
 SOURCES = $(prelink_SOURCES)
@@ -218,18 +206,7 @@
 distclean-compile:
 	-rm -f *.tab.c
 
-include ./$(DEPDIR)/arch-alpha.Po
-include ./$(DEPDIR)/arch-arm.Po
 include ./$(DEPDIR)/arch-i386.Po
-include ./$(DEPDIR)/arch-ia64.Po
-include ./$(DEPDIR)/arch-ppc.Po
-include ./$(DEPDIR)/arch-ppc64.Po
-include ./$(DEPDIR)/arch-s390.Po
-include ./$(DEPDIR)/arch-s390x.Po
-include ./$(DEPDIR)/arch-sh.Po
-include ./$(DEPDIR)/arch-sparc.Po
-include ./$(DEPDIR)/arch-sparc64.Po
-include ./$(DEPDIR)/arch-x86_64.Po
 include ./$(DEPDIR)/cache.Po
 include ./$(DEPDIR)/conflict.Po
 include ./$(DEPDIR)/crc32.Po
