diff -purN db-4.1.25/env/env_region.c db/env/env_region.c
--- db-4.1.25/env/env_region.c	2002-07-31 15:19:40.000000000 -0400
+++ db/env/env_region.c	2003-01-03 16:00:16.000000000 -0500
@@ -130,7 +130,7 @@ loop:	renv = NULL;
 	 */
 	if (F_ISSET(dbenv, DB_ENV_CREATE)) {
 		if ((ret = __os_open(dbenv, infop->name,
-		    DB_OSO_CREATE | DB_OSO_DIRECT | DB_OSO_EXCL | DB_OSO_REGION,
+		    DB_OSO_CREATE | DB_OSO_EXCL | DB_OSO_REGION,
 		    dbenv->db_mode, dbenv->lockfhp)) == 0)
 			goto creation;
 		if (ret != EEXIST) {
@@ -144,7 +144,7 @@ loop:	renv = NULL;
 	 * If we couldn't create the file, try and open it.  (If that fails,
 	 * we're done.)
 	 */
-	if ((ret = __os_open(dbenv, infop->name, DB_OSO_REGION | DB_OSO_DIRECT,
+	if ((ret = __os_open(dbenv, infop->name, DB_OSO_REGION,
 	    dbenv->db_mode, dbenv->lockfhp)) != 0)
 		goto err;
 
diff -purN db-4.1.25/mp/mp_fopen.c db/mp/mp_fopen.c
--- db-4.1.25/mp/mp_fopen.c	2002-12-19 11:39:08.000000000 -0500
+++ db/mp/mp_fopen.c	2003-01-03 16:21:16.000000000 -0500
@@ -247,7 +247,7 @@ __memp_fopen(dbmfp, path, flags, mode, p
 
 	/* Validate arguments. */
 	if ((ret = __db_fchk(dbenv, "memp_fopen", flags,
-	    DB_CREATE | DB_DIRECT | DB_EXTENT |
+	    DB_CREATE | DB_EXTENT |
 	    DB_NOMMAP | DB_ODDFILESIZE | DB_RDONLY | DB_TRUNCATE)) != 0)
 		return (ret);
 
diff -purN db-4.1.25/os/os_map.c db/os/os_map.c
--- db-4.1.25/os/os_map.c	2002-08-22 17:58:05.000000000 -0400
+++ db/os/os_map.c	2003-01-03 16:24:45.000000000 -0500
@@ -146,7 +146,7 @@ __os_r_sysattach(dbenv, infop, rp)
 	 * of that.
 	 */
 	if ((ret = __os_open(dbenv, infop->name,
-	    DB_OSO_REGION | DB_OSO_DIRECT |
+	    DB_OSO_REGION |
 	    (F_ISSET(infop, REGION_CREATE_OK) ? DB_OSO_CREATE : 0),
 	    infop->mode, &fh)) != 0)
 		__db_err(dbenv, "%s: %s", infop->name, db_strerror(ret));
diff -purN db-4.1.25/os_win32/os_map.c db/os_win32/os_map.c
--- db-4.1.25/os_win32/os_map.c	2002-09-13 07:56:35.000000000 -0400
+++ db/os_win32/os_map.c	2003-01-03 16:23:28.000000000 -0500
@@ -36,7 +36,6 @@ __os_r_sysattach(dbenv, infop, rp)
 	 * properly ordered, our caller has already taken care of that.
 	 */
 	if ((ret = __os_open(dbenv, infop->name,
-	    DB_OSO_DIRECT |
 	    F_ISSET(infop, REGION_CREATE_OK) ? DB_OSO_CREATE: 0,
 	    infop->mode, &fh)) != 0) {
 		__db_err(dbenv, "%s: %s", infop->name, db_strerror(ret));
