diff -Naur gtk-sharp-1.0.6.orig/configure.in gtk-sharp-1.0.6/configure.in
--- gtk-sharp-1.0.6.orig/configure.in	2005-03-29 10:49:55.206617008 -0800
+++ gtk-sharp-1.0.6/configure.in	2005-03-29 11:16:23.316187656 -0800
@@ -153,11 +153,11 @@
 RSVG_REQUIRED_VERSION=2.0.1
 PKG_CHECK_MODULES(RSVG_DEPENDENCIES, librsvg-2.0 >= $RSVG_REQUIRED_VERSION, enable_rsvg=yes, enable_rsvg=no)
 
-GTKHTML_REQUIRED_VERSION=3.0.10
-PKG_CHECK_MODULES(GTKHTML_DEPENDENCIES, libgtkhtml-3.0 >= $GTKHTML_REQUIRED_VERSION, enable_gtkhtml=yes, enable_gtkhtml=no)
+GTKHTML_REQUIRED_VERSION=3.6.1
+PKG_CHECK_MODULES(GTKHTML_DEPENDENCIES, libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION, enable_gtkhtml=yes, enable_gtkhtml=no)
 if test "x$enable_gtkhtml" = "xyes"; then
-	GTKHTML_VERSION=3.0
-	GTKHTML_SOVERSION=4
+	GTKHTML_VERSION=3.6
+	GTKHTML_SOVERSION=18
 else
 	GTKHTML_REQUIRED_VERSION=3.2.0
 	PKG_CHECK_MODULES(GTKHTML_DEPENDENCIES, libgtkhtml-3.1 >= $GTKHTML_REQUIRED_VERSION, enable_gtkhtml=yes, enable_gtkhtml=no)
