diff -u -r1.7 -r1.8
--- libetpan/libetpan/src/engine/mailprivacy_gnupg.c	2005/11/30 11:08:43	1.7
+++ libetpan/libetpan/src/engine/mailprivacy_gnupg.c	2005/12/07 14:57:57	1.8
@@ -30,7 +30,7 @@
  */
 
 /*
- * $Id: mailprivacy_gnupg.c,v 1.7 2005/11/30 11:08:43 hoa Exp $
+ * $Id: mailprivacy_gnupg.c,v 1.8 2005/12/07 14:57:57 hoa Exp $
  */
 
 /* passphrase is needed when private key is needed
@@ -509,7 +509,7 @@
   }
   
   snprintf(command, sizeof(command),
-      "gpg --passphrase-fd=0 --batch --yes --decrypt %s",
+      "gpg --passphrase-fd=0 --batch --yes --decrypt '%s'",
       quoted_encrypted_filename);
   
   decrypt_ok = 0;
@@ -706,7 +706,7 @@
     goto unlink_decrypted;
   }
   
-  snprintf(command, sizeof(command), "gpg --batch --yes --verify %s %s",
+  snprintf(command, sizeof(command), "gpg --batch --yes --verify '%s' '%s'",
       quoted_signature_filename, quoted_signed_filename);
 
   sign_ok = 0;
@@ -877,7 +877,7 @@
   }
   
   snprintf(command, sizeof(command),
-      "gpg --batch --yes --decrypt %s", quoted_signed_filename);
+      "gpg --batch --yes --decrypt '%s'", quoted_signed_filename);
   
   sign_ok = 0;
   r = gpg_command_passphrase(privacy, msg, command, NULL,
@@ -1099,7 +1099,7 @@
   }
   
   snprintf(command, sizeof(command),
-      "gpg --passphrase-fd=0 --batch --yes --decrypt %s",
+      "gpg --passphrase-fd=0 --batch --yes --decrypt '%s'",
       quoted_encrypted_filename);
   
   sign_ok = 0;
@@ -1524,7 +1524,7 @@
   }
   
   snprintf(command, sizeof(command),
-      "gpg --passphrase-fd=0 -a --batch --yes --digest-algo sha1 %s -b %s",
+      "gpg --passphrase-fd=0 -a --batch --yes --digest-algo sha1 %s -b '%s'",
       default_key, quoted_to_sign_filename);
   
   sign_ok = 0;
@@ -1905,7 +1905,7 @@
   }
   
   snprintf(command, sizeof(command),
-      "gpg --passphrase-fd=0 %s -a --batch --yes --digest-algo sha1 -s %s -e %s",
+      "gpg --passphrase-fd=0 %s -a --batch --yes --digest-algo sha1 -s %s -e '%s'",
       recipient, default_key, quoted_original_filename);
   
   encrypt_ok = 0;
@@ -2126,7 +2126,7 @@
     goto unlink_encrypted;
   }
   
-  snprintf(command, sizeof(command), "gpg %s -a --batch --yes -e %s",
+  snprintf(command, sizeof(command), "gpg %s -a --batch --yes -e '%s'",
       recipient, quoted_original_filename);
   
   encrypt_ok = 0;
@@ -2339,7 +2339,7 @@
   }
 
   snprintf(command, sizeof(command),
-      "gpg --passphrase-fd=0 --batch --yes --digest-algo sha1 %s --clearsign %s",
+      "gpg --passphrase-fd=0 --batch --yes --digest-algo sha1 %s --clearsign '%s'",
       default_key, quoted_original_filename);
   
   sign_ok = 0;
@@ -2518,7 +2518,7 @@
     goto unlink_description;
   }
   
-  snprintf(command, sizeof(command), "gpg %s --batch --yes -e --armor %s",
+  snprintf(command, sizeof(command), "gpg %s --batch --yes -e --armor '%s'",
       recipient, quoted_original_filename);
   
   encrypt_ok = 0;
@@ -2711,7 +2711,7 @@
 
   snprintf(command, sizeof(command),
       "gpg --passphrase-fd=0 %s --batch --yes --digest-algo sha1 "
-      "%s -e -s -a %s",
+      "%s -e -s -a '%s'",
       recipient, default_key, quoted_original_filename);
 
   encrypt_ok = 0;
diff -u -r1.11 -r1.12
--- libetpan/libetpan/src/engine/mailprivacy_smime.c	2005/11/30 11:08:43	1.11
+++ libetpan/libetpan/src/engine/mailprivacy_smime.c	2005/12/07 14:57:57	1.12
@@ -30,7 +30,7 @@
  */
 
 /*
- * $Id: mailprivacy_smime.c,v 1.11 2005/11/30 11:08:43 hoa Exp $
+ * $Id: mailprivacy_smime.c,v 1.12 2005/12/07 14:57:57 hoa Exp $
  */
 
 #include "mailprivacy_smime.h"
@@ -295,8 +295,8 @@
     }
     
     sign_ok = 0;
-    snprintf(command, PATH_MAX,
-        "openssl smime -decrypt -passin fd:0 -in %s -inkey %s -recip %s",
+    snprintf(command, sizeof(command),
+        "openssl smime -decrypt -passin fd:0 -in '%s' -inkey '%s' -recip '%s'",
         quoted_smime_filename, quoted_smime_key, quoted_smime_cert);
     
     unlink(description_filename);
@@ -449,7 +449,7 @@
       goto err;
     }
     
-    snprintf(check_CA, sizeof(check_CA), "-CAfile %s", quoted_CAfile);
+    snprintf(check_CA, sizeof(check_CA), "-CAfile '%s'", quoted_CAfile);
   }
   
   * noverify = '\0';
@@ -493,7 +493,7 @@
   }
 
   sign_ok = 0;
-  snprintf(command, PATH_MAX, "openssl smime -verify -in %s %s %s",
+  snprintf(command, sizeof(command), "openssl smime -verify -in '%s' %s %s",
       quoted_smime_filename, check_CA, noverify);
   
   r = smime_command_passphrase(privacy, msg, command,
@@ -777,7 +777,7 @@
   }
 
   snprintf(command, sizeof(command),
-      "openssl smime -sign -passin fd:0 -in %s -signer %s -inkey %s",
+      "openssl smime -sign -passin fd:0 -in '%s' -signer '%s' -inkey '%s'",
       quoted_signed_filename,
       quoted_smime_cert, quoted_smime_key);
 
@@ -854,9 +854,13 @@
   buflen = strlen(quoted_filename + 1);
   if (buflen >= * len)
     return MAIL_ERROR_MEMORY;
-    
+  
+  strncat(recipient, "\'", * len);
+  (* len) --;
   strncat(recipient, quoted_filename, * len);
   (* len) -= buflen;
+  strncat(recipient, "\'", * len);
+  (* len) --;
   strncat(recipient, " ", * len);
   (* len) --;
   
@@ -1059,7 +1063,7 @@
     goto unlink_description;
   }
   
-  snprintf(command, sizeof(command), "openssl smime -encrypt -in %s %s",
+  snprintf(command, sizeof(command), "openssl smime -encrypt -in '%s' %s",
       quoted_decrypted_filename, recipient);
   
   r = smime_command_passphrase(privacy, msg, command,
@@ -1367,16 +1371,20 @@
     return;
   
   while ((ent = readdir(dir)) != NULL) {
+    char quoted_filename[PATH_MAX];
     char filename[PATH_MAX];
     char command[PATH_MAX];
     char buf[MAX_EMAIL_SIZE];
     FILE * p;
+    int r;
     
     snprintf(filename, sizeof(filename),
         "%s/%s", directory, ent->d_name);
     
+    r = mail_quote_filename(quoted_filename, sizeof(quoted_filename), filename);
+    
     snprintf(command, sizeof(command),
-        "openssl x509 -email -noout -in %s 2>/dev/null", filename);
+        "openssl x509 -email -noout -in '%s' 2>/dev/null", quoted_filename);
     
     p = popen(command, "r");
     if (p == NULL)
@@ -1560,7 +1568,7 @@
   }
 
   snprintf(command, sizeof(command),
-      "openssl pkcs7 -inform DER -in %s -out %s -print_certs 2>/dev/null",
+      "openssl pkcs7 -inform DER -in '%s' -out '%s' -print_certs 2>/dev/null",
       quoted_signature_filename, quoted_store_cert_filename);
   
   r = system(command);
diff -u -r1.7 -r1.8
--- libetpan/libetpan/src/engine/mailprivacy_tools.c	2005/12/16 02:15:31	1.7
+++ libetpan/libetpan/src/engine/mailprivacy_tools.c	2005/12/21 23:49:23	1.8
@@ -380,7 +380,7 @@
   privacy can be set to NULL to disable privacy check.
 */
 
-static int mailprivacy_get_mime(struct mailprivacy * privacy,
+int mailprivacy_get_mime(struct mailprivacy * privacy,
     int check_privacy, int reencode,
     char * content, size_t content_len,
     struct mailmime ** result_mime)
diff -u -r1.5 -r1.6
--- libetpan/libetpan/src/engine/mailprivacy_tools.h	2005/11/30 11:08:43	1.5
+++ libetpan/libetpan/src/engine/mailprivacy_tools.h	2005/12/21 23:49:23	1.6
@@ -30,7 +30,7 @@
  */
 
 /*
- * $Id: mailprivacy_tools.h,v 1.5 2005/11/30 11:08:43 hoa Exp $
+ * $Id: mailprivacy_tools.h,v 1.6 2005/12/21 23:49:23 hoa Exp $
  */
 
 #ifndef MAIL_PRIVACY_TOOLS_H
@@ -97,4 +97,9 @@
     char * filename, size_t size,
     mailmessage * msg, struct mailmime * mime);
 
+int mailprivacy_get_mime(struct mailprivacy * privacy,
+    int check_privacy, int reencode,
+    char * content, size_t content_len,
+    struct mailmime ** result_mime);
+
 #endif
diff -u -r1.51 -r1.52
--- libetpan/libetpan/configure.in	2005/12/02 10:15:22	1.51
+++ libetpan/libetpan/configure.in	2005/12/04 13:29:41	1.52
@@ -1,5 +1,5 @@
 dnl Process this file with autoconf to create configure
-dnl $Id: configure.in,v 1.51 2005/12/02 10:15:22 hoa Exp $
+dnl $Id: configure.in,v 1.52 2005/12/04 13:29:41 hoa Exp $
 
 AC_INIT(src/main/libetpan_version.h.in)
 
@@ -23,7 +23,7 @@
 dnl    interface.
 API_CURRENT=6
 API_REVISION=0
-API_COMPATIBILITY=0
+API_COMPATIBILITY=6
 
 API_AGE=`expr $API_CURRENT - $API_COMPATIBILITY`
 API_VERSION="$API_CURRENT:$API_REVISION:$API_AGE"

