diff -r -U2 stegdetect.ori/file/Makefile.am stegdetect/file/Makefile.am
--- stegdetect.ori/file/Makefile.am	2001-10-22 19:38:36.000000000 +0200
+++ stegdetect/file/Makefile.am	2004-05-25 18:18:30.513719096 +0200
@@ -40,5 +40,5 @@
 magic.inc: magic
 	echo -n "\"" > $@
-	sed -e 's/\\/\\\\/g' -e 's/"/\\\"/g' magic >> $@
+	sed -e 's/\\/\\\\/g' -e 's/"/\\\"/g' -e 's/$$/\ \\/' magic >> $@
 	echo -n "\"" >> $@
 
diff -r -U2 stegdetect.ori/file/Makefile.in stegdetect/file/Makefile.in
--- stegdetect.ori/file/Makefile.in	2001-10-22 19:38:36.000000000 +0200
+++ stegdetect/file/Makefile.in	2004-05-25 18:17:46.788366360 +0200
@@ -506,5 +506,5 @@
 magic.inc: magic
 	echo -n "\"" > $@
-	sed -e 's/\\/\\\\/g' -e 's/"/\\\"/g' magic >> $@
+	sed -e 's/\\/\\\\/g' -e 's/"/\\\"/g' -e 's/$$/\ \\/' magic >> $@
 	echo -n "\"" >> $@
 
