diff -urp nload-0.5.0/src/form_field.cpp nload-0.5.0.fixed/src/form_field.cpp
--- nload-0.5.0/src/form_field.cpp	2002-08-01 19:21:31.000000000 +0200
+++ nload-0.5.0.fixed/src/form_field.cpp	2003-01-21 23:19:17.000000000 +0100
@@ -133,7 +133,7 @@ bool operator==( const Field& field1, co
 }
 
 
-Form::Form( Slots* slots = 0 ) : m_slots( slots ), m_form(0), m_visible( false )
+Form::Form( Slots* slots ) : m_slots( slots ), m_form(0), m_visible( false )
 {
 	m_instances.push_back( this );
 }
