diff -Nur kaffe-1.0.6-org/libraries/clib/math/BigInteger.c kaffe-1.0.6/libraries/clib/math/BigInteger.c
--- kaffe-1.0.6-org/libraries/clib/math/BigInteger.c	Sat Jul 22 00:41:26 2000
+++ kaffe-1.0.6/libraries/clib/math/BigInteger.c	Fri Mar 22 03:46:00 2002
@@ -508,7 +508,7 @@
 
 	src = (mpz_srcptr)(*env)->GetObjectField(env, s, number);
 
-	return ((jint)mpz_get_si(src));
+	return ((jint)mpz_get_ui(src));
 }
 
 jint
