diff -Nur krb5-1.2.8.ori/src/include/kerberosIV/krb.h krb5-1.2.8/src/include/kerberosIV/krb.h
--- krb5-1.2.8.ori/src/include/kerberosIV/krb.h	2000-06-03 00:05:20.000000000 +0200
+++ krb5-1.2.8/src/include/kerberosIV/krb.h	2003-06-10 18:28:34.000000000 +0200
@@ -26,6 +26,8 @@
  * Include file for the Kerberos V4 library. 
  */
 
+#include <errno.h>
+
 /* Only one time, please */
 #ifndef	KRB_DEFS
 #define KRB_DEFS
diff -Nur krb5-1.2.8.ori/src/util/ss/ss.h krb5-1.2.8/src/util/ss/ss.h
--- krb5-1.2.8.ori/src/util/ss/ss.h	1998-03-30 19:10:33.000000000 +0200
+++ krb5-1.2.8/src/util/ss/ss.h	2003-06-10 18:44:53.000000000 +0200
@@ -4,6 +4,8 @@
  * For copyright information, see mit-sipb-copyright.h.
  */
 
+#include <errno.h>
+
 #ifndef _ss_h
 #define _ss_h __FILE__
 
