--- sysdeps/unix/sysv/linux/sys/kd.h.orig	2005-12-17 00:19:18.000000000 -0600
+++ sysdeps/unix/sysv/linux/sys/kd.h		2005-12-17 00:20:43.000000000 -0600
@@ -21,9 +21,15 @@
 
 /* Make sure the <linux/types.h> header is not loaded.  */
 #ifndef _LINUX_TYPES_H
-# define _LINUX_TYPES_H	1
+#define _LINUX_TYPES_H 1
+#define __undef_LINUX_TYPES_H 1
 #endif
 
 #include <linux/kd.h>
 
+#ifdef __undef_LINUX_TYPES_H
+#undef _LINUX_TYPES_H
+#undef __undef_LINUX_TYPES_H
+#endif
+
 #endif	/* sys/kd.h */
