diff -Nur linux/videodev2.h /usr/include/linux/videodev2.h
--- linux/videodev2.h	2003-11-26 21:44:42.000000000 +0100
+++ /usr/include/linux/videodev2.h	2003-12-16 08:44:09.000000000 +0100
@@ -13,7 +13,9 @@
  *		Justin Schoeman
  *		et al.
  */
+#ifndef _STRUCT_TIMEVAL
 #include <linux/time.h> /* need struct timeval */
+#endif
 
 /*
  *	M I S C E L L A N E O U S
diff -Nur linux/videodev.h /usr/include/linux/videodev.h
--- linux/videodev.h	2003-11-26 21:46:13.000000000 +0100
+++ /usr/include/linux/videodev.h	2003-12-16 08:41:05.000000000 +0100
@@ -2,14 +2,14 @@
 #define __LINUX_VIDEODEV_H
 
 #include <linux/types.h>
-#include <linux/version.h>
-#include <linux/device.h>
 
 #define HAVE_V4L2 1
 #include <linux/videodev2.h>
 
 #ifdef __KERNEL__
 
+#include <linux/version.h>
+#include <linux/device.h>
 #include <linux/poll.h>
 #include <linux/mm.h>
 
