diff -ur digikamplugins/printwizard/frmprintwizardbase.ui digikamplugins-mine/printwizard/frmprintwizardbase.ui
--- digikamplugins/printwizard/frmprintwizardbase.ui	2003-11-08 03:48:32.000000000 +0100
+++ digikamplugins-mine/printwizard/frmprintwizardbase.ui	2004-05-04 17:07:20.000000000 +0200
@@ -279,6 +279,11 @@
                                         <string>A4</string>
                                     </property>
                                 </item>
+                                <item>
+                                    <property name="text">
+                                        <string>10x15cm</string>
+                                    </property>
+                                </item>
                                 <property name="name">
                                     <cstring>CmbPaperSize</cstring>
                                 </property>
diff -ur digikamplugins/printwizard/frmprintwizard.cpp digikamplugins-mine/printwizard/frmprintwizard.cpp
--- digikamplugins/printwizard/frmprintwizard.cpp	2004-02-19 21:56:14.000000000 +0100
+++ digikamplugins-mine/printwizard/frmprintwizard.cpp	2004-05-05 09:27:31.000000000 +0200
@@ -524,7 +524,9 @@
 
   int pageSize = config.readNumEntry("PageSize", (int)m_pageSize);
   initPhotoSizes((KPrinter::PageSize)pageSize);
-  if (m_pageSize == KPrinter::A4)
+  if (m_pageSize == KPrinter::A6)
+    CmbPaperSize->setCurrentItem(2);
+  else if (m_pageSize == KPrinter::A4)
     CmbPaperSize->setCurrentItem(1);
   else
     CmbPaperSize->setCurrentItem(0);
@@ -625,6 +627,8 @@
              break;
     case 1 : pageSize = KPrinter::A4;
              break;
+    case 2 : pageSize = KPrinter::A6;
+             break;
   }
   initPhotoSizes(pageSize);
 }
@@ -778,6 +782,37 @@
     // add to the list
     m_photoSizes.append(p);
   } // A4
+  else if (pageSize == KPrinter::A6)
+  {
+    // A6 is 10.5 x 14.8 cm
+    // but let's pretend it is 10.2 x 15.3 cm
+    TPhotoSize *p;
+    // ========== 9x13
+    p = new TPhotoSize;
+    p->label = i18n("9 x 13cm");
+    // page size
+//    p->layouts.append(new QRect(0, 0, 1050, 1480));
+    p->layouts.append(new QRect(0, 0, 1020, 1530));
+
+    // photo layouts
+    p->layouts.append(new QRect( 50,  100, 900, 1300));
+    
+    // add to the list
+    m_photoSizes.append(p);
+
+    // ========== 10x15cm
+    p = new TPhotoSize;
+    p->label = i18n("10 x 15cm");
+    // page size
+//     p->layouts.append(new QRect(0, 0, 1050, 1480));
+    p->layouts.append(new QRect(0, 0, 1020, 1530));
+
+    // photo layouts
+    p->layouts.append(new QRect( 0, 0, 1020, 1530));
+    
+    // add to the list
+    m_photoSizes.append(p);
+ }
   else
   {
     kdDebug() << "Initializing Unsupported page layouts\n";
