--- openwebmail-tool.pl.org	2003-04-26 22:44:09.000000000 +0200
+++ openwebmail-tool.pl	2003-04-26 22:46:02.000000000 +0200
@@ -229,8 +229,8 @@
    $err++ if (check_dbm_option($dbm_ext, $dbmopen_ext, $dbmopen_haslock, %config)<0);
    $err++ if (check_savedsuid_support()<0);
    if ($err>0) {
-      print qq|And execute '$SCRIPT_DIR/openwebmail-tool.pl --init' again!.\n\n|;
-      return -1;
+#      print qq|And execute '$SCRIPT_DIR/openwebmail-tool.pl --init' again!.\n\n|;
+#      return -1;
    }
 
    if ($config{'b2g_map'} ne 'none' && !-f "$config{'ow_etcdir'}/b2g$config{'dbm_ext'}") {
