diff -Nurp middleman.orig/include/proto.h middleman/include/proto.h
--- middleman.orig/include/proto.h	2003-06-08 02:00:52.000000000 +0200
+++ middleman/include/proto.h	2003-06-08 02:03:43.000000000 +0200
@@ -3,8 +3,8 @@
 
 #include "flags.h"
 #include "../config.h"
-#include "../pcre/pcre.h"
-#include "../pcre/pcreposix.h"
+#include "pcre.h"
+#include "pcreposix.h"
 #include <pthread.h>
 
 #ifdef HAVE_POLL
diff -Nurp middleman.orig/Makefile.in middleman/Makefile.in
--- middleman.orig/Makefile.in	2003-06-08 02:00:52.000000000 +0200
+++ middleman/Makefile.in	2003-06-08 02:03:43.000000000 +0200
@@ -1,12 +1,11 @@
 CC = @CC@
 CFLAGS = -Iinclude -I. @CFLAGS@ -Wall -D_GNU_SOURCE
 LDFLAGS = @LDFLAGS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lpcre -lpcreposix
 DEFS = @DEFS@ -D_REENTRANT
 MMAN_OBJS = src/global.o src/cache.o src/access.o src/filter.o src/log.o src/main.o src/network.o src/socket.o src/header.o src/template.o src/cookies.o src/xml.o src/misc.o src/mem.o src/signal.o src/protocol.o src/compat.o src/hash.o src/rewrite.o src/external.o src/mime.o src/regexp.o src/interface.o src/redirect.o src/forward.o src/keywords.o src/limits.o src/dirlist.o src/pool.o src/timers.o src/base64.o
-PCRE_OBJS = pcre/get.o pcre/study.o pcre/chartables.o pcre/pcre.o pcre/pcreposix.o
 LIBNTLM_OBJS = libntlm/smbdes.o libntlm/smbencrypt.o libntlm/smbmd4.o libntlm/smbutil.o libntlm/version.o
-OBJS = $(MMAN_OBJS) $(PCRE_OBJS) $(LIBNTLM_OBJS)
+OBJS = $(MMAN_OBJS) $(LIBNTLM_OBJS)
 TARGET= mman
 STRIP = @STRIP@
 ld = ld
@@ -14,12 +13,6 @@ ld = ld
 
 all: $(TARGET)
 
-pcre/dftables:
-	$(CC) -o pcre/dftables $(CFLAGS) pcre/dftables.c
-	
-pcre/chartables.c:	pcre/dftables
-	pcre/dftables > pcre/chartables.c
-
 %.o: %.c
 	$(CC) -c $(CFLAGS) $(DEFS) $< -o $@
 
@@ -27,8 +20,11 @@ $(TARGET): $(OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
 	$(STRIP) $(TARGET)
 
+install: $(TARGET)
+	install -m 755 mman /usr/sbin
+
 clean:
-	rm -f *~ $(OBJS) core config.log config.status $(TARGET) pcre/dftables pcre/chartables.c
+	rm -f *~ $(OBJS) core config.log config.status $(TARGET)
 
 distclean: clean
 	rm config.h Makefile
diff -Nurp middleman.orig/src/regexp.c middleman/src/regexp.c
--- middleman.orig/src/regexp.c	2003-06-08 02:00:52.000000000 +0200
+++ middleman/src/regexp.c	2003-06-08 02:03:43.000000000 +0200
@@ -30,7 +30,7 @@ regex_t *reg_compile(char *pattern, int 
 
 	re = xmalloc(sizeof(regex_t));
 
-	rerr = regcomp2(re, pattern, flags);
+	rerr = regcomp(re, pattern, flags);
 	if (rerr != 0) {
 		putlog(MMLOG_ERROR, "failed to compile regular expression: %s", pattern);
 		xfree(re);
@@ -47,7 +47,7 @@ void reg_free(regex_t * re)
 {
 	ASSERT(re != NULL);
 
-	regfree2(re);
+	regfree(re);
 	xfree(re);
 }
 
@@ -58,7 +58,7 @@ int reg_exec(regex_t * re, char *string)
 {
 	int rerr;
 
-	rerr = regexec2(re, string, 0, NULL, 0);
+	rerr = regexec(re, string, 0, NULL, 0);
 
 	return rerr;
 }
