diff  -U  configure  configure
--- configure	Thu Feb 21 00:46:33 2002
+++ configure	Thu Feb 21 00:46:33 2002
@@ -13834,7 +13834,7 @@
   #find Mozilla headers
 echo "$as_me:13835: checking for Mozilla development headers" >&5
 echo $ECHO_N "checking for Mozilla development headers... $ECHO_C" >&6
-  mozilla_incldirs="/usr/include /usr/include/mozilla /usr/local/include /opt/include"
+  mozilla_incldirs="/usr/include /usr/include/mozilla /usr/include/mozilla/gtkembedmoz /usr/local/include /opt/include"
 
 mozilla_incdir=NO
 for i in $mozilla_incldirs;
