--- adriver.h	2003-10-11 10:43:03.000000000 +0200
+++ adriver.h.copy	2003-10-11 11:04:29.000000000 +0200
@@ -127,6 +127,9 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 3) && !defined(need_resched)
 #define need_resched() (current->need_resched)
 #endif
+#if LINUX_VERSION_CODE == KERNEL_VERSION(2, 4, 23)
+#define CONFIG_HAVE_PDE
+#endif
 #ifndef CONFIG_HAVE_PDE
 #include <linux/fs.h>
 static inline struct proc_dir_entry *PDE(const struct inode *inode)
