diff -Naur vim61.orig/src/auto/configure vim61/src/auto/configure
--- vim61.orig/src/auto/configure	2002-03-09 13:14:07.000000000 -0500
+++ vim61/src/auto/configure	2002-05-26 20:36:16.000000000 -0400
@@ -1161,12 +1161,6 @@
     LDFLAGS="$LDFLAGS -L/usr/local/lib"
   fi
 fi
-if test -d /usr/local/include; then
-  tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
-  if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-  fi
-fi
 
 echo $ac_n "checking --with-vim-name argument""... $ac_c" 1>&6
 echo "configure:1173: checking --with-vim-name argument" >&5
