diff -urN webmin-0.970/setup.sh webmin/setup.sh
--- webmin-0.970/setup.sh	2002-05-07 01:08:45.000000000 -0400
+++ webmin/setup.sh	2002-05-28 10:25:19.000000000 -0400
@@ -46,9 +46,11 @@
 echo "Unless you want to run multiple versions of Webmin at the same time"
 echo "you can just accept the defaults."
 echo ""
-printf "Config file directory [/etc/webmin]: "
+printf "Config file directory [${config_dir:-/etc/webmin}]: "
 if [ "$config_dir" = "" ]; then
 	read config_dir
+else
+	echo ""
 fi
 if [ "$config_dir" = "" ]; then
 	config_dir=/etc/webmin
@@ -133,9 +135,11 @@
 	fi
 
 	# Ask for log directory
-	printf "Log file directory [/var/webmin]: "
+	printf "Log file directory [${var_dir:-/var/webmin}]: "
 	if [ "$var_dir" = "" ]; then
 		read var_dir
+	else
+		echo ""
 	fi
 	if [ "$var_dir" = "" ]; then
 		var_dir=/var/webmin
@@ -183,7 +187,7 @@
 				exit 4
 			fi
 		else
-			printf "Full path to perl (default $perldef): "
+			printf "Full path to perl (default ${perl:-$perldef}): "
 			read perl
 			if [ "$perl" = "" ]; then
 				perl=$perldef
@@ -256,12 +260,14 @@
 	echo " - If the webserver should use SSL (if your system supports it)."
 	echo " - Whether to start webmin at boot time."
 	echo ""
-	printf "Web server port (default 10000): "
+	printf "Web server port (default ${port:-10000}): "
 	if [ "$port" = "" ]; then
 		read port
 		if [ "$port" = "" ]; then
 			port=10000
 		fi
+	else
+		echo ""
 	fi
 	if [ $port -lt 1 ]; then
 		echo "ERROR: $port is not a valid port number"
@@ -280,12 +286,14 @@
 		echo ""
 		exit 13
 	fi
-	printf "Login name (default admin): "
+	printf "Login name (default ${login:-admin}): "
 	if [ "$login" = "" ]; then
 		read login
 		if [ "$login" = "" ]; then
 			login="admin"
 		fi
+	else
+		echo ""
 	fi
 	echo $login | grep : >/dev/null
 	if [ "$?" = "0" ]; then
@@ -293,7 +301,7 @@
 		echo ""
 		exit 14
 	fi
-	printf "Login password: "
+	printf "Login password [$password]: "
 	if [ "$password" = "" -a "$crypt" = "" ]; then
 		stty -echo
 		read password
@@ -315,14 +323,18 @@
 			echo ""
 			exit 14
 		fi
+	else
+		echo ""
 	fi
 	defhost=`hostname`
-	printf "Web server host name (default $defhost): "
+	printf "Web server host name (default ${host:-$defhost}): "
 	if [ "$host" = "" ]; then
 		read host
 		if [ "$host" = "" ]; then
 			host=$defhost
 		fi
+	else
+		echo ""
 	fi
 	if [ "$ssl" = "" ]; then
 		ssl=0
@@ -368,7 +380,7 @@
 	echo "addtype_cgi=internal/cgi" >> $cfile
 	echo "realm=Webmin Server" >> $cfile
 	echo "logfile=$var_dir/miniserv.log" >> $cfile
-	echo "pidfile=$var_dir/miniserv.pid" >> $cfile
+	echo "pidfile=/var/run/miniserv.pid" >> $cfile
 	echo "logtime=168" >> $cfile
 	echo "ppath=$ppath" >> $cfile
 	echo "ssl=$ssl" >> $cfile
