--- electricsheep-2.3b11/Makefile.am~	2002-12-12 03:32:51.000000000 -0600
+++ electricsheep-2.3b11/Makefile.am	2003-02-17 17:37:48.000000000 -0600
@@ -10,4 +10,4 @@
 man1_MANS = electricsheep.man
 EXTRA_DIST = $(man1_MANS) $(data_DATA) electricsheep.xml electricsheep.spec timers.c.patch
 install-data-local:
-	${INSTALL} electricsheep.xml /usr/X11R6/lib/xscreensaver/config
+	${INSTALL} electricsheep.xml /usr/lib/xscreensaver/config
