Index: libmpcodecs/Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/Makefile,v
retrieving revision 1.108
diff -u -r1.108 Makefile
--- libmpcodecs/Makefile	4 Oct 2003 22:00:24 -0000	1.108
+++ libmpcodecs/Makefile	5 Oct 2003 23:31:39 -0000
@@ -37,7 +37,7 @@
 SRCS2=$(ENCODER_SRCS)
 OBJS2=$(SRCS2:.c=.o)
 
-CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE
+CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE -I/usr/include/ffmpeg
 ifneq ($(MPFLAC),none)
 CFLAGS += -I../libmpflac
 endif
Index: libmpcodecs/vf_pp.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/vf_pp.c,v
retrieving revision 1.30
diff -u -r1.30 vf_pp.c
--- libmpcodecs/vf_pp.c	31 Aug 2003 22:18:27 -0000	1.30
+++ libmpcodecs/vf_pp.c	5 Oct 2003 23:31:40 -0000
@@ -20,10 +20,10 @@
 
 #define EMU_OLD
 
-#include "../libavcodec/libpostproc/postprocess.h"
+#include "postproc/postprocess.h"
 
 #ifdef EMU_OLD
-#include "../libavcodec/libpostproc/postprocess_internal.h"
+#include "postproc/postprocess_internal.h"
 #endif
 
 struct vf_priv_s {
