Index: configure
===================================================================
RCS file: /cvsroot/mplayer/main/configure,v
retrieving revision 1.792
diff -u -r1.792 configure
--- configure	22 Oct 2003 18:24:57 -0000	1.792
+++ configure	22 Oct 2003 23:47:01 -0000
@@ -4829,6 +4829,8 @@
 elif test "$_libavcodecso" = yes ; then
   _def_libavcodec='#define USE_LIBAVCODEC 1'
   _def_libavcodecso='#define USE_LIBAVCODEC_SO 1'
+  _def_ffpostprocess='#define FF_POSTPROCESS 1'
+  _def_haveffpostprocess='yes'
   _ld_libavcodec='-lavcodec'
   _codecmodules="libavcodec.so $_codecmodules"
 else
Index: libmpcodecs/Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/Makefile,v
retrieving revision 1.110
diff -u -r1.110 Makefile
--- libmpcodecs/Makefile	22 Oct 2003 21:21:52 -0000	1.110
+++ libmpcodecs/Makefile	22 Oct 2003 23:47:05 -0000
@@ -41,7 +41,7 @@
 SRCS2=$(ENCODER_SRCS)
 OBJS2=$(SRCS2:.c=.o)
 
-CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE
+CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE -I/usr/include/ffmpeg
 ifneq ($(MPFLAC),none)
 CFLAGS += -I../libmpflac
 endif
Index: libmpcodecs/vf_pp.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/vf_pp.c,v
retrieving revision 1.30
diff -u -r1.30 vf_pp.c
--- libmpcodecs/vf_pp.c	31 Aug 2003 22:18:27 -0000	1.30
+++ libmpcodecs/vf_pp.c	22 Oct 2003 23:47:06 -0000
@@ -20,10 +20,10 @@
 
 #define EMU_OLD
 
-#include "../libavcodec/libpostproc/postprocess.h"
+#include "postproc/postprocess.h"
 
 #ifdef EMU_OLD
-#include "../libavcodec/libpostproc/postprocess_internal.h"
+#include "postproc/postprocess_internal.h"
 #endif
 
 struct vf_priv_s {
