Index: configure
===================================================================
RCS file: /cvsroot/mplayer/main/configure,v
retrieving revision 1.797
diff -u -r1.797 configure
--- configure	27 Oct 2003 01:06:35 -0000	1.797
+++ configure	27 Oct 2003 11:39:11 -0000
@@ -4832,6 +4832,8 @@
 elif test "$_libavcodecso" = yes ; then
   _def_libavcodec='#define USE_LIBAVCODEC 1'
   _def_libavcodecso='#define USE_LIBAVCODEC_SO 1'
+  _def_ffpostprocess='#define FF_POSTPROCESS 1'
+  _def_haveffpostprocess='yes'
   _ld_libavcodec='-lavcodec'
   _codecmodules="libavcodec.so $_codecmodules"
 else
Index: libmpcodecs/Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/Makefile,v
retrieving revision 1.112
diff -u -r1.112 Makefile
--- libmpcodecs/Makefile	26 Oct 2003 23:07:21 -0000	1.112
+++ libmpcodecs/Makefile	27 Oct 2003 11:39:15 -0000
@@ -41,7 +41,7 @@
 SRCS2=$(ENCODER_SRCS)
 OBJS2=$(SRCS2:.c=.o)
 
-CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE
+CFLAGS  = $(OPTFLAGS) -I. -Inative -I.. -I../libmpdemux -I../loader $(EXTRA_INC) -D_GNU_SOURCE -I/usr/include/ffmpeg
 ifneq ($(MPFLAC),none)
 CFLAGS += -I../libmpflac
 endif
Index: libmpcodecs/vf_pp.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/vf_pp.c,v
retrieving revision 1.30
diff -u -r1.30 vf_pp.c
--- libmpcodecs/vf_pp.c	31 Aug 2003 22:18:27 -0000	1.30
+++ libmpcodecs/vf_pp.c	27 Oct 2003 11:39:16 -0000
@@ -20,10 +20,10 @@
 
 #define EMU_OLD
 
-#include "../libavcodec/libpostproc/postprocess.h"
+#include "postproc/postprocess.h"
 
 #ifdef EMU_OLD
-#include "../libavcodec/libpostproc/postprocess_internal.h"
+#include "postproc/postprocess_internal.h"
 #endif
 
 struct vf_priv_s {
Index: libmpcodecs/vf_spp.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/vf_spp.c,v
retrieving revision 1.3
diff -u -r1.3 vf_spp.c
--- libmpcodecs/vf_spp.c	27 Oct 2003 00:14:33 -0000	1.3
+++ libmpcodecs/vf_spp.c	27 Oct 2003 11:39:16 -0000
@@ -32,8 +32,8 @@
 #include "../config.h"
 #include "../mp_msg.h"
 #include "../cpudetect.h"
-#include "../libavcodec/avcodec.h"
-#include "../libavcodec/dsputil.h"
+#include "avcodec.h"
+#include "dsputil.h"
 
 #ifdef HAVE_MALLOC_H
 #include <malloc.h>
