Index: configure
===================================================================
RCS file: /cvsroot/mplayer/main/configure,v
retrieving revision 1.818
diff -u -r1.818 configure
--- configure	7 Dec 2003 15:26:12 -0000	1.818
+++ configure	7 Dec 2003 16:22:08 -0000
@@ -4928,6 +4928,8 @@
 elif test "$_libavcodecso" = yes ; then
   _def_libavcodec='#define USE_LIBAVCODEC 1'
   _def_libavcodecso='#define USE_LIBAVCODEC_SO 1'
+  _def_ffpostprocess='#define FF_POSTPROCESS 1'
+  _def_haveffpostprocess='yes'
   _ld_libavcodec='-lavcodec'
   _codecmodules="libavcodec.so $_codecmodules"
 else
Index: libmpcodecs/vf_pp.c
===================================================================
RCS file: /cvsroot/mplayer/main/libmpcodecs/vf_pp.c,v
retrieving revision 1.30
diff -u -r1.30 vf_pp.c
--- libmpcodecs/vf_pp.c	31 Aug 2003 22:18:27 -0000	1.30
+++ libmpcodecs/vf_pp.c	7 Dec 2003 16:22:12 -0000
@@ -20,10 +20,10 @@
 
 #define EMU_OLD
 
-#include "../libavcodec/libpostproc/postprocess.h"
+#include "postproc/postprocess.h"
 
 #ifdef EMU_OLD
-#include "../libavcodec/libpostproc/postprocess_internal.h"
+#include "postproc/postprocess_internal.h"
 #endif
 
 struct vf_priv_s {
