diff -Nur cinelerra-1.1.7/build/Makefile.cinelerra cinelerra-1.1.7-shared/build/Makefile.cinelerra
--- cinelerra-1.1.7/build/Makefile.cinelerra	2003-08-09 22:23:19.000000000 +0200
+++ cinelerra-1.1.7-shared/build/Makefile.cinelerra	2003-09-02 13:22:14.000000000 +0200
@@ -19,12 +19,10 @@
 DIRS := \
 	$(AVIFILE_DIR) \
 	doc \
-	libmpeg3 \
 	libsndfile \
 	mpeg2enc \
 	mplexhi \
 	mplexlo \
-	quicktime \
 	guicast \
 	cinelerra \
 	plugins
@@ -40,10 +38,7 @@
 
 all: $(OBJDIR)/soundtest
 	make -f build/Makefile.libavc
-	make -f build/Makefile.libraw
 	make -f build/Makefile.esound
-	make -f build/Makefile.freetype
-	make -f build/Makefile.tiff
 	make -f build/Makefile.toolame
 	make -f build/Makefile.uuid
 ifeq ($(HAVE_ALSA), yes)
@@ -82,7 +77,6 @@
 install:
 	make -C cinelerra install
 	make -C plugins install
-	make -C libmpeg3 install
 	make -C mplexhi install
 	make -C mplexlo install
 	cp $(OBJDIR)/soundtest /usr/bin
diff -Nur cinelerra-1.1.7/cinelerra/Makefile cinelerra-1.1.7-shared/cinelerra/Makefile
--- cinelerra-1.1.7/cinelerra/Makefile	2003-08-09 22:24:28.000000000 +0200
+++ cinelerra-1.1.7-shared/cinelerra/Makefile	2003-09-02 13:23:02.000000000 +0200
@@ -276,11 +276,8 @@
 
 STATICLIBS = \
 	../libavc/$(OBJDIR)/libavc.a \
-	../quicktime/$(OBJDIR)/libquicktime.a \
         $(ESOUNDLIBS) \
-	$(TIFF_DIR)/$(OBJDIR)/libtiff.a \
 	../guicast/$(OBJDIR)/libguicast.a \
-	../libmpeg3/$(OBJDIR)/libmpeg3.a \
 	../libsndfile/$(OBJDIR)/libsndfile.a \
 	$(UUID_DIR)/$(OBJDIR)/libuuid.a \
 	../mpeg2enc/$(OBJDIR)/mpeg2enc.a \
@@ -302,7 +299,6 @@
 
 LIBS += \
 	$(ALSA_LIBS) \
-	$(LIBRAW_DIR)/$(OBJDIR)/libraw1394.a \
 	$(GLLIBS) \
 	-lX11 \
 	-lXext \
@@ -312,7 +308,11 @@
 	$(XXF86VM) \
 	-lpng \
 	-lz \
-	-ldl
+	-ldl \
+	-lmpeg3 \
+	-lquicktime \
+	-ltiff \
+	-lvorbisfile
 
 
 CFLAGS += \
diff -Nur cinelerra-1.1.7/esound/esd.h cinelerra-1.1.7-shared/esound/esd.h
--- cinelerra-1.1.7/esound/esd.h	2002-06-21 14:35:25.000000000 +0200
+++ cinelerra-1.1.7-shared/esound/esd.h	2003-09-02 13:29:01.000000000 +0200
@@ -8,7 +8,7 @@
 
 /* path and name of the default EsounD domain socket */
 #define ESD_UNIX_SOCKET_DIR	"/tmp/.esd"
-#define ESD_UNIX_SOCKET_NAME	ESD_UNIX_SOCKET_DIR ## "/" ## "socket"
+#define ESD_UNIX_SOCKET_NAME	ESD_UNIX_SOCKET_DIR"/socket"
 
 /* length of the audio buffer size */
 #define ESD_BUF_SIZE (4 * 1024)
diff -Nur cinelerra-1.1.7/mpeg2enc/Makefile cinelerra-1.1.7-shared/mpeg2enc/Makefile
--- cinelerra-1.1.7/mpeg2enc/Makefile	2003-07-28 06:38:11.000000000 +0200
+++ cinelerra-1.1.7-shared/mpeg2enc/Makefile	2003-09-02 13:24:55.000000000 +0200
@@ -1,6 +1,6 @@
 include ../global_config
 CC = gcc
-CFLAGS += -I../libmpeg3 -I../quicktime
+CFLAGS += -I/usr/include/libmpeg3 -I/usr/include/quicktime
 
 
 ifeq ($(OBJDIR), i686)
@@ -48,8 +48,8 @@
 	$(OBJDIR)/writepic.o
 
 LIBS = \
-	../../quicktime/$(OBJDIR)/libquicktime.a \
-	../../libmpeg3/$(OBJDIR)/libmpeg3.a \
+        -lquicktime \
+	-lmpeg3 \
 	-lpthread \
 	-lpng \
 	-lz \
diff -Nur cinelerra-1.1.7/mplexlo/Makefile cinelerra-1.1.7-shared/mplexlo/Makefile
--- cinelerra-1.1.7/mplexlo/Makefile	2002-07-12 15:51:49.000000000 +0200
+++ cinelerra-1.1.7-shared/mplexlo/Makefile	2003-09-02 13:23:53.000000000 +0200
@@ -7,7 +7,7 @@
 OUTPUT := $(OBJDIR)/mplexlo
 
 $(OUTPUT): $(OBJS)
-	$(CC) -o $(OUTPUT) $(OBJS) ../libmpeg3/$(OBJDIR)/libmpeg3.a -lpthread -lm
+	$(CC) -o $(OUTPUT) $(OBJS) -lmpeg3 -lpthread -lm
 
 $(OBJS):
 	$(CC) -c $(CFLAGS) $(subst $(OBJDIR)/,, $*.c) -o $*.o
diff -Nur cinelerra-1.1.7/plugins/titler/Makefile cinelerra-1.1.7-shared/plugins/titler/Makefile
--- cinelerra-1.1.7/plugins/titler/Makefile	2003-07-20 11:05:04.000000000 +0200
+++ cinelerra-1.1.7-shared/plugins/titler/Makefile	2003-09-02 13:25:57.000000000 +0200
@@ -15,7 +15,7 @@
 $(OUTPUT_PLUGIN): $(OBJS)
 	$(LDLINKER) -o $(OUTPUT_PLUGIN) \
 		$(OBJS) \
-		$(FREETYPE_DIR)/$(OBJDIR)/libfreetype.a
+		-lfreetype
 	strip $(OUTPUT_PLUGIN)
 
 include ../../plugin_config
