diff -purN horde-2.2.4/config/registry.php.dist horde/config/registry.php.dist
--- horde-2.2.4/config/registry.php.dist	2003-05-06 08:28:01.000000000 -0400
+++ horde/config/registry.php.dist	2003-09-10 15:01:22.000000000 -0400
@@ -83,14 +83,14 @@ $this->registry['filter']['whitelistFrom
  */
 
 $this->applications['horde'] = array(
-    'fileroot' => dirname(__FILE__) . '/..',
+    'fileroot' => HORDE_BASE,
     'webroot' => '/horde',
     'initial_page' => 'login.php',
     'icon' => '/horde/graphics/home.gif',
     'name' => _("Horde"),
     'allow_guests' => true,
     'status' => 'active',
-    'templates' => dirname(__FILE__) . '/../templates',
+    'templates' => HORDE_BASE . '/templates',
     'cookie_domain' => $_SERVER['SERVER_NAME'],
     // ** If IE will be used to access Horde modules, you should read
     //    this discussion about the cookie_path setting (discussing issues
@@ -102,7 +102,7 @@ $this->applications['horde'] = array(
 );
 
 $this->applications['logout'] = array(
-    'fileroot' => dirname(__FILE__) . '/..',
+    'fileroot' => HORDE_BASE,
     'webroot' => $this->applications['horde']['webroot'],
     'initial_page' => 'login.php?reason=logout',
     'icon' => $this->applications['horde']['webroot'] . '/graphics/logout.gif',
@@ -112,7 +112,7 @@ $this->applications['logout'] = array(
 );
 
 $this->applications['imp'] = array(
-    'fileroot' => dirname(__FILE__) . '/../imp',
+    'fileroot' => HORDE_BASE . '/imp',
     'webroot' => $this->applications['horde']['webroot'] . '/imp',
     'icon' => $this->applications['horde']['webroot'] . '/imp/graphics/imp.gif',
     'name' => _("Mail"),
@@ -121,7 +121,7 @@ $this->applications['imp'] = array(
 );
 
 $this->applications['ingo'] = array(
-    'fileroot' => dirname(__FILE__) . '/../ingo',
+    'fileroot' => HORDE_BASE . '/ingo',
     'webroot' => $this->applications['horde']['webroot'] . '/ingo',
     'icon' => $this->applications['horde']['webroot'] . '/ingo/graphics/ingo.gif',
     'name' => _("Filters"),
@@ -130,7 +130,7 @@ $this->applications['ingo'] = array(
 );
 
 $this->applications['turba'] = array(
-    'fileroot' => dirname(__FILE__) . '/../turba',
+    'fileroot' => HORDE_BASE . '/turba',
     'webroot' => $this->applications['horde']['webroot'] . '/turba',
     'icon' => $this->applications['horde']['webroot'] . '/turba/graphics/turba.gif',
     'name' => _("Address Book"),
@@ -139,7 +139,7 @@ $this->applications['turba'] = array(
 );
 
 $this->applications['kronolith'] = array(
-    'fileroot' => dirname(__FILE__) . '/../kronolith',
+    'fileroot' => HORDE_BASE . '/kronolith',
     'webroot' => $this->applications['horde']['webroot'] . '/kronolith',
     'icon' => $this->applications['horde']['webroot'] . '/kronolith/graphics/kronolith.gif',
     'name' => _("Calendar"),
@@ -148,7 +148,7 @@ $this->applications['kronolith'] = array
 );
 
 $this->applications['mnemo'] = array(
-    'fileroot' => dirname(__FILE__) . '/../mnemo',
+    'fileroot' => HORDE_BASE . '/mnemo',
     'webroot' => $this->applications['horde']['webroot'] . '/mnemo',
     'icon' => $this->applications['horde']['webroot'] . '/mnemo/graphics/mnemo.gif',
     'name' => _("Memos"),
@@ -157,7 +157,7 @@ $this->applications['mnemo'] = array(
 );
 
 $this->applications['nag'] = array(
-    'fileroot' => dirname(__FILE__) . '/../nag',
+    'fileroot' => HORDE_BASE . '/nag',
     'webroot' => $this->applications['horde']['webroot'] . '/nag',
     'icon' => $this->applications['horde']['webroot'] . '/nag/graphics/nag.gif',
     'name' => _("Tasks"),
@@ -166,7 +166,7 @@ $this->applications['nag'] = array(
 );
 
 $this->applications['chora'] = array(
-    'fileroot' => dirname(__FILE__) . '/../chora',
+    'fileroot' => HORDE_BASE . '/chora',
     'webroot' => $this->applications['horde']['webroot'] . '/chora',
     'icon' => $this->applications['horde']['webroot'] . '/chora/graphics/chora.gif',
     'name' => _("CVS"),
@@ -175,7 +175,7 @@ $this->applications['chora'] = array(
 );
 
 $this->applications['klutz'] = array(
-    'fileroot' => dirname(__FILE__) . '/../klutz',
+    'fileroot' => HORDE_BASE . '/klutz',
     'webroot' => $this->applications['horde']['webroot'] . '/klutz',
     'icon' => $this->applications['horde']['webroot'] . '/klutz/graphics/klutz.gif',
     'name' => _("Comics"),
