Index: programs/Imakefile
===================================================================
RCS file: /cvs/xorg/xc/programs/Imakefile,v
retrieving revision 1.8
diff -u -r1.8 Imakefile
--- programs/Imakefile	2 Sep 2004 01:10:29 -0000	1.8
+++ programs/Imakefile	4 Oct 2004 21:20:41 -0000
@@ -123,7 +123,7 @@
 #endif
 
 #if BuildServersOnly || !BuildClients
-SUBDIRS = $(XSSRCDIR)
+SUBDIRS = $(XSSRCDIR) $(XDRIINFO) $(GLXCLIENTDIRS)
 #else
 SUBDIRS = \
 	appres bdftopcf bitmap \
