--- php-5.0.5/TSRM/TSRM.h	2005-03-11 06:12:07.000000000 -0500
+++ php-5.1.0RC1/TSRM/TSRM.h	2005-10-03 09:52:21.000000000 -0400
@@ -137,7 +137,7 @@
 
 #define TSRMLS_FETCH()			void ***tsrm_ls = (void ***) ts_resource_ex(0, NULL)
 #define TSRMLS_FETCH_FROM_CTX(ctx)	void ***tsrm_ls = (void ***) ctx
-#define TSRMLS_SET_CTX(ctx)		(void ***) ctx = tsrm_ls
+#define TSRMLS_SET_CTX(ctx)		ctx = (void ***) tsrm_ls
 #define TSRMG(id, type, element)	(((type) (*((void ***) tsrm_ls))[TSRM_UNSHUFFLE_RSRC_ID(id)])->element)
 #define TSRMLS_D	void ***tsrm_ls
 #define TSRMLS_DC	, TSRMLS_D
