diff -uprN mnemo-1.1/config/prefs.php.dist mnemo/config/prefs.php.dist
--- mnemo-1.1/config/prefs.php.dist	2002-12-28 14:54:07.000000000 -0800
+++ mnemo/config/prefs.php.dist	2003-08-04 10:12:35.000000000 -0700
@@ -2,7 +2,7 @@
 // $Horde: mnemo/config/prefs.php.dist,v 1.1.2.1 2002/12/28 22:54:07 jan Exp $
 
 // Make sure that constants are defined.
-require_once dirname(__FILE__) . '/../lib/constants.php';
+require_once HORDE_BASE . '/mnemo/lib/constants.php';
 
 $prefGroups['language'] = array(
     'column' => _("Your Information"),
