diff -Nur rxvt-unicode-2.8/src/iom.h rxvt-unicode-2.8-fixed/src/iom.h
--- rxvt-unicode-2.8/src/iom.h	2004-04-02 18:46:21.000000000 +0200
+++ rxvt-unicode-2.8-fixed/src/iom.h	2004-05-30 14:27:32.973560109 +0200
@@ -72,10 +72,10 @@
 
   void erase_unordered (unsigned int pos)
   {
-    watcher *w = (*this)[size () - 1];
-    pop_back ();
+    watcher *w = (*this)[this->size () - 1];
+    this->pop_back ();
 
-    if (size ())
+    if (this->size ())
       if ((*this)[pos] = w)
         w->active = pos + 1;
   }
diff -Nur rxvt-unicode-2.8/src/rxvtcolor.C rxvt-unicode-2.8-fixed/src/rxvtcolor.C
--- rxvt-unicode-2.8/src/rxvtcolor.C	2004-04-08 22:38:25.000000000 +0200
+++ rxvt-unicode-2.8-fixed/src/rxvtcolor.C	2004-05-30 14:27:44.968031348 +0200
@@ -57,7 +57,7 @@
 template<class T>
 T *refcache<T>::get (const char *id)
 {
-  for (T **i = begin (); i < end (); ++i)
+  for (T **i = this->begin (); i < this->end (); ++i)
     {
       if (!strcmp (id, (*i)->id))
         {
@@ -90,7 +90,7 @@
 
   if (!--obj->referenced)
     {
-      erase (find (begin (), end (), obj));
+      erase (find (this->begin (), this->end (), obj));
       delete obj;
     }
 }
@@ -98,8 +98,8 @@
 template<class T>
 refcache<T>::~refcache ()
 {
-  while (size ())
-    put (*begin ());
+  while (this->size ())
+    put (*this->begin ());
 }
 
 /////////////////////////////////////////////////////////////////////////////
@@ -309,7 +309,7 @@
 
 /////////////////////////////////////////////////////////////////////////////
 
-template refcache<rxvt_display>;
+template class refcache<rxvt_display>;
 refcache<rxvt_display> displays;
 
 /////////////////////////////////////////////////////////////////////////////
