diff -ruN coreutils-5.0.orig/src/copy.c coreutils-5.0/src/copy.c
--- coreutils-5.0.orig/src/copy.c	2003-03-12 05:02:23.000000000 +1100
+++ coreutils-5.0/src/copy.c	2003-11-14 19:00:06.000000000 +1100
@@ -236,7 +236,7 @@
 
   /* Compare the source dev/ino from the open file to the incoming,
      saved ones obtained via a previous call to stat.  */
-  if (! SAME_INODE (*src_sb, src_open_sb))
+  if (! SAME_INODE_ONLY (*src_sb, src_open_sb))
     {
       error (0, 0,
 	     _("skipping file %s, as it was replaced while being copied"),
diff -ruN coreutils-5.0.orig/src/sys2.h coreutils-5.0/src/sys2.h
--- coreutils-5.0.orig/src/sys2.h	2003-03-10 21:41:07.000000000 +1100
+++ coreutils-5.0/src/sys2.h	2003-11-14 18:52:50.000000000 +1100
@@ -455,6 +455,9 @@
   ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino \
    && (Stat_buf_1).st_dev == (Stat_buf_2).st_dev)
 
+#define SAME_INODE_ONLY(Stat_buf_1, Stat_buf_2) \
+ ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino)
+
 #define DOT_OR_DOTDOT(Basename) \
   (Basename[0] == '.' && (Basename[1] == '\0' \
 			  || (Basename[1] == '.' && Basename[2] == '\0')))
