diff -urN coreutils-5.3.0/src/copy.c coreutils/src/copy.c
--- coreutils-5.3.0/src/copy.c	2004-09-20 03:27:12.000000000 -0400
+++ coreutils/src/copy.c	2005-10-02 23:34:19.000000000 -0400
@@ -231,7 +231,7 @@
 
   /* Compare the source dev/ino from the open file to the incoming,
      saved ones obtained via a previous call to stat.  */
-  if (! SAME_INODE (*src_sb, src_open_sb))
+  if (! SAME_INODE_ONLY (*src_sb, src_open_sb))
     {
       error (0, 0,
 	     _("skipping file %s, as it was replaced while being copied"),
diff -urN coreutils-5.3.0/src/system.h coreutils/src/system.h
--- coreutils-5.3.0/src/system.h	2005-01-04 03:57:13.000000000 -0500
+++ coreutils/src/system.h	2005-10-02 23:34:19.000000000 -0400
@@ -564,6 +564,9 @@
   ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino \
    && (Stat_buf_1).st_dev == (Stat_buf_2).st_dev)
 
+#define SAME_INODE_ONLY(Stat_buf_1, Stat_buf_2) \
+ ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino)
+
 #define DOT_OR_DOTDOT(Basename) \
   (Basename[0] == '.' && (Basename[1] == '\0' \
 			  || (Basename[1] == '.' && Basename[2] == '\0')))
