Index: libavcodec/avcodec.h
===================================================================
RCS file: /cvsroot/ffmpeg/ffmpeg/libavcodec/avcodec.h,v
retrieving revision 1.231
diff -u -r1.231 avcodec.h
--- libavcodec/avcodec.h	15 Oct 2003 03:20:40 -0000	1.231
+++ libavcodec/avcodec.h	17 Oct 2003 12:17:37 -0000
@@ -1749,7 +1749,9 @@
 #define av_mallocz_static(p, s) __av_mallocz_static((void **)(p), s)
 
 /* add by bero : in adx.c */
+/*
 int is_adx(const unsigned char *buf,size_t bufsize);
+*/
 
 #ifdef __cplusplus
 }
