diff -u -a -b -B -r gdm-orig/configure gdm-2.8.0.0/configure
--- gdm-orig/configure	2005-06-12 08:15:51.000000000 -0500
+++ gdm-2.8.0.0/configure	2005-06-12 08:17:32.000000000 -0500
@@ -30215,7 +30215,7 @@
         HALT_COMMAND="/usr/bin/poweroff;/sbin/poweroff;/sbin/shutdown -h now;/usr/sbin/shutdown -h now"
         REBOOT_COMMAND="/usr/bin/reboot;/sbin/reboot;/sbin/shutdown -r now;/usr/sbin/shutdown -r now"
         SUSPEND_COMMAND=""
-        XSESSION_SHELL=/bin/ksh
+        XSESSION_SHELL=/bin/bash
         ;;
 esac
 
diff -u -a -b -B -r gdm-orig/configure.in gdm-2.8.0.0/configure.in
--- gdm-orig/configure.in	2005-06-12 08:15:51.000000000 -0500
+++ gdm-2.8.0.0/configure.in	2005-06-12 08:17:01.000000000 -0500
@@ -782,7 +782,7 @@
         HALT_COMMAND="/usr/bin/poweroff;/sbin/poweroff;/sbin/shutdown -h now;/usr/sbin/shutdown -h now"
         REBOOT_COMMAND="/usr/bin/reboot;/sbin/reboot;/sbin/shutdown -r now;/usr/sbin/shutdown -r now"
         SUSPEND_COMMAND=""
-        XSESSION_SHELL=/bin/ksh
+        XSESSION_SHELL=/bin/bash
         ;;
 esac
 
