--- configure.orig      2002-12-20 11:13:39.000000000 +0000
+++ configure   2002-12-20 11:14:04.000000000 +0000
@@ -14435,7 +14435,7 @@
   if test x$enable_qt = xyes; then
      PLUGINS="${PLUGINS} qt"
      ALIASES="${ALIASES} qvlc"
-     qt_LDFLAGS="${qt_LDFLAGS} -lqt -L${QTDIR}/lib"
+     qt_LDFLAGS="${qt_LDFLAGS} -lqt-mt -L${QTDIR}/lib"
      qt_CFLAGS="${qt_CFLAGS} -I/usr/include/qt -I${QTDIR}/include"
      if test -x ${QTDIR}/bin/moc
      then

