diff -Nur vlc-0.6.2/modules/codec/ffmpeg/ffmpeg.c vlc-0.6.2-ffmpeg/modules/codec/ffmpeg/ffmpeg.c
--- vlc-0.6.2/modules/codec/ffmpeg/ffmpeg.c	2003-08-08 19:08:32.000000000 +0200
+++ vlc-0.6.2-ffmpeg/modules/codec/ffmpeg/ffmpeg.c	2003-10-24 00:16:03.000000000 +0200
@@ -297,7 +297,7 @@
         b_ffmpeginit = 1;
 
         msg_Dbg( p_decoder->p_fifo, "libavcodec initialized (interface "
-                                    LIBAVCODEC_BUILD_STR ")" );
+                                    LIBAVCODEC_IDENT ")" );
     }
     else
     {
diff -Nur vlc-0.6.2/modules/codec/ffmpeg/video.c vlc-0.6.2-ffmpeg/modules/codec/ffmpeg/video.c
--- vlc-0.6.2/modules/codec/ffmpeg/video.c	2003-08-13 20:39:53.000000000 +0200
+++ vlc-0.6.2-ffmpeg/modules/codec/ffmpeg/video.c	2003-10-24 00:21:08.000000000 +0200
@@ -116,7 +116,11 @@
         i_chroma = VLC_FOURCC('I','4','2','0');
     }
 
+#if LIBAVCODEC_BUILD >= 4687
+    i_aspect = VOUT_ASPECT_FACTOR * av_q2d(p_context->sample_aspect_ratio) * i_width / i_height;
+#else
     i_aspect = VOUT_ASPECT_FACTOR * p_context->aspect_ratio;
+#endif
     if( i_aspect == 0 )
     {
         i_aspect = VOUT_ASPECT_FACTOR * i_width / i_height;
