diff -Nur vlc-0.7.1/include/vlc_common.h vlc-0.7.1-gcc34/include/vlc_common.h
--- vlc-0.7.1/include/vlc_common.h	2004-02-29 14:08:56.000000000 +0100
+++ vlc-0.7.1-gcc34/include/vlc_common.h	2004-05-09 15:58:54.000000000 +0200
@@ -456,13 +456,13 @@
 #define TAB_APPEND( count, tab, p )             \
     if( (count) > 0 )                           \
     {                                           \
-        (void *)(tab) = realloc( tab, sizeof( void ** ) * ( (count) + 1 ) ); \
+        (tab) = realloc( tab, sizeof( void ** ) * ( (count) + 1 ) ); \
     }                                           \
     else                                        \
     {                                           \
-        (void *)(tab) = malloc( sizeof( void ** ) );    \
+        (tab) = malloc( sizeof( void ** ) );    \
     }                                           \
-    ((void**)(tab))[count] = (void*)(p);        \
+    (tab)[count] = (p);                  \
     (count)++
 
 #define TAB_FIND( count, tab, p, index )        \
@@ -471,7 +471,7 @@
         (index) = -1;                           \
         for( _i_ = 0; _i_ < (count); _i_++ )    \
         {                                       \
-            if( ((void**)(tab))[_i_] == (void*)(p) )  \
+            if( (tab)[_i_] == (void*)(p) )  \
             {                                   \
                 (index) = _i_;                  \
                 break;                          \
diff -Nur vlc-0.7.1/modules/demux/livedotcom.cpp vlc-0.7.1-gcc34/modules/demux/livedotcom.cpp
--- vlc-0.7.1/modules/demux/livedotcom.cpp	2004-02-24 00:01:31.000000000 +0100
+++ vlc-0.7.1-gcc34/modules/demux/livedotcom.cpp	2004-05-09 15:59:05.000000000 +0200
@@ -623,9 +623,11 @@
 
         if( tk->p_es || tk->b_quicktime )
         {
-            TAB_APPEND( p_sys->i_track, p_sys->track, tk );
             tk->readSource = sub->readSource();
             tk->rtpSource  = sub->rtpSource();
+
+            p_sys->track = (live_track_t**)realloc( p_sys->track, sizeof( live_track_t ) * ( p_sys->i_track + 1 ) );
+            p_sys->track[p_sys->i_track++] = tk;
         }
         else
         {
diff -Nur vlc-0.7.1/modules/gui/wxwindows/preferences_widgets.cpp vlc-0.7.1-gcc34/modules/gui/wxwindows/preferences_widgets.cpp
--- vlc-0.7.1/modules/gui/wxwindows/preferences_widgets.cpp	2004-01-29 19:15:40.000000000 +0100
+++ vlc-0.7.1-gcc34/modules/gui/wxwindows/preferences_widgets.cpp	2004-05-09 15:59:23.000000000 +0200
@@ -456,7 +456,8 @@
 
     vlc_value_t val;
     wxString value = GetPszValue();
-    (const char *)val.psz_string = value.mb_str();
+    const char *myval = (const char *)val.psz_string;
+    myval = value.mb_str();
     p_item->ppf_action[i_action]( p_this, GetName().mb_str(), val, val, 0 );
 
     if( p_item->b_dirty )
diff -Nur vlc-0.7.1/modules/video_chroma/i420_rgb_mmx.h vlc-0.7.1-gcc34/modules/video_chroma/i420_rgb_mmx.h
--- vlc-0.7.1/modules/video_chroma/i420_rgb_mmx.h	2003-10-01 05:38:24.000000000 +0200
+++ vlc-0.7.1-gcc34/modules/video_chroma/i420_rgb_mmx.h	2004-05-09 15:59:45.000000000 +0200
@@ -25,7 +25,7 @@
 
 /* hope these constant values are cache line aligned */
 #define UNUSED_U64(foo) \
-    static const uint64_t foo __asm__ (#foo) __attribute__((unused))
+    const uint64_t foo __asm__ (#foo) __attribute__((unused))
 UNUSED_U64(mmx_80w)     = 0x0080008000800080ULL;
 UNUSED_U64(mmx_10w)     = 0x1010101010101010ULL;
 UNUSED_U64(mmx_00ffw)   = 0x00ff00ff00ff00ffULL;
