diff -r -U2 cheops-ng-0.1.10.ori/Makefile.in cheops-ng-0.1.10/Makefile.in
--- cheops-ng-0.1.10.ori/Makefile.in	2002-08-27 05:58:08.000000000 +0200
+++ cheops-ng-0.1.10/Makefile.in	2002-08-27 10:43:19.000000000 +0200
@@ -5,5 +5,5 @@
 DEBUG_FLAGS=# -DDEBUG_DISCOVER -DDEBUG_EVENT -DDEBUG_IO -DDEBUG_SCHEDULER
 
-CFLAGS = @CFLAGS@ -Iadns-1.0/src
+CFLAGS = @CFLAGS@ 
 LIBS = @LIBS@
 
@@ -50,6 +50,6 @@
             agent-map.o probe.o agent-probe.o agent-auth.o agent-nmapxml.o
 	
-AGENTLIBS = ./adns-1.0/src/libadns.a
-AGENTLIBFLAGS = -lm $(LIBS) -L./adns-1.0/src/ -lcrypt
+AGENTLIBS = 
+AGENTLIBFLAGS = -lm $(LIBS) -lcrypt
 
 # Objects for Cheops Agent Shell
@@ -74,5 +74,5 @@
 COMLIBS = #@COMLIBS@
 CC_LIBS = 
-CFLAGS += -Wall -g -Iadns-1.0/src $(DEBUG_FLAGS) $(GTKCFLAGS)  $(XMLCFLAGS)
+CFLAGS += -Wall -g $(DEBUG_FLAGS) $(GTKCFLAGS)  $(XMLCFLAGS)
 
 ifdef DEBUG
@@ -84,5 +84,5 @@
 #PROGS = test-sched test-io cheops-sh
 
-all: version.h installdir.h  ./adns-1.0/src/libadns.a $(PROGS)
+all: version.h installdir.h $(PROGS)
 
 strip: cheops-ng cheops-agent
@@ -105,12 +105,6 @@
 	echo "#endif" >> version.h
 
-./adns-1.0/src/Makefile: ./adns-1.0/src/Makefile.in
-	cd adns-1.0; ./configure
-
-./adns-1.0/src/libadns.a: ./adns-1.0/src/Makefile
-	$(MAKE) -C ./adns-1.0/src/
-
 cheops-ng: version.h $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS)
-	$(CC) -o cheops-ng $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(CC_LIBS) ./adns-1.0/src/libadns.a $(STATIC_FLAGS)
+	$(CC) -o cheops-ng $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(CC_LIBS) $(STATIC_FLAGS)
 
 cheops-agent: version.h $(AGENTOBJS) $(AGENTLIBS) $(COMOBJS) $(NORMOBJS) $(COMLIBS)
