diff -Nur xmovie-1.9.8/build/Makefile.xmovie xmovie-1.9.8-shared/build/Makefile.xmovie
--- xmovie-1.9.8/build/Makefile.xmovie	2003-02-10 04:12:18.000000000 +0100
+++ xmovie-1.9.8-shared/build/Makefile.xmovie	2003-04-25 22:16:35.000000000 +0200
@@ -1,9 +1,7 @@
 include global_config
 
 DIRS := \
-	libmpeg3 \
 	libsndfile \
-	quicktime \
 	guicast \
 	xmovie \
 
diff -Nur xmovie-1.9.8/guicast/bctheme.C xmovie-1.9.8-shared/guicast/bctheme.C
--- xmovie-1.9.8/guicast/bctheme.C	2002-12-21 04:01:43.000000000 +0100
+++ xmovie-1.9.8-shared/guicast/bctheme.C	2003-04-25 22:17:57.000000000 +0200
@@ -86,7 +86,7 @@
 }
 
 
-void BC_Theme::overlay(VFrame *dst, VFrame *src, int in_x1 = -1, int in_x2 = -1)
+void BC_Theme::overlay(VFrame *dst, VFrame *src, int in_x1, int in_x2)
 {
 	int w;
 	int h;
diff -Nur xmovie-1.9.8/xmovie/Makefile xmovie-1.9.8-shared/xmovie/Makefile
--- xmovie-1.9.8/xmovie/Makefile	2003-01-29 07:33:17.000000000 +0100
+++ xmovie-1.9.8-shared/xmovie/Makefile	2003-04-25 22:22:17.000000000 +0200
@@ -36,11 +36,9 @@
 	-I../libsndfile
 
 LIBS += -L/usr/X11R6/lib \
-	../libmpeg3/$(OBJDIR)/libmpeg3.a \
-	../quicktime/$(OBJDIR)/libquicktime.a \
 	../guicast/$(OBJDIR)/libguicast.a \
 	../libsndfile/$(OBJDIR)/libsndfile.a \
- 	$(GLLIBS) -lX11 -lXext -lXv -lpthread -lglib -lm -lpng -lz -ldl $(XXF86VM)
+ 	$(GLLIBS) -lX11 -lXext -lXv -lpthread -lglib -lm -lpng -lz -ldl $(XXF86VM) -lquicktime -lmpeg3
 
 # Static libraries for distribution
 ifeq ($(USE_STATIC),y)
