diff -u -r1.18.2.2 prefix.linux
--- builds/posix/prefix.linux	9 Oct 2003 23:08:34 -0000	1.18.2.2
+++ builds/posix/prefix.linux	31 Oct 2003 19:18:56 -0000
@@ -22,7 +22,7 @@
 
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-O3 -march=i586 -mcpu=i686 -fomit-frame-pointer -fno-builtin -DNDEBUG -DLINUX -pipe -MMD -fPIC
+PROD_FLAGS=@CXXFLAGS@ -DNDEBUG -DLINUX -pipe -MMD -fPIC
 DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
 
 OS_ServerFiles=inet_server.cpp
