--- src/fe-text/fe-text.c	2004-11-21 17:02:08.000000000 +0000
+++ src/fe-text/fe-text.c	2004-11-21 17:03:34.000000000 +0000
@@ -815,3 +815,13 @@
 void fe_confirm (const char *message, void (*yesproc)(void *), void (*noproc)(void *), void *ud)
 {
 }
+char *fe_get_inputbox_contents (struct session *sess)
+{
+        return NULL;
+}
+void fe_set_inputbox_contents (struct session *sess, char *text)
+{
+}
+void fe_set_inputbox_cursor (struct session *sess, int delta, int pos)
+{
+}
