diff -urpN httpd-2.0.52.orig/modules/ssl/Makefile.in httpd-2.0.52/modules/ssl/Makefile.in
--- httpd-2.0.52.orig/modules/ssl/Makefile.in	2004-11-30 17:16:06.729373400 -0800
+++ httpd-2.0.52/modules/ssl/Makefile.in	2004-11-30 17:19:48.781616328 -0800
@@ -23,10 +23,6 @@ include $(top_srcdir)/build/special.mk
 #   (we really don't expect end users to use these targets!)
 #
 
-ssl_expr_scan.c: $(top_srcdir)/modules/ssl/ssl_expr_scan.l ssl_expr_parse.h
-	flex -Pssl_expr_yy -s -B $(top_srcdir)/modules/ssl/ssl_expr_scan.l
-	sed -e '/$$Header:/d' <lex.ssl_expr_yy.c >ssl_expr_scan.c && rm -f lex.ssl_expr_yy.c
-
 ssl_expr_parse.c ssl_expr_parse.h: $(top_srcdir)/modules/ssl/ssl_expr_parse.y
 	yacc -d $(top_srcdir)/modules/ssl/ssl_expr_parse.y
 	sed -e 's;yy;ssl_expr_yy;g' \
