diff -Nur motion-3.0.5-orig/ffmpeg.c motion-3.0.5/ffmpeg.c
--- motion-3.0.5-orig/ffmpeg.c	2002-12-14 17:18:34.000000000 +0100
+++ motion-3.0.5/ffmpeg.c	2003-05-14 19:33:41.000000000 +0200
@@ -55,7 +55,7 @@
 	c->height = height;
 
 	/* frames per second */
-	c->frame_rate = rate * FRAME_RATE_BASE;  /* default rate is only 5 frames per second */
+	c->frame_rate = rate * DEFAULT_FRAME_RATE_BASE;  /* default rate is only 5 frames per second */
 	c->gop_size = 10; /* emit one intra frame every ten frames */
 //	c->qmin = 3;
 //	c->max_qdiff = 3;
@@ -78,7 +78,7 @@
 	/* alloc image and output buffer */
 
 	ffmpeg->outbuf_size = width * height * 10; // large outbuf_buf
-	if ((ffmpeg->outbuf = (UINT8*) malloc(ffmpeg->outbuf_size)) == NULL) return NULL;
+	if ((ffmpeg->outbuf = (uint8_t*) malloc(ffmpeg->outbuf_size)) == NULL) return NULL;
 
 	ffmpeg->picture->data[0] = y;
 	ffmpeg->picture->data[1] = u;
diff -Nur motion-3.0.5-orig/ffmpeg.h motion-3.0.5/ffmpeg.h
--- motion-3.0.5-orig/ffmpeg.h	2002-12-14 17:18:34.000000000 +0100
+++ motion-3.0.5/ffmpeg.h	2003-05-14 19:31:48.000000000 +0200
@@ -11,7 +11,7 @@
 	AVFrame *picture;
 	int outbuf_size;
 	FILE *f;
-	UINT8 *outbuf;
+	uint8_t *outbuf;
 	void *udata;
 #else
 	int dummy;
