--- Makefile-old	2003-10-28 04:43:23.000000000 +0100
+++ Makefile	2003-11-13 13:21:23.000000000 +0100
@@ -50,11 +50,7 @@
 
 
 
-include Makefile.jpeg
-include Makefile.vorbis
 include Makefile.ffmpeg
-include Makefile.lame
-include Makefile.dv
 include Makefile.encore50
 
 
@@ -68,7 +64,9 @@
 CFLAGS += -I. \
 	-I$(JPEG_DIR) \
 	-I$(MPEG3_DIR) \
-	-I$(LAME_DIR)/include
+	-I$(LAME_DIR)/include \
+	-I/usr/include/libmpeg3 \
+	-I/usr/include/lame
 
 
 #	$(OBJDIR)/div3.o \
@@ -222,7 +220,7 @@
 	$(OBJDIR)/$(ENCORE_DIR)/mbtransquant/intel_mmx
 
 
-LIBS := $(OUTPUT) $(MPEG3_DIR)/$(OBJDIR)/libmpeg3.a -lpng -lz -lpthread -ldl -lm
+LIBS := $(OUTPUT) -lmpeg3 -lraw1394 -lmp3lame -logg -lvorbis -lvorbisenc -ldv -ljpeg -lpng -lz -lpthread -ldl -lm
 
 
 $(shell if ! test -d $(OBJDIR) \; then mkdir -p $(DIRS) \; fi )
@@ -232,12 +230,14 @@
 $(shell echo  $(OBJS) $(DV_OBJS) $(DVMMXOBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(JPEG_FAIL) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) > $(OBJDIR)/objs)
 #$(shell echo  $(OBJS) $(DV_OBJS) $(DVMMXOBJS) $(DECOREMMX_OBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(JPEG_FAIL) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) > $(OBJDIR)/objs)
 
-all: $(OBJDIR) $(OUTPUT)
+all: $(OBJDIR) $(OUTPUT) $(SHAREDOUTPUT) $(UTILS)
 
 #$(OUTPUT): $(DV_OBJS) $(DVMMXOBJ) $(OBJS) $(DVMMXOBJS) $(DECOREMMX_OBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) $(JPEG_FAIL) 
 $(OUTPUT): $(DV_OBJS) $(DVMMXOBJ) $(OBJS) $(DVMMXOBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) $(JPEG_FAIL) 
 	ar rcs $(OUTPUT) `cat $(OBJDIR)/objs`
 
+$(SHAREDOUTPUT): $(OUTPUT)
+	ld -shared -o $(SHAREDOUTPUT) `cat $(OBJDIR)/objs` $(LIBS)
 
 $(OBJDIR)/qtdump: $(OBJDIR)/dump.o $(OUTPUT)
 	$(CC) -o $(OBJDIR)/qtdump $(OBJDIR)/dump.o $(LIBS)
@@ -274,7 +274,11 @@
 	cd .. && \
 	tar zcvf quicktime.tar.gz quicktime
 
-install:
+install: $(OUTPUT) $(SHAREDOUTPUT) $(UTILS)
+	mkdir -p /usr/include/quicktime
+	cp quicktime.h qtprivate.h colormodels.h /usr/include/quicktime
+	cp i686/libquicktime.* /usr/lib
+	ldconfig
 	cp $(UTILS) /usr/bin
 
 $(OBJDIR)/dump.o \
