--- flamerobin.bkl	2005-08-13 13:29:37.000000000 +0000
+++ flamerobin.bkl.cp	2005-08-14 10:31:28.000000000 +0000
@@ -212,17 +212,17 @@
 
     <set var="IBPPDYNLIBRARY">
         <if cond="PLATFORM_WIN32=='1'">ibpp</if>
+        <if cond="PLATFORM_UNIX=='1'">ibpp</if>
     </set>
     <set var="IBPPSTATICLIBRARY">
         <if cond="PLATFORM_MACOSX=='1'">$(IBPPLIBPATH)/libibpp.a</if>
-        <if cond="PLATFORM_UNIX=='1'">$(IBPPLIBPATH)/libibpp.a</if>
     </set>
 
     <set var="FIREBIRDFRAMEWORK">
         <if cond="PLATFORM_MACOSX=='1'">-framework Firebird</if>  -->
     </set>
     <set var="FIREBIRDLIBRARY">
-        <if cond="PLATFORM_UNIX=='1'">fbclient</if>
+        <!-- PLATFORM_UNIX: ibpp library loads firebird DLL dynamically -->
         <!-- PLATFORM_WIN32: ibpp library loads firebird DLL dynamically -->
     </set>
 
