diff -ruN libggi-2.0.3.orig/default/linear_16/crossblit.c libggi-2.0.3/default/linear_16/crossblit.c
--- libggi-2.0.3.orig/default/linear_16/crossblit.c	2002-10-28 15:57:04.000000000 +1100
+++ libggi-2.0.3/default/linear_16/crossblit.c	2003-06-25 02:04:29.000000000 +1000
@@ -1278,7 +1278,7 @@
 				while (stopcol > dstp + 11) {
 				  void *dummy;
 				  __asm__ __volatile__(
-				   "movq  (%1), %%mm0\n\t
+				    movq  (%1), %%mm0\n\t
                                     pxor  %%mm5, %%mm5\n\t
                                     movq  8(%1), %%mm1\n\t
                                     pxor  %%mm6, %%mm6\n\t
@@ -1339,7 +1339,7 @@
                                     jmp   .Lright%=\n\t
 
                                     .Lout%=:\n\t
-                                    emms\n\t"
+                                    emms\n\t
 				   : "=qom" (dstp), "=qom" (srcp), "=q" (dummy)
 				   : "qom" (dstp), "qom" (srcp), "q" (tab)
 				   : "cc", "memory");
@@ -1438,7 +1438,7 @@
 				  void *dummy;
 				  __asm__ __volatile__(
 
-				   "movq  (%1), %%mm0\n\t
+				    movq  (%1), %%mm0\n\t
                                     movq  8(%1), %%mm1\n\t
                                     movq  16(%1), %%mm2\n\t
                                     movq  24(%1), %%mm3\n\t
@@ -1505,7 +1505,7 @@
                                     jmp   .Lright%=\n\t
 
                                     .Lout%=:\n\t
-                                    emms\n\t"
+                                    emms\n\t
 				   : "=qom" (dstp), "=qom" (srcp), "=q" (dummy)
 				   : "qom" (dstp), "qom" (srcp), "q" (tab)
 				   : "cc", "memory");
