--- patch-1.4.dgc.deepif.1	2002-06-18 12:32:50.000000000 -0500
+++ patch-1.4.dgc.deepif.1.smgl	2005-03-19 19:20:18.339039032 -0600
@@ -1,3 +1,8 @@
+diff -Pur mutt-1.4-base/PATCHES mutt-1.4/PATCHES
+--- mutt-1.4-base/PATCHES	Mon Mar 25 05:28:58 2002
++++ mutt-1.4/PATCHES	Fri Jun 14 10:59:54 2002
+@@ -1,0 +1 @@
++patch-1.4.dgc.deepif.1
 diff -Pur mutt-1.4-base/muttlib.c mutt-1.4/muttlib.c
 --- mutt-1.4-base/muttlib.c	Mon Mar 25 05:28:58 2002
 +++ mutt-1.4/muttlib.c	Fri Jun 14 10:59:54 2002
