--- mpich-1.2.4/configure   Fri May 10 01:30:11 2002
+++ mpich-1.2.4/configure   Fri May 10 01:29:40 2002
@@ -563,6 +563,9 @@
      -srcdir | --srcdir | --srcdi | --srcd | --src | --sr | --s)
        next_srcdir=yes ;;

+     -sysconfdir=* | --sysconfdir=* )
+       sysconfdir=`echo $arg | sed 's/[-a-z_]*=//'` ;;
+
      -datadir=* | --datadir=* )
        datadir=`echo $arg | sed 's/[-a-z_]*=//'` ;;

