diff -uprN fbdesk-1.1.4.old/src/Icon.cc fbdesk-1.1.4/src/Icon.cc
--- fbdesk-1.1.4.old/src/Icon.cc	2003-12-08 10:59:37.535895808 -0800
+++ fbdesk-1.1.4/src/Icon.cc	2003-12-08 11:01:13.773265504 -0800
@@ -36,6 +36,8 @@
 #include <X11/xpm.h>
 #include <X11/Xatom.h>
 
+#include <cassert>
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif // HAVE_CONFIG_H
diff -uprN fbdesk-1.1.4.old/src/InputField.cc fbdesk-1.1.4/src/InputField.cc
--- fbdesk-1.1.4.old/src/InputField.cc	2003-12-08 10:59:37.543894592 -0800
+++ fbdesk-1.1.4/src/InputField.cc	2003-12-08 11:01:19.720361408 -0800
@@ -27,6 +27,9 @@
 
 #include <X11/Xutil.h>
 #include <X11/keysym.h>
+
+#include <cassert>
+
 namespace {
 
 void setNoResize(FbTk::FbWindow &win) {
