diff -Nur kaffe-1.0.6/libraries/clib/native/System.c kaffe-1.0.6-org/libraries/clib/native/System.c
--- kaffe-1.0.6/libraries/clib/native/System.c	Fri Mar 22 14:41:30 2002
+++ kaffe-1.0.6-org/libraries/clib/native/System.c	Thu Jun 22 16:17:11 2000
@@ -20,8 +20,8 @@
 #if defined(HAVE_PWD_H)
 #include <pwd.h>
 #endif
+#include <time.h>
 #if defined(HAVE_TIME_H)
-#include <time.h>
 #endif
 #include <ctype.h>
 #include "../../../kaffe/kaffevm/classMethod.h"
