==== DB_File/DB_File.xs#60 (text) ====
Index: DB_File/DB_File.xs
--- DB_File/DB_File.xs   Wed Sep  4 06:35:36 2002
+++ DB_File/DB_File.xs   Wed Sep  4 07:32:19 2002
@@ -774,6 +774,7 @@
 char * buffer;
 #endif
 {
+    dTHX;
     SV * sv = perl_get_sv(ERR_BUFF, FALSE) ;
     if (sv) {
         if (db_errpfx)
@@ -1455,6 +1456,7 @@
 
 BOOT:
   {
+    dTHX;
     SV * sv_err = perl_get_sv(ERR_BUFF, GV_ADD|GV_ADDMULTI) ;    
     MY_CXT_INIT;
     __getBerkeleyDBInfo() ;
