diff -u -r1.770.2.27 khtml_part.cpp
--- khtml_part.cpp	3 Jul 2003 17:13:02 -0000	1.770.2.27
+++ khtml_part.cpp	10 Jul 2003 11:45:10 -0000
@@ -4171,7 +4171,27 @@
 
 QString KHTMLPart::referrer() const
 {
-   return d->m_pageReferrer;
+   return d->m_referrer;
+}
+
+QString KHTMLPart::pageReferrer() const
+{
+   KURL referrerURL = d->m_pageReferrer;
+   if (referrerURL.isValid())
+   {
+      QString protocol = referrerURL.protocol();
+
+      if ((protocol == "http") ||
+         ((protocol == "https") && (m_url.protocol() == "https")))
+      {
+          referrerURL.setRef(QString::null);
+          referrerURL.setUser(QString::null);
+          referrerURL.setPass(QString::null);
+          return referrerURL.url();
+      }
+   }
+   
+   return QString::null;
 }
 
 QString KHTMLPart::lastModified() const
