diff -urNad /home/ultrotter/docs/projects/deb/directory-administrator-1.5.1/src/profile.c directory-administrator-1.5.1/src/profile.c
--- /home/ultrotter/docs/projects/deb/directory-administrator-1.5.1/src/profile.c	2003-06-12 14:52:09.000000000 +0200
+++ directory-administrator-1.5.1/src/profile.c	2003-06-24 12:00:56.000000000 +0200
@@ -308,6 +308,10 @@
 #   endif
   }
 
+ version = LDAP_VERSION3;
+    if (ldap_set_option(connectionhandler, LDAP_OPT_PROTOCOL_VERSION,
+        &version)!= LDAP_OPT_SUCCESS) g_print("Could not set protocol version 3, continuing anyway\n" );
+
   if (userid && strlen(userid) > 0 && strchr(userid,'=') == NULL) {
    // nos pasaron un UNIX user ID - tratemos de buscarlo
      ldap_errors = ldap_simple_bind_s (connectionhandler, NULL, NULL);
@@ -347,9 +351,6 @@
   ldap_errors = ldap_simple_bind_s (connectionhandler,
 				    userid,
 				    connection_profile_get_password (conn));
-    version = LDAP_VERSION3;
-    if (ldap_set_option(connectionhandler, LDAP_OPT_PROTOCOL_VERSION,
-        &version)!= LDAP_OPT_SUCCESS) g_print("Could not set protocol version 3, continuing anyway\n" );
 
   if (ldap_errors)
     {
