diff -Nur xpdf-3.00.orig/configure xpdf-3.00/configure
--- xpdf-3.00.orig/configure	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/configure	2004-07-10 20:36:15.209988753 +0200
@@ -9124,6 +9124,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
+#include <ft2build.h>
 #include <$ac_header>
 _ACEOF
 rm -f conftest.$ac_objext
@@ -9159,6 +9160,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
+#include <ft2build.h>
 #include <$ac_header>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
diff -Nur xpdf-3.00.orig/splash/SplashFTFontEngine.h xpdf-3.00/splash/SplashFTFontEngine.h
--- xpdf-3.00.orig/splash/SplashFTFontEngine.h	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/splash/SplashFTFontEngine.h	2004-07-10 20:36:28.256325926 +0200
@@ -15,7 +15,8 @@
 #pragma interface
 #endif
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "gtypes.h"
 
 class SplashFontFile;
diff -Nur xpdf-3.00.orig/splash/SplashFTFontFile.h xpdf-3.00/splash/SplashFTFontFile.h
--- xpdf-3.00.orig/splash/SplashFTFontFile.h	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/splash/SplashFTFontFile.h	2004-07-10 20:36:28.258325671 +0200
@@ -15,7 +15,8 @@
 #pragma interface
 #endif
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "SplashFontFile.h"
 
 class SplashFontFileID;
diff -Nur xpdf-3.00.orig/splash/SplashFTFont.h xpdf-3.00/splash/SplashFTFont.h
--- xpdf-3.00.orig/splash/SplashFTFont.h	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/splash/SplashFTFont.h	2004-07-10 20:36:28.261325289 +0200
@@ -15,7 +15,8 @@
 #pragma interface
 #endif
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "SplashFont.h"
 
 class SplashFTFontFile;
