--- src/frontend/common/TemplateBuilder.hh.orig	2004-11-13 10:26:24.112111056 -0600
+++ src/frontend/common/TemplateBuilder.hh	2004-11-13 10:28:42.109132320 -0600
@@ -790,10 +790,18 @@
 			 const SmartPtr<Value>& tableAlign,
 			 unsigned columnIndex)
     {
-      return
+      if (cellAlign)
+	return cellAlign;
+      else if (rowAlign)
+	return GetComponent(rowAlign, columnIndex);
+      else if (tableAlign)
+	return GetComponent(tableAlign, columnIndex);
+      else
+	return 0;
+/*      return
 	(cellAlign ? cellAlign : 
 	 (rowAlign ? GetComponent(rowAlign, columnIndex) :
-	  (tableAlign ? GetComponent(tableAlign, columnIndex) : 0)));
+	  (tableAlign ? GetComponent(tableAlign, columnIndex) : 0))); */
     }
 
     static void
