--- configure.old	Wed Mar 20 13:55:34 2002
+++ configure	Wed Mar 20 13:55:45 2002
@@ -4609,9 +4609,9 @@
 
   if test x"$enable_ltdl_install" = x"yes"; then
-    ac_configure_args="$ac_configure_args --enable-ltdl-install"
+    ac_configure_args="--enable-ltdl-install"
     LIBLTDL='${top_builddir}/''libltdl'/libltdl.la
     LTDLINCL='-I${top_srcdir}/''libltdl'
   else
-    ac_configure_args="$ac_configure_args --enable-ltdl-install=no"
+    ac_configure_args="--enable-ltdl-install=no"
     LIBLTDL="-lltdl"
     LTDLINCL=
